<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Amino Acid Tools | GTO 2: The genomics-proteomics toolkit</title>
  <meta name="description" content="To do" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Amino Acid Tools | GTO 2: The genomics-proteomics toolkit" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="To do" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Amino Acid Tools | GTO 2: The genomics-proteomics toolkit" />
  
  <meta name="twitter:description" content="To do" />
  

<meta name="author" content="J. R. Almeida, A. J. Pinho, J. L. Oliveira, D. Pratas" />


<meta name="date" content="2021-04-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="fastq-tools.html"/>
<link rel="next" href="genomic-tools.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="/">GTO 2: The genomics-proteomics toolkit</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#testing"><i class="fa fa-check"></i><b>1.2</b> Testing</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#execution-control"><i class="fa fa-check"></i><b>1.3</b> Execution control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="fasta-tools.html"><a href="fasta-tools.html"><i class="fa fa-check"></i><b>2</b> FASTA Tools</a><ul>
<li class="chapter" data-level="2.1" data-path="fasta-tools.html"><a href="fasta-tools.html#gto2_fa_to_fq"><i class="fa fa-check"></i><b>2.1</b> gto2_fa_to_fq</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="fastq-tools.html"><a href="fastq-tools.html"><i class="fa fa-check"></i><b>3</b> FASTQ Tools</a><ul>
<li class="chapter" data-level="3.1" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_to_fa"><i class="fa fa-check"></i><b>3.1</b> Program gto2_fq_to_fa</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_to_mfa"><i class="fa fa-check"></i><b>3.2</b> Program gto2_fq_to_mfa</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-1"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-1"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_exclude_n"><i class="fa fa-check"></i><b>3.3</b> Program gto2_fq_exclude_n</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-2"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-2"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_extract_quality_scores"><i class="fa fa-check"></i><b>3.4</b> Program gto2_fq_extract_quality_scores</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-3"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-3"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_info"><i class="fa fa-check"></i><b>3.5</b> Program gto2_fq_info</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-4"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-4"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_maximum_read_size"><i class="fa fa-check"></i><b>3.6</b> Program gto2_fq_maximum_read_size</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-5"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-5"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_minimum_quality_score"><i class="fa fa-check"></i><b>3.7</b> Program gto2_fq_minimum_quality_score</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-6"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-6"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_minimum_read_size"><i class="fa fa-check"></i><b>3.8</b> Program gto2_fq_minimum_read_size</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-7"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-7"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_rand_extra_chars"><i class="fa fa-check"></i><b>3.9</b> Program gto2_fq_rand_extra_chars</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-8"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-8"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.10" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_from_seq"><i class="fa fa-check"></i><b>3.10</b> Program gto2_fq_from_seq</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-9"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-9"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.11" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_mutate"><i class="fa fa-check"></i><b>3.11</b> Program gto2_fq_mutate</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-10"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-10"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.12" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_split"><i class="fa fa-check"></i><b>3.12</b> Program gto2_fq_split</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-11"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-11"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.13" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_pack"><i class="fa fa-check"></i><b>3.13</b> Program gto2_fq_pack</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-12"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-12"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.14" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_unpack"><i class="fa fa-check"></i><b>3.14</b> Program gto2_fq_unpack</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-13"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-13"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.15" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_quality_score_info"><i class="fa fa-check"></i><b>3.15</b> Program gto2_fq_quality_score_info</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-14"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-14"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.16" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_quality_score_min"><i class="fa fa-check"></i><b>3.16</b> Program gto2_fq_quality_score_min</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-15"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-15"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.17" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_quality_score_max"><i class="fa fa-check"></i><b>3.17</b> Program gto2_fq_quality_score_max</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-16"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-16"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.18" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_cut"><i class="fa fa-check"></i><b>3.18</b> Program gto2_fq_cut</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-17"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-17"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.19" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_minimum_local_quality_score_forward"><i class="fa fa-check"></i><b>3.19</b> Program gto2_fq_minimum_local_quality_score_forward</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-18"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-18"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.20" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_minimum_local_quality_score_reverse"><i class="fa fa-check"></i><b>3.20</b> Program gto2_fq_minimum_local_quality_score_reverse</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-19"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-19"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.21" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_xs"><i class="fa fa-check"></i><b>3.21</b> Program gto2_fq_xs</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-20"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-20"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.22" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_complement"><i class="fa fa-check"></i><b>3.22</b> Program gto2_fq_complement</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-21"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-21"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.23" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_reverse"><i class="fa fa-check"></i><b>3.23</b> Program gto2_fq_reverse</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-22"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-22"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.24" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_variation_map"><i class="fa fa-check"></i><b>3.24</b> Program gto2_fq_variation_map</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-23"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-23"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.25" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_variation_filter"><i class="fa fa-check"></i><b>3.25</b> Program gto2_fq_variation_filter</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-24"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-24"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.26" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_variation_visual"><i class="fa fa-check"></i><b>3.26</b> Program gto2_fq_variation_visual</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-25"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-25"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.27" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_metagenomics"><i class="fa fa-check"></i><b>3.27</b> Program gto2_fq_metagenomics</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-26"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-26"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html"><i class="fa fa-check"></i><b>4</b> Amino Acid Tools</a><ul>
<li class="chapter" data-level="4.1" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#program-gto2_aa_to_group"><i class="fa fa-check"></i><b>4.1</b> Program gto2_aa_to_group</a><ul>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#input-parameters-27"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#output-27"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#program-gto2_aa_to_pseudo_dna"><i class="fa fa-check"></i><b>4.2</b> Program gto2_aa_to_pseudo_dna</a><ul>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#input-parameters-28"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#output-28"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#program-gto2_aa_compressor"><i class="fa fa-check"></i><b>4.3</b> Program gto2_aa_compressor</a></li>
<li class="chapter" data-level="4.4" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#program-gto2_aa_from_fa"><i class="fa fa-check"></i><b>4.4</b> Program gto2_aa_from_fa</a><ul>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#input-parameters-29"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#output-29"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#program-gto2_aa_from_fq"><i class="fa fa-check"></i><b>4.5</b> Program gto2_aa_from_fq</a><ul>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#input-parameters-30"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#output-30"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#program-gto2_aa_from_seq"><i class="fa fa-check"></i><b>4.6</b> Program gto2_aa_from_seq</a><ul>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#input-parameters-31"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#output-31"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genomic-tools.html"><a href="genomic-tools.html"><i class="fa fa-check"></i><b>5</b> Genomic Tools</a><ul>
<li class="chapter" data-level="5.1" data-path="genomic-tools.html"><a href="genomic-tools.html#gto2_dna"><i class="fa fa-check"></i><b>5.1</b> gto2_dna</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="general-purpose-tools.html"><a href="general-purpose-tools.html"><i class="fa fa-check"></i><b>6</b> General Purpose Tools</a><ul>
<li class="chapter" data-level="6.1" data-path="general-purpose-tools.html"><a href="general-purpose-tools.html#gto2_"><i class="fa fa-check"></i><b>6.1</b> gto2_</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GTO 2: The genomics-proteomics toolkit</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="amino-acid-tools" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Amino Acid Tools</h1>
<p>A more specific subset of tools is the Amino Acid Sequence tools, designed to manipulate amino acid sequences. The main features of those tools are grouping sequences, for instance by their properties, such as electric charge (positive and negative), uncharged side chains, hydrophobic side chains and special cases. It is also possible generating pseudo-DNA with characteristics passed by amino acid sequences, or for data compression, using cooperation between multiple contexts and substitutional tolerant context models. The current available amino acid sequence tools, for analysis and manipulation, are:</p>
<ul>
<li><strong>gto2_aa_to_group</strong>: to convert an amino acid sequence to a group sequence.</li>
<li><strong>gto2_aa_to_pseudo_dna</strong>: to convert an amino acid (protein) sequence to a pseudo DNA sequence.</li>
<li><strong>gto2_aa_compressor</strong>: a new lossless compressor to compress efficiently amino acid sequences (proteins).</li>
<li><strong>gto2_aa_from_fa</strong>: to convert DNA sequences in FASTA or Multi-FASTA file format to an amino acid sequence.</li>
<li><strong>gto2_aa_from_fq</strong>: to convert DNA sequences in the FASTQ file format to an amino acid sequence.</li>
<li><strong>gto2_aa_from_seq</strong>: to convert DNA sequences to an amino acid sequence.</li>
</ul>
<div id="program-gto2_aa_to_group" class="section level2">
<h2><span class="header-section-number">4.1</span> Program gto2_aa_to_group</h2>
<p>The <strong>gto2_aa_to_group</strong> converts an amino acid sequence to a group sequence.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_aa_to_group</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-27" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_aa_to_group</strong> program needs two streams for the computation, namely the input and output standard. The input stream is an amino acid sequence. The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_aa_to_group [options] [[--] args]
   <span class="ex">or</span>: ./gto2_aa_to_group [options]

<span class="ex">It</span> converts a amino acid sequence to a group sequence.

    <span class="ex">-h</span>, --help        show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.prot</span>      Input amino acid sequence file (stdin)
    <span class="op">&gt;</span> <span class="ex">output.group</span>    Output group sequence file (stdout)

<span class="ex">Example</span>: ./gto2_aa_to_group <span class="op">&lt;</span> input.prot <span class="op">&gt;</span> output.group
<span class="ex">Table</span>:
<span class="ex">Prot</span>    Group
<span class="ex">R</span>       P
<span class="ex">H</span>       P  Electric charged side chains: POSITIVE
<span class="ex">K</span>       P
<span class="ex">-</span>       -
<span class="ex">D</span>       N
<span class="ex">E</span>       N  Electric charged side chains: NEGATIVE
<span class="ex">-</span>       -
<span class="ex">S</span>       U
<span class="ex">T</span>       U
<span class="ex">N</span>       U  Amino acids with electric UNCHARGED side chains
<span class="ex">Q</span>       U
<span class="ex">-</span>       -
<span class="ex">C</span>       S
<span class="ex">U</span>       S
<span class="ex">G</span>       S  Special cases
<span class="ex">P</span>       S
<span class="ex">-</span>       -
<span class="ex">A</span>       H
<span class="ex">V</span>       H
<span class="ex">I</span>       H
<span class="ex">L</span>       H
<span class="ex">M</span>       H  Amino acids with hydrophobic side chains
<span class="ex">F</span>       H
<span class="ex">Y</span>       H
<span class="ex">W</span>       H
<span class="ex">-</span>       -
<span class="ex">*</span>       *  Others
<span class="ex">X</span>       X  Unknown</code></pre></div>
<p>This tool can be used to group amino acids by properties, such as electric charge (positive and negative), uncharged side chains, hydrophobic side chains and special cases. An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">IPFLLKKQFALADKLVLSKLRQLLGGRIKMMPCGGAKLEPAIGLFFHAIGINIKLGYGMT</span>
<span class="ex">ETTATVSCWHDFQFNPNSIGTLMPKAEVKIGENNEILVRGGMVMKGYYKKPEETAQAFTE</span>
<span class="ex">DGFLKTGDAGEFDEQGNLFITDRIKELMKTSNGKYIAPQYIESKIGKDKFIEQIAIIADA</span>
<span class="ex">KKYVSALIVPCFDSLEEYAKQLNIKYHDRLELLKNSDILKMFE</span></code></pre></div>
</div>
<div id="output-27" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_aa_to_group</strong> program is a group sequence. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">HSHHHPPUHHHHNPHHHUPHPUHHSSPHPHHSSSSHPHNSHHSHHHPHHSHUHPHSHSHU</span>
<span class="ex">NUUHUHUSHPNHUHUSUUHSUHHSPHNHPHSNUUNHHHPSSHHHPSHHPPSNNUHUHHUN</span>
<span class="ex">NSHHPUSNHSNHNNUSUHHHUNPHPNHHPUUUSPHHHSUHHNUPHSPNPHHNUHHHHHNH</span>
<span class="ex">PPHHUHHHHSSHNUHNNHHPUHUHPHPNPHNHHPUUNHHPHHN</span></code></pre></div>
</div>
</div>
<div id="program-gto2_aa_to_pseudo_dna" class="section level2">
<h2><span class="header-section-number">4.2</span> Program gto2_aa_to_pseudo_dna</h2>
<p>The <strong>gto2_aa_to_pseudo_dna</strong> converts an amino acid (protein) sequence to a pseudo DNA sequence.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_aa_to_pseudo_dna</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-28" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_aa_to_pseudo_dna</strong> program needs two streams for the computation, namely the input and output standard. The input stream is an amino acid sequence. The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_aa_to_pseudo_dna [options] [[--] args]
   <span class="ex">or</span>: ./gto2_aa_to_pseudo_dna [options]

<span class="ex">It</span> converts a protein sequence to a pseudo DNA sequence.

    <span class="ex">-h</span>, --help        show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.prot</span>      Input amino acid sequence file (stdin)
    <span class="op">&gt;</span> <span class="ex">output.dna</span>      Output DNA sequence file (stdout)

<span class="ex">Example</span>: ./gto2_aa_to_pseudo_dna <span class="op">&lt;</span> input.prot <span class="op">&gt;</span> output.dna
<span class="ex">Table</span>:
<span class="ex">Prot</span>    DNA
<span class="ex">A</span>       GCA
<span class="ex">C</span>       TGC
<span class="ex">D</span>       GAC
<span class="ex">E</span>       GAG
<span class="ex">F</span>       TTT
<span class="ex">G</span>       GGC
<span class="ex">H</span>       CAT
<span class="ex">I</span>       ATC
<span class="ex">K</span>       AAA
<span class="ex">L</span>       CTG
<span class="ex">M</span>       ATG
<span class="ex">N</span>       AAC
<span class="ex">P</span>       CCG
<span class="ex">Q</span>       CAG
<span class="ex">R</span>       CGT
<span class="ex">S</span>       TCT
<span class="ex">T</span>       ACG
<span class="ex">V</span>       GTA
<span class="ex">W</span>       TGG
<span class="ex">Y</span>       TAC
<span class="ex">*</span>       TAG
<span class="ex">X</span>       GGG</code></pre></div>
<p>It can be used to generate pseudo-DNA with characteristics passed by amino acid (protein) sequences. An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">IPFLLKKQFALADKLVLSKLRQLLGGRIKMMPCGGAKLEPAIGLFFHAIGINIKLGYGMT</span>
<span class="ex">ETTATVSCWHDFQFNPNSIGTLMPKAEVKIGENNEILVRGGMVMKGYYKKPEETAQAFTE</span>
<span class="ex">DGFLKTGDAGEFDEQGNLFITDRIKELMKTSNGKYIAPQYIESKIGKDKFIEQIAIIADA</span>
<span class="ex">KKYVSALIVPCFDSLEEYAKQLNIKYHDRLELLKNSDILKMFE</span></code></pre></div>
</div>
<div id="output-28" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_aa_to_pseudo_dna</strong> program is a DNA sequence. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">ATCCCGTTTCTGCTGAAAAAACAGTTTGCACTGGCAGACAAACTGGTACTGTCTAAACTG</span>
<span class="ex">CGTCAGCTGCTGGGCGGCCGTATCAAAATGATGCCGTGCGGCGGCGCAAAACTGGAGCCG</span>
<span class="ex">GCAATCGGCCTGTTTTTTCATGCAATCGGCATCAACATCAAACTGGGCTACGGCATGACG</span>
<span class="ex">GAGACGACGGCAACGGTATCTTGCTGGCATGACTTTCAGTTTAACCCGAACTCTATCGGC</span>
<span class="ex">ACGCTGATGCCGAAAGCAGAGGTAAAAATCGGCGAGAACAACGAGATCCTGGTACGTGGC</span>
<span class="ex">GGCATGGTAATGAAAGGCTACTACAAAAAACCGGAGGAGACGGCACAGGCATTTACGGAG</span>
<span class="ex">GACGGCTTTCTGAAAACGGGCGACGCAGGCGAGTTTGACGAGCAGGGCAACCTGTTTATC</span>
<span class="ex">ACGGACCGTATCAAAGAGCTGATGAAAACGTCTAACGGCAAATACATCGCACCGCAGTAC</span>
<span class="ex">ATCGAGTCTAAAATCGGCAAAGACAAATTTATCGAGCAGATCGCAATCATCGCAGACGCA</span>
<span class="ex">AAAAAATACGTATCTGCACTGATCGTACCGTGCTTTGACTCTCTGGAGGAGTACGCAAAA</span>
<span class="ex">CAGCTGAACATCAAATACCATGACCGTCTGGAGCTGCTGAAAAACTCTGACATCCTGAAA</span>
<span class="ex">ATGTTTGAG</span></code></pre></div>
</div>
</div>
<div id="program-gto2_aa_compressor" class="section level2">
<h2><span class="header-section-number">4.3</span> Program gto2_aa_compressor</h2>
<p>The <strong>gto2_aa_compressor</strong> is a new lossless compressor to compress efficiently amino acid sequences (proteins). It uses a cooperation between multiple context and substitutional tolerant context models. The cooperation between models is balanced with weights that benefit the models with better performance according to a forgetting function specific for each model.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_aa_compressor</span> -h</code></pre></div>
<p>The <strong>gto2_aa_compressor</strong> program needs a file with amino acid sequences to compress. In the following example, it will be downloaded nine amino acid sequences and compress and decompress one of the smallest (HI). Finally, it compares if the uncompressed sequence is equal to the original.</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="fu">wget</span> http://sweet.ua.pt/pratas/datasets/AminoAcidsCorpus.zip
<span class="fu">unzip</span> AminoAcidsCorpus.zip
<span class="fu">cp</span> AminoAcidsCorpus/HI .
<span class="ex">./gto2_aa_compressor</span> -v -l 2 HI
<span class="ex">./gto2_aa_decompressor</span> -v HI.co
<span class="fu">cmp</span> HI HI.de</code></pre></div>
</div>
<div id="program-gto2_aa_from_fa" class="section level2">
<h2><span class="header-section-number">4.4</span> Program gto2_aa_from_fa</h2>
<p>The <strong>gto2_aa_from_fasta</strong> converts DNA sequences in FASTA or Multi-FASTA file format to an amino acid sequence.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_aa_from_fasta</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-29" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_aa_from_fasta</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTA or Multi-FASTA file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_aa_from_fasta [options] [[--] args]
   <span class="ex">or</span>: ./gto2_aa_from_fasta [options]

<span class="ex">It</span> converts FASTA or Multi-FASTA file format to an amino 
<span class="ex">acid</span> sequence (translation)<span class="ex">.</span>

    <span class="ex">-h</span>, --help        Show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.mfasta</span>    Input FASTA or Multi-FASTA file format 
                      <span class="kw">(</span><span class="ex">stdin</span><span class="kw">)</span>
    <span class="op">&gt;</span> <span class="ex">output.prot</span>     Output amino acid sequence file 
                      <span class="kw">(</span><span class="ex">stdout</span><span class="kw">)</span>

<span class="ex">Optional</span>
    <span class="ex">-f</span>                Translation codon frame (1, 2 or 3)

<span class="ex">Example</span>: ./gto2_aa_from_fasta <span class="op">&lt;</span> input.mfasta <span class="op">&gt;</span> output.prot</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="op">&gt;</span><span class="ex">AB000264</span> <span class="kw">|</span><span class="va">acc=</span>AB000264<span class="kw">|</span><span class="va">descr=</span>Homo <span class="ex">sapiens</span> mRNA 
<span class="ex">ACAAGACGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCCTGGAGGGTCCACCGCT</span>
<span class="ex">GCCCTGCTGCCATTGTCCCCGGCCCCACCTAAGGAAAAGCAGCCTCCTGACTTTCCTCGC</span>
<span class="ex">TTGGGCCGAGACAGCGAGCATATGCAGGAAGCGGCAGGAAGTGGTTTGAGTGGACCTCCG</span>
<span class="ex">GGCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCGGCAGGAAGCAGGCCAGTGCC</span>
<span class="ex">GCGAATCCGCGCGCCGGGACAGAATCTCCTGCAAAGCCCTGCAGGAACTTCTTCTGGAAG</span>
<span class="ex">ACCTTCTCCACCCCCCCAGCTAAAACCTCACCCATGAATGCTCACGCAAGTTTAATTACA</span>
<span class="ex">GACCTGAA</span>
<span class="op">&gt;</span><span class="ex">AB000263</span> <span class="kw">|</span><span class="va">acc=</span>AB000263<span class="kw">|</span><span class="va">descr=</span>Homo <span class="ex">sapiens</span> mRNA 
<span class="ex">ACAAGATGCCATTGTCCCCCGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCACCG</span>
<span class="ex">CTGCCCTGCCCCTGGAGGGTGGCCCCACCGGCCGAGACAGCGAGCATATGCAGGAAGCGG</span>
<span class="ex">CAGGAATAAGGAAAAGCAGCCTCCTGACTTTCCTCGCTTGGTGGTTTGAGTGGACCTCCC</span>
<span class="ex">AGGCCAGTGCCGGGCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCGGCAGGAAG</span>
<span class="ex">GCGCACCCCCCCAGCAATCCGCGCGCCGGGACAGAATGCCCTGCAGGAACTTCTTCTGGA</span>
<span class="ex">AGACCTTCTCCTCCTGCAAATAAAACCTCACCCATGAATGCTCACGCAAGTTTAATTACA</span>
<span class="ex">GACCTGAA</span></code></pre></div>
</div>
<div id="output-29" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_aa_from_fasta</strong> program is an amino acid sequence. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">TRRPPAAAALRGHGPGGSTAALLPLSPAPPKEKQPPDFPRLGRDSEHMQEAAGSGLSGPP</span>
<span class="ex">GPS-ERKLGRWPGGRKQASAANPRAGTESPAKPCRNFFWKTFSTPPAKTSPMNAHASLIT</span>
<span class="ex">DLTRCHCPPASCCCCSPGPRPPLPCPWRVAPPAETASICRKRQE-GKAAS-LSSLGGLSG</span>
<span class="ex">PPRPVPGPS-ERKLGRWPGGRKAHPPSNPRAGTECPAGTSSGRPSPPANKTSPMNAHASL</span>
<span class="ex">ITDL</span></code></pre></div>
</div>
</div>
<div id="program-gto2_aa_from_fq" class="section level2">
<h2><span class="header-section-number">4.5</span> Program gto2_aa_from_fq</h2>
<p>The <strong>gto2_aa_from_fastq</strong> converts DNA sequences in the FASTQ file format to an amino acid sequence.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_aa_from_fastq</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-30" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_aa_from_fastq</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ../../bin/gto2_aa_from_fastq [options] [[--] args]
   <span class="ex">or</span>: ../../bin/gto2_aa_from_fastq [options]

<span class="ex">It</span> converts FASTQ file format to an amino acid sequence 
<span class="kw">(</span><span class="ex">translation</span><span class="kw">)</span><span class="ex">.</span>

    <span class="ex">-h</span>, --help      Show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>   Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.prot</span>   Output amino acid sequence file (stdout)

<span class="ex">Optional</span>
    <span class="ex">-f</span>              Translation codon frame (1, 2 or 3)

<span class="ex">Example</span>: ./gto2_aa_from_fastq <span class="op">&lt;</span> input.fastq <span class="op">&gt;</span> output.prot</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-30" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_aa_from_fastq</strong> program is an amino acid sequence. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">G-WPLPMASNPTKLPLTT-GFSNRVQGYDVCILRI-SRSR--YASFYH</span></code></pre></div>
</div>
</div>
<div id="program-gto2_aa_from_seq" class="section level2">
<h2><span class="header-section-number">4.6</span> Program gto2_aa_from_seq</h2>
<p>The <strong>gto2_aa_from_seq</strong> converts DNA sequence to an amino acid sequence.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_aa_from_seq</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-31" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_aa_from_seq</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a DNA sequence.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_aa_from_seq [options] [[--] args]
   <span class="ex">or</span>: ./gto2_aa_from_seq [options]

<span class="ex">It</span> converts DNA sequence to an amino acid sequence 
<span class="kw">(</span><span class="ex">translation</span><span class="kw">)</span><span class="ex">.</span>

    <span class="ex">-h</span>, --help      Show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.seq</span>     Input sequence file (stdin)
    <span class="op">&gt;</span> <span class="ex">output.prot</span>   Output amino acid sequence file (stdout)

<span class="ex">Optional</span>
    <span class="ex">-f</span>              Translation codon frame (1, 2 or 3)

<span class="ex">Example</span>: ./gto2_aa_from_seq <span class="op">&lt;</span> input.seq <span class="op">&gt;</span> output.prot</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">ACAAGACGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCCTGGAGGGTCCACCGCT</span>
<span class="ex">GCCCTGCTGCCATTGTCCCCGGCCCCACCTAAGGAAAAGCAGCCTCCTGACTTTCCTCGC</span>
<span class="ex">TTGGGCCGAGACAGCGAGCATATGCAGGAAGCGGCAGGAAGTGGTTTGAGTGGACCTCCG</span>
<span class="ex">GGCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCGGCAGGAAGCAGGCCAGTGCC</span>
<span class="ex">GCGAATCCGCGCGCCGGGACAGAATCTCCTGCAAAGCCCTGCAGGAACTTCTTCTGGAAG</span>
<span class="ex">ACCTTCTCCACCCCCCCAGCTAAAACCTCACCCATGAATGCTCACGCAAGTTTAATTACA</span>
<span class="ex">GACCTGAAACAAGATGCCATTGTCCCCCGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCAC</span>
<span class="ex">GGCCACCGCTGCCCTGCCCCTGGAGGGTGGCCCCACCGGCCGAGACAGCGAGCATATGCA</span>
<span class="ex">GGAAGCGGCAGGAATAAGGAAAAGCAGCCTCCTGACTTTCCTCGCTTGGTGGTTTGAGTG</span>
<span class="ex">GACCTCCCAGGCCAGTGCCGGGCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCG</span>
<span class="ex">GCAGGAAGGCGCACCCCCCCAGCAATCCGCGCGCCGGGACAGAATGCCCTGCAGGAACTT</span>
<span class="ex">CTTCTGGAAGACCTTCTCCTCCTGCAAATAAAACCTCACCCATGAATGCTCACGCAAGTT</span>
<span class="ex">TAATTACAGACCTGAA</span></code></pre></div>
</div>
<div id="output-31" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_aa_from_seq</strong> program is an amino acid sequence. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">TRRPPAAAALRGHGPGGSTAALLPLSPAPPKEKQPPDFPRLGRDSEHMQEAAGSGLSGPP</span>
<span class="ex">GPS-ERKLGRWPGGRKQASAANPRAGTESPAKPCRNFFWKTFSTPPAKTSPMNAHASLIT</span>
<span class="ex">DLKQDAIVPRPPAAAALRGHGHRCPAPGGWPHRPRQRAYAGSGRNKEKQPPDFPRLVV-V</span>
<span class="ex">DLPGQCRAPHRRGSSGGGQAAGRRTPPAIRAPGQNALQELLLEDLLLLQIKPHP-MLTQV</span>
<span class="ex">-LQT-</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="fastq-tools.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="genomic-tools.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["GTO2.pdf"],
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
