<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 FASTQ Tools | GTO 2: The genomics-proteomics toolkit</title>
  <meta name="description" content="To do" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 FASTQ Tools | GTO 2: The genomics-proteomics toolkit" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="To do" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 FASTQ Tools | GTO 2: The genomics-proteomics toolkit" />
  
  <meta name="twitter:description" content="To do" />
  

<meta name="author" content="J. R. Almeida, A. J. Pinho, J. L. Oliveira, D. Pratas" />


<meta name="date" content="2021-04-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="fasta-tools.html"/>
<link rel="next" href="amino-acid-tools.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="/">GTO 2: The genomics-proteomics toolkit</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#testing"><i class="fa fa-check"></i><b>1.2</b> Testing</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#execution-control"><i class="fa fa-check"></i><b>1.3</b> Execution control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="fasta-tools.html"><a href="fasta-tools.html"><i class="fa fa-check"></i><b>2</b> FASTA Tools</a><ul>
<li class="chapter" data-level="2.1" data-path="fasta-tools.html"><a href="fasta-tools.html#gto2_fa_to_fq"><i class="fa fa-check"></i><b>2.1</b> gto2_fa_to_fq</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="fastq-tools.html"><a href="fastq-tools.html"><i class="fa fa-check"></i><b>3</b> FASTQ Tools</a><ul>
<li class="chapter" data-level="3.1" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_to_fa"><i class="fa fa-check"></i><b>3.1</b> gto2_fq_to_fa</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_to_mfa"><i class="fa fa-check"></i><b>3.2</b> gto2_fq_to_mfa</a></li>
<li class="chapter" data-level="3.3" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_exclude_n"><i class="fa fa-check"></i><b>3.3</b> gto2_fq_exclude_n</a></li>
<li class="chapter" data-level="3.4" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_extract_quality_scores"><i class="fa fa-check"></i><b>3.4</b> gto2_fq_extract_quality_scores</a></li>
<li class="chapter" data-level="3.5" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_info"><i class="fa fa-check"></i><b>3.5</b> gto2_fq_info</a></li>
<li class="chapter" data-level="3.6" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_maximum_read_size"><i class="fa fa-check"></i><b>3.6</b> gto2_fq_maximum_read_size</a></li>
<li class="chapter" data-level="3.7" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_minimum_quality_score"><i class="fa fa-check"></i><b>3.7</b> gto2_fq_minimum_quality_score</a></li>
<li class="chapter" data-level="3.8" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_minimum_read_size"><i class="fa fa-check"></i><b>3.8</b> gto2_fq_minimum_read_size</a></li>
<li class="chapter" data-level="3.9" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_rand_extra_chars"><i class="fa fa-check"></i><b>3.9</b> gto2_fq_rand_extra_chars</a></li>
<li class="chapter" data-level="3.10" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_from_seq"><i class="fa fa-check"></i><b>3.10</b> gto2_fq_from_seq</a></li>
<li class="chapter" data-level="3.11" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_mutate"><i class="fa fa-check"></i><b>3.11</b> gto2_fq_mutate</a></li>
<li class="chapter" data-level="3.12" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_split"><i class="fa fa-check"></i><b>3.12</b> gto2_fq_split</a></li>
<li class="chapter" data-level="3.13" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_pack"><i class="fa fa-check"></i><b>3.13</b> gto2_fq_pack</a></li>
<li class="chapter" data-level="3.14" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_unpack"><i class="fa fa-check"></i><b>3.14</b> gto2_fq_unpack</a></li>
<li class="chapter" data-level="3.15" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_quality_score_info"><i class="fa fa-check"></i><b>3.15</b> gto2_fq_quality_score_info</a></li>
<li class="chapter" data-level="3.16" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_quality_score_min"><i class="fa fa-check"></i><b>3.16</b> gto2_fq_quality_score_min</a></li>
<li class="chapter" data-level="3.17" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_quality_score_max"><i class="fa fa-check"></i><b>3.17</b> gto2_fq_quality_score_max</a></li>
<li class="chapter" data-level="3.18" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_cut"><i class="fa fa-check"></i><b>3.18</b> gto2_fq_cut</a></li>
<li class="chapter" data-level="3.19" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_minimum_local_quality_score_forward"><i class="fa fa-check"></i><b>3.19</b> gto2_fq_minimum_local_quality_score_forward</a></li>
<li class="chapter" data-level="3.20" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_minimum_local_quality_score_reverse"><i class="fa fa-check"></i><b>3.20</b> gto2_fq_minimum_local_quality_score_reverse</a></li>
<li class="chapter" data-level="3.21" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_xs"><i class="fa fa-check"></i><b>3.21</b> gto2_fq_xs</a></li>
<li class="chapter" data-level="3.22" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_complement"><i class="fa fa-check"></i><b>3.22</b> gto2_fq_complement</a></li>
<li class="chapter" data-level="3.23" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_reverse"><i class="fa fa-check"></i><b>3.23</b> gto2_fq_reverse</a></li>
<li class="chapter" data-level="3.24" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_variation_map"><i class="fa fa-check"></i><b>3.24</b> gto2_fq_variation_map</a></li>
<li class="chapter" data-level="3.25" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_variation_filter"><i class="fa fa-check"></i><b>3.25</b> gto2_fq_variation_filter</a></li>
<li class="chapter" data-level="3.26" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_variation_visual"><i class="fa fa-check"></i><b>3.26</b> gto2_fq_variation_visual</a></li>
<li class="chapter" data-level="3.27" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_metagenomics"><i class="fa fa-check"></i><b>3.27</b> gto2_fq_metagenomics</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html"><i class="fa fa-check"></i><b>4</b> Amino Acid Tools</a><ul>
<li class="chapter" data-level="4.1" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#gto2_aa"><i class="fa fa-check"></i><b>4.1</b> gto2_aa</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genomic-tools.html"><a href="genomic-tools.html"><i class="fa fa-check"></i><b>5</b> Genomic Tools</a><ul>
<li class="chapter" data-level="5.1" data-path="genomic-tools.html"><a href="genomic-tools.html#gto2_dna"><i class="fa fa-check"></i><b>5.1</b> gto2_dna</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="general-purpose-tools.html"><a href="general-purpose-tools.html"><i class="fa fa-check"></i><b>6</b> General Purpose Tools</a><ul>
<li class="chapter" data-level="6.1" data-path="general-purpose-tools.html"><a href="general-purpose-tools.html#gto2_"><i class="fa fa-check"></i><b>6.1</b> gto2_</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GTO 2: The genomics-proteomics toolkit</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="fastq-tools" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> FASTQ Tools</h1>
<p>The toolkit has a set of tools dedicated to manipulating FASTQ files. Some of these tools allow the data conversion to/from different formats, i. e., there are tools designed to convert a FASTQ file into a sequence or a FASTA/Multi-FASTA format, or converting DNA in some of those formats to FASTQ.</p>
<p>There are also tools for data manipulation in this format, which are designed to exclude 'N', remove low quality scored reads, following different metrics and randomize DNA sequences. Succeeding the manipulation, it is also possible to perform analyses over these files, simulations and mutations. The current available tools for FASTQ format analysis and manipulation include:</p>
<ul>
<li><strong>gto2_fq_to_fa</strong>: it converts a FASTQ file format to a pseudo FASTA file.</li>
<li><strong>gto2_fq_to_mfa</strong>: it converts a FASTQ file format to a pseudo Multi-FASTA file.</li>
<li><strong>gto2_fq_exclude_n</strong>: it discards the FASTQ reads with the minimum number of &quot;N&quot; symbols.</li>
<li><strong>gto2_fq_extract_quality_scores</strong>: it extracts all the quality-scores from FASTQ reads.</li>
<li><strong>gto2_fq_info</strong>: it analyses the basic information of FASTQ file format.</li>
<li><strong>gto2_fq_maximum_read_size</strong>: it filters the FASTQ reads with the length higher than the value defined.</li>
<li><strong>gto2_fq_minimum_quality_score</strong>: it discards reads with average quality-score below of the defined.</li>
<li><strong>gto2_fq_minimum_read_size</strong>: it filters the FASTQ reads with the length smaller than the value defined.</li>
<li><strong>gto2_fq_rand_extra_chars</strong>: it substitues in the FASTQ files, the DNA sequence the outside ACGT chars by random ACGT symbols.</li>
<li><strong>gto2_fq_from_seq</strong>: it converts a genomic sequence to pseudo FASTQ file format.</li>
<li><strong>gto2_fq_mutate</strong>: it creates a synthetic mutation of a FASTQ file given specific rates of mutations, deletions and additions.</li>
<li><strong>gto2_fq_split</strong>: it splits Paired End files according to the direction of the strand ('/1' or '/2').</li>
<li><strong>gto2_fq_pack</strong>: it packages each FASTQ read in a single line.</li>
<li><strong>gto2_fq_unpack</strong>: it unpacks the FASTQ reads packaged using the <strong>gto2_fastq_pack</strong> tool.</li>
<li><strong>gto2_fq_quality_score_info</strong>: it analyses the quality-scores of a FASTQ file.</li>
<li><strong>gto2_fq_quality_score_min</strong>: it analyses the minimal quality-scores of a FASTQ file.</li>
<li><strong>gto2_fq_quality_score_max</strong>: it analyses the maximal quality-scores of a FASTQ file.</li>
<li><strong>gto2_fq_cut</strong>: it cuts read sequences in a FASTQ file.</li>
<li><strong>gto2_fq_minimum_local_quality_score_forward</strong>: it filters the reads considering the quality score average of a defined window size of bases.</li>
<li><strong>gto2_fq_minimum_local_quality_score_reverse</strong>: it filters the reverse reads, considering the average window size score defined by the bases.</li>
<li><strong>gto2_fq_xs</strong>: it is a skilled FASTQ read simulation tool, flexible, portable and tunable in terms of sequence complexity.</li>
<li><strong>gto2_fq_complement</strong>: it replaces the ACGT bases with their complements in a FASTQ file format.</li>
<li><strong>gto2_fq_reverse</strong>: it reverses the ACGT bases order for each read in a FASTQ file format.</li>
<li><strong>gto2_fq_variation_map</strong>: it identifies the variation that occours in the sequences relative to the reads or a set of reads.</li>
<li><strong>gto2_fq_variation_filter</strong>: it filters and segments the regions of singularity from the output of <strong>gto2_fq_variation_map</strong>.</li>
<li><strong>gto2_fq_variation_visual</strong>: it depites the regions of singularity using the output from <strong>gto2_fq_variation_filter</strong> into an SVG image.</li>
<li><strong>gto2_fq_metagenomics</strong>: it measures similarity between any FASTQ file, independently from the size, against any multi-FASTA database.</li>
</ul>
<div id="gto2_fq_to_fa" class="section level2">
<h2><span class="header-section-number">3.1</span> gto2_fq_to_fa</h2>
<p>The <strong>gto2_fq_to_fa</strong> converts a FASTQ file format to a pseudo FASTA file. However, it does not align the sequence. Also, it extracts the sequence and adds a pseudo header.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_to_fa</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output paramters.</p>
<div id="input-parameters" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_to_fa</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_to_fa [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_to_fa [options]

<span class="ex">It</span> converts a FASTQ file format to a pseudo FASTA file.
<span class="ex">It</span> does NOT align the sequence.
<span class="ex">It</span> extracts the sequence and adds a pseudo header.

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fasta</span>        Output FASTA file format (stdout)

<span class="ex">Example</span>: ./gto2_fq_to_fa <span class="op">&lt;</span> input.fastq <span class="op">&gt;</span> output.fasta</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_to_fa</strong> program a FASTA file. Using the input above, an output example for this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="op">&gt;</span> <span class="ex">Computed</span> with Fastq2Fasta
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span></code></pre></div>
</div>
</div>
<div id="gto2_fq_to_mfa" class="section level2">
<h2><span class="header-section-number">3.2</span> gto2_fq_to_mfa</h2>
<p>to do</p>
</div>
<div id="gto2_fq_exclude_n" class="section level2">
<h2><span class="header-section-number">3.3</span> gto2_fq_exclude_n</h2>
<p>to do</p>
</div>
<div id="gto2_fq_extract_quality_scores" class="section level2">
<h2><span class="header-section-number">3.4</span> gto2_fq_extract_quality_scores</h2>
<p>to do</p>
</div>
<div id="gto2_fq_info" class="section level2">
<h2><span class="header-section-number">3.5</span> gto2_fq_info</h2>
<p>to do</p>
</div>
<div id="gto2_fq_maximum_read_size" class="section level2">
<h2><span class="header-section-number">3.6</span> gto2_fq_maximum_read_size</h2>
<p>to do</p>
</div>
<div id="gto2_fq_minimum_quality_score" class="section level2">
<h2><span class="header-section-number">3.7</span> gto2_fq_minimum_quality_score</h2>
<p>to do</p>
</div>
<div id="gto2_fq_minimum_read_size" class="section level2">
<h2><span class="header-section-number">3.8</span> gto2_fq_minimum_read_size</h2>
<p>to do</p>
</div>
<div id="gto2_fq_rand_extra_chars" class="section level2">
<h2><span class="header-section-number">3.9</span> gto2_fq_rand_extra_chars</h2>
<p>to do</p>
</div>
<div id="gto2_fq_from_seq" class="section level2">
<h2><span class="header-section-number">3.10</span> gto2_fq_from_seq</h2>
<p>to do</p>
</div>
<div id="gto2_fq_mutate" class="section level2">
<h2><span class="header-section-number">3.11</span> gto2_fq_mutate</h2>
<p>to do</p>
</div>
<div id="gto2_fq_split" class="section level2">
<h2><span class="header-section-number">3.12</span> gto2_fq_split</h2>
<p>to do</p>
</div>
<div id="gto2_fq_pack" class="section level2">
<h2><span class="header-section-number">3.13</span> gto2_fq_pack</h2>
<p>to do</p>
</div>
<div id="gto2_fq_unpack" class="section level2">
<h2><span class="header-section-number">3.14</span> gto2_fq_unpack</h2>
<p>to do</p>
</div>
<div id="gto2_fq_quality_score_info" class="section level2">
<h2><span class="header-section-number">3.15</span> gto2_fq_quality_score_info</h2>
<p>to do</p>
</div>
<div id="gto2_fq_quality_score_min" class="section level2">
<h2><span class="header-section-number">3.16</span> gto2_fq_quality_score_min</h2>
<p>to do</p>
</div>
<div id="gto2_fq_quality_score_max" class="section level2">
<h2><span class="header-section-number">3.17</span> gto2_fq_quality_score_max</h2>
<p>to do</p>
</div>
<div id="gto2_fq_cut" class="section level2">
<h2><span class="header-section-number">3.18</span> gto2_fq_cut</h2>
<p>to do</p>
</div>
<div id="gto2_fq_minimum_local_quality_score_forward" class="section level2">
<h2><span class="header-section-number">3.19</span> gto2_fq_minimum_local_quality_score_forward</h2>
<p>to do</p>
</div>
<div id="gto2_fq_minimum_local_quality_score_reverse" class="section level2">
<h2><span class="header-section-number">3.20</span> gto2_fq_minimum_local_quality_score_reverse</h2>
<p>to do</p>
</div>
<div id="gto2_fq_xs" class="section level2">
<h2><span class="header-section-number">3.21</span> gto2_fq_xs</h2>
<p>to do</p>
</div>
<div id="gto2_fq_complement" class="section level2">
<h2><span class="header-section-number">3.22</span> gto2_fq_complement</h2>
<p>to do</p>
</div>
<div id="gto2_fq_reverse" class="section level2">
<h2><span class="header-section-number">3.23</span> gto2_fq_reverse</h2>
<p>to do</p>
</div>
<div id="gto2_fq_variation_map" class="section level2">
<h2><span class="header-section-number">3.24</span> gto2_fq_variation_map</h2>
<p>to do</p>
</div>
<div id="gto2_fq_variation_filter" class="section level2">
<h2><span class="header-section-number">3.25</span> gto2_fq_variation_filter</h2>
<p>to do</p>
</div>
<div id="gto2_fq_variation_visual" class="section level2">
<h2><span class="header-section-number">3.26</span> gto2_fq_variation_visual</h2>
<p>to do</p>
</div>
<div id="gto2_fq_metagenomics" class="section level2">
<h2><span class="header-section-number">3.27</span> gto2_fq_metagenomics</h2>
<p>to do</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="fasta-tools.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="amino-acid-tools.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["GTO2.pdf"],
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
