<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 FASTQ Tools | GTO 2: The genomics-proteomics toolkit</title>
  <meta name="description" content="To do" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 FASTQ Tools | GTO 2: The genomics-proteomics toolkit" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="To do" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 FASTQ Tools | GTO 2: The genomics-proteomics toolkit" />
  
  <meta name="twitter:description" content="To do" />
  

<meta name="author" content="J. R. Almeida, A. J. Pinho, J. L. Oliveira, D. Pratas" />


<meta name="date" content="2021-04-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="fasta-tools.html"/>
<link rel="next" href="amino-acid-tools.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="/">GTO 2: The genomics-proteomics toolkit</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#testing"><i class="fa fa-check"></i><b>1.2</b> Testing</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#execution-control"><i class="fa fa-check"></i><b>1.3</b> Execution control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="fasta-tools.html"><a href="fasta-tools.html"><i class="fa fa-check"></i><b>2</b> FASTA Tools</a><ul>
<li class="chapter" data-level="2.1" data-path="fasta-tools.html"><a href="fasta-tools.html#gto2_fa_to_fq"><i class="fa fa-check"></i><b>2.1</b> gto2_fa_to_fq</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="fastq-tools.html"><a href="fastq-tools.html"><i class="fa fa-check"></i><b>3</b> FASTQ Tools</a><ul>
<li class="chapter" data-level="3.1" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_to_fa"><i class="fa fa-check"></i><b>3.1</b> Program gto2_fq_to_fa</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_to_mfa"><i class="fa fa-check"></i><b>3.2</b> Program gto2_fq_to_mfa</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-1"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-1"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_exclude_n"><i class="fa fa-check"></i><b>3.3</b> Program gto2_fq_exclude_n</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-2"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-2"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_extract_quality_scores"><i class="fa fa-check"></i><b>3.4</b> Program gto2_fq_extract_quality_scores</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-3"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-3"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_info"><i class="fa fa-check"></i><b>3.5</b> Program gto2_fq_info</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-4"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-4"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_maximum_read_size"><i class="fa fa-check"></i><b>3.6</b> Program gto2_fq_maximum_read_size</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-5"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-5"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_minimum_quality_score"><i class="fa fa-check"></i><b>3.7</b> Program gto2_fq_minimum_quality_score</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-6"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-6"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_minimum_read_size"><i class="fa fa-check"></i><b>3.8</b> Program gto2_fq_minimum_read_size</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-7"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-7"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_rand_extra_chars"><i class="fa fa-check"></i><b>3.9</b> Program gto2_fq_rand_extra_chars</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-8"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-8"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.10" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_from_seq"><i class="fa fa-check"></i><b>3.10</b> Program gto2_fq_from_seq</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-9"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-9"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.11" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_mutate"><i class="fa fa-check"></i><b>3.11</b> Program gto2_fq_mutate</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-10"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-10"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.12" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_split"><i class="fa fa-check"></i><b>3.12</b> Program gto2_fq_split</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-11"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-11"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.13" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_pack"><i class="fa fa-check"></i><b>3.13</b> Program gto2_fq_pack</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-12"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-12"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.14" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_unpack"><i class="fa fa-check"></i><b>3.14</b> Program gto2_fq_unpack</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-13"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-13"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.15" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_quality_score_info"><i class="fa fa-check"></i><b>3.15</b> Program gto2_fq_quality_score_info</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-14"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-14"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.16" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_quality_score_min"><i class="fa fa-check"></i><b>3.16</b> Program gto2_fq_quality_score_min</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-15"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-15"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.17" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_quality_score_max"><i class="fa fa-check"></i><b>3.17</b> Program gto2_fq_quality_score_max</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-16"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-16"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.18" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_cut"><i class="fa fa-check"></i><b>3.18</b> Program gto2_fq_cut</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-17"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-17"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.19" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_minimum_local_quality_score_forward"><i class="fa fa-check"></i><b>3.19</b> Program gto2_fq_minimum_local_quality_score_forward</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-18"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-18"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.20" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_minimum_local_quality_score_reverse"><i class="fa fa-check"></i><b>3.20</b> Program gto2_fq_minimum_local_quality_score_reverse</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-19"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-19"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.21" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_xs"><i class="fa fa-check"></i><b>3.21</b> Program gto2_fq_xs</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-20"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-20"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.22" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_complement"><i class="fa fa-check"></i><b>3.22</b> Program gto2_fq_complement</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-21"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-21"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.23" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_reverse"><i class="fa fa-check"></i><b>3.23</b> Program gto2_fq_reverse</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-22"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-22"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.24" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_variation_map"><i class="fa fa-check"></i><b>3.24</b> Program gto2_fq_variation_map</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-23"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-23"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.25" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_variation_filter"><i class="fa fa-check"></i><b>3.25</b> Program gto2_fq_variation_filter</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-24"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-24"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.26" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_variation_visual"><i class="fa fa-check"></i><b>3.26</b> Program gto2_fq_variation_visual</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-25"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-25"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.27" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_metagenomics"><i class="fa fa-check"></i><b>3.27</b> Program gto2_fq_metagenomics</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-26"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-26"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html"><i class="fa fa-check"></i><b>4</b> Amino Acid Tools</a><ul>
<li class="chapter" data-level="4.1" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#gto2_aa"><i class="fa fa-check"></i><b>4.1</b> gto2_aa</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genomic-tools.html"><a href="genomic-tools.html"><i class="fa fa-check"></i><b>5</b> Genomic Tools</a><ul>
<li class="chapter" data-level="5.1" data-path="genomic-tools.html"><a href="genomic-tools.html#gto2_dna"><i class="fa fa-check"></i><b>5.1</b> gto2_dna</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="general-purpose-tools.html"><a href="general-purpose-tools.html"><i class="fa fa-check"></i><b>6</b> General Purpose Tools</a><ul>
<li class="chapter" data-level="6.1" data-path="general-purpose-tools.html"><a href="general-purpose-tools.html#gto2_"><i class="fa fa-check"></i><b>6.1</b> gto2_</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GTO 2: The genomics-proteomics toolkit</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="fastq-tools" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> FASTQ Tools</h1>
<p>The toolkit has a set of tools dedicated to manipulating FASTQ files. Some of these tools allow the data conversion to/from different formats, i. e., there are tools designed to convert a FASTQ file into a sequence or a FASTA/Multi-FASTA format, or converting DNA in some of those formats to FASTQ.</p>
<p>There are also tools for data manipulation in this format, which are designed to exclude 'N', remove low quality scored reads, following different metrics and randomize DNA sequences. Succeeding the manipulation, it is also possible to perform analyses over these files, simulations and mutations. The current available tools for FASTQ format analysis and manipulation include:</p>
<ul>
<li><strong>gto2_fq_to_fa</strong>: to convert a FASTQ file format to a pseudo FASTA file.</li>
<li><strong>gto2_fq_to_mfa</strong>: to convert a FASTQ file format to a pseudo Multi-FASTA file.</li>
<li><strong>gto2_fq_exclude_n</strong>: to discard the FASTQ reads with the minimum number of &quot;N&quot; symbols.</li>
<li><strong>gto2_fq_extract_quality_scores</strong>: to extract all the quality-scores from FASTQ reads.</li>
<li><strong>gto2_fq_info</strong>: to analyse the basic information of FASTQ file format.</li>
<li><strong>gto2_fq_maximum_read_size</strong>: to filter the FASTQ reads with the length higher than the value defined.</li>
<li><strong>gto2_fq_minimum_quality_score</strong>: to discard reads with average quality-score below of the defined.</li>
<li><strong>gto2_fq_minimum_read_size</strong>: to filter the FASTQ reads with the length smaller than the value defined.</li>
<li><strong>gto2_fq_rand_extra_chars</strong>: to substitue in the FASTQ files, the DNA sequence the outside ACGT chars by random ACGT symbols.</li>
<li><strong>gto2_fq_from_seq</strong>: to convert a genomic sequence to pseudo FASTQ file format.</li>
<li><strong>gto2_fq_mutate</strong>: to create a synthetic mutation of a FASTQ file given specific rates of mutations, deletions and additions.</li>
<li><strong>gto2_fq_split</strong>: to split Paired End files according to the direction of the strand ('/1' or '/2').</li>
<li><strong>gto2_fq_pack</strong>: to package each FASTQ read in a single line.</li>
<li><strong>gto2_fq_unpack</strong>: to unpack the FASTQ reads packaged using the <strong>gto2_fq_pack</strong> tool.</li>
<li><strong>gto2_fq_quality_score_info</strong>: to analyse the quality-scores of a FASTQ file.</li>
<li><strong>gto2_fq_quality_score_min</strong>: to analyse the minimal quality-scores of a FASTQ file.</li>
<li><strong>gto2_fq_quality_score_max</strong>: to analyse the maximal quality-scores of a FASTQ file.</li>
<li><strong>gto2_fq_cut</strong>: to cut read sequences in a FASTQ file.</li>
<li><strong>gto2_fq_minimum_local_quality_score_forward</strong>: to filter the reads considering the quality score average of a defined window size of bases.</li>
<li><strong>gto2_fq_minimum_local_quality_score_reverse</strong>: to filter the reverse reads, considering the average window size score defined by the bases.</li>
<li><strong>gto2_fq_xs</strong>: a skilled FASTQ read simulation tool, flexible, portable and tunable in terms of sequence complexity.</li>
<li><strong>gto2_fq_complement</strong>: to replace the ACGT bases with their complements in a FASTQ file format.</li>
<li><strong>gto2_fq_reverse</strong>: to reverse the ACGT bases order for each read in a FASTQ file format.</li>
<li><strong>gto2_fq_variation_map</strong>: to identify the variation that occours in the sequences relative to the reads or a set of reads.</li>
<li><strong>gto2_fq_variation_filter</strong>: to filter and segments the regions of singularity from the output of <strong>gto2_fq_variation_map</strong>.</li>
<li><strong>gto2_fq_variation_visual</strong>: to depict the regions of singularity using the output from <strong>gto2_fq_variation_filter</strong> into an SVG image.</li>
<li><strong>gto2_fq_metagenomics</strong>: to measure the similarity between any FASTQ file, independently from the size, against any multi-FASTA database.</li>
</ul>
<div id="program-gto2_fq_to_fa" class="section level2">
<h2><span class="header-section-number">3.1</span> Program gto2_fq_to_fa</h2>
<p>The <strong>gto2_fq_to_fa</strong> converts a FASTQ file format to a pseudo FASTA file. However, this tool does not align the sequence, instead, it extracts the sequence and adds a pseudo-header.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_to_fa</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_to_fa</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_to_fa [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_to_fa [options]

<span class="ex">It</span> converts a FASTQ file format to a pseudo FASTA file.
<span class="ex">It</span> does NOT align the sequence.
<span class="ex">It</span> extracts the sequence and adds a pseudo header.

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fasta</span>        Output FASTA file format (stdout)

<span class="ex">Example</span>: ./gto2_fq_to_fa <span class="op">&lt;</span> input.fastq <span class="op">&gt;</span> output.fasta</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_to_fa</strong> program is a FASTA file. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="op">&gt;</span> <span class="ex">Computed</span> with Fastq2Fasta
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span></code></pre></div>
</div>
</div>
<div id="program-gto2_fq_to_mfa" class="section level2">
<h2><span class="header-section-number">3.2</span> Program gto2_fq_to_mfa</h2>
<p>The <strong>gto2_fq_to_mfa</strong> converts a FASTQ file format to a pseudo Multi-FASTA file. However, this tool does not align the sequence, instead, it extracts the sequence and adds a pseudo header.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_to_mfa</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-1" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_to_mfa</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_to_mfa [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_to_mfa [options]

<span class="ex">It</span> converts a FASTQ file format to a pseudo Multi-FASTA 
<span class="ex">file.</span> It does NOT align the sequence. It extracts the 
<span class="ex">sequence</span> and adds each header in a Multi-FASTA format.

    <span class="ex">-h</span>, --help       show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>    Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.mfasta</span>  Output Multi-FASTA file format (stdout)

<span class="ex">Example</span>: ./gto2_fq_to_mfa <span class="op">&lt;</span> input.fastq <span class="op">&gt;</span> output.mfasta</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-1" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_to_mfa</strong> program is a Multi-FASTA file. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="op">&gt;</span><span class="ex">SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="op">&gt;</span><span class="ex">SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span></code></pre></div>
</div>
</div>
<div id="program-gto2_fq_exclude_n" class="section level2">
<h2><span class="header-section-number">3.3</span> Program gto2_fq_exclude_n</h2>
<p>The <strong>gto2_fq_exclude_n</strong> discards the FASTQ reads with the minimum number of ''N'' symbols, and it will erase the second header (after +), if presented.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_exclude_n</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-2" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_exclude_n</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_exclude_n [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_exclude_n [options]

<span class="ex">It</span> discards the FASTQ reads with the minimum number of <span class="st">&quot;N&quot;</span> 
<span class="ex">symbols.</span> 
<span class="ex">If</span> present, it will erase the second header (after +)<span class="ex">.</span>

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="ex">-m</span>, --max=<span class="op">&lt;</span>int<span class="op">&gt;</span>       The maximum of of <span class="st">&quot;N&quot;</span> symbols in 
                          <span class="ex">the</span> read
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>        Output FASTQ file format (stdout)

<span class="ex">Example</span>: ./gto2_fq_exclude_n -m <span class="op">&lt;</span>max<span class="op">&gt;</span> <span class="op">&lt;</span> input.fastq <span class="op">&gt;</span> 
<span class="ex">output.fastq</span>

<span class="ex">Console</span> output example :
<span class="op">&lt;</span><span class="ex">FASTQ</span> non-filtered reads<span class="op">&gt;</span>
<span class="ex">Total</span> reads    : value
<span class="ex">Filtered</span> reads : value</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-2" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_exclude_n</strong> program is a set of all the filtered FASTQ reads, followed by the execution report. The execution report only appears in the console.</p>
<p>Using the input above with the max value as 5, an output example for this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span>
<span class="ex">Total</span> reads    : 2
<span class="ex">Filtered</span> reads : 1</code></pre></div>
</div>
</div>
<div id="program-gto2_fq_extract_quality_scores" class="section level2">
<h2><span class="header-section-number">3.4</span> Program gto2_fq_extract_quality_scores</h2>
<p>The <strong>gto2_fq_extract_quality_scores</strong> extracts all the quality-scores from FASTQ reads.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_extract_quality_scores</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-3" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_extract_quality_scores</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_extract_quality_scores [options] [[--]args]
   <span class="ex">or</span>: ./gto2_fq_extract_quality_scores [options]

<span class="ex">It</span> extracts all the quality-scores from FASTQ reads.

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>        Output FASTQ file format (stdout)

<span class="ex">Example</span>: ./gto2_fq_extract_quality_scores <span class="op">&lt;</span> input.fastq <span class="op">&gt;</span>
<span class="ex">output.fastq</span>

<span class="ex">Console</span> output example:
<span class="op">&lt;</span><span class="ex">FASTQ</span> quality scores<span class="op">&gt;</span>
<span class="ex">Total</span> reads          : value
<span class="ex">Total</span> Quality-Scores : value</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-3" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_extract_quality_scores</strong> program is a set of all the quality scores from the FASTQ reads, followed by the execution report. The execution report only appears in the console. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span>
<span class="ex">Total</span> reads          : 2
<span class="ex">Total</span> Quality-Scores : 144</code></pre></div>
</div>
</div>
<div id="program-gto2_fq_info" class="section level2">
<h2><span class="header-section-number">3.5</span> Program gto2_fq_info</h2>
<p>The <strong>gto2_fq_info</strong> analyses the basic information of FASTQ file format.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_info</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-4" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_info</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_info [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_info [options]

<span class="ex">It</span> analyses the basic information of FASTQ file format.

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output</span>              Output read information (stdout)

<span class="ex">Example</span>: ./gto2_fq_info <span class="op">&lt;</span> input.fastq <span class="op">&gt;</span> output

<span class="ex">Output</span> example:
<span class="ex">Total</span> reads     : value
<span class="ex">Max</span> read length : value
<span class="ex">Min</span> read length : value
<span class="ex">Min</span> QS value    : value
<span class="ex">Max</span> QS value    : value
<span class="ex">QS</span> range        : value</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-4" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_info</strong> program is a set of information related to the file read. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Total</span> reads     : 2
<span class="ex">Max</span> read length : 72
<span class="ex">Min</span> read length : 72
<span class="ex">Min</span> QS value    : 41
<span class="ex">Max</span> QS value    : 73
<span class="ex">QS</span> range        : 33</code></pre></div>
</div>
</div>
<div id="program-gto2_fq_maximum_read_size" class="section level2">
<h2><span class="header-section-number">3.6</span> Program gto2_fq_maximum_read_size</h2>
<p>The <strong>gto2_fq_maximum_read_size</strong> filters the FASTQ reads with the length higher than the value defined.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_maximum_read_size</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-5" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_maximum_read_size</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_maximum_read_size [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_maximum_read_size [options]

<span class="ex">It</span> filters the FASTQ reads with the length higher than the 
<span class="ex">value</span> defined. 
<span class="ex">If</span> present, it will erase the second header (after +)<span class="ex">.</span>

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="ex">-s</span>, --size=<span class="op">&lt;</span>int<span class="op">&gt;</span>      The maximum read length
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>        Output FASTQ file format (stdout)

<span class="ex">Example</span>: ./gto2_fq_maximum_read_size -s <span class="op">&lt;</span>size<span class="op">&gt;</span> <span class="op">&lt;</span> input.fastq 
<span class="op">&gt;</span> <span class="ex">output.fastq</span>

<span class="ex">Console</span> output example :
<span class="op">&lt;</span><span class="ex">FASTQ</span> non-filtered reads<span class="op">&gt;</span>
<span class="ex">Total</span> reads    : value
<span class="ex">Filtered</span> reads : value</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=59
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-5" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_maximum_read_size</strong> program is a set of all the filtered FASTQ reads, followed by the execution report. The execution report only appears in the console.</p>
<p>Using the input above with the size values as 59, an output example for this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=59
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDII</span>
<span class="ex">Total</span> reads    : 2
<span class="ex">Filtered</span> reads : 1</code></pre></div>
</div>
</div>
<div id="program-gto2_fq_minimum_quality_score" class="section level2">
<h2><span class="header-section-number">3.7</span> Program gto2_fq_minimum_quality_score</h2>
<p>The <strong>gto2_fq_minimum_quality_score</strong> discards reads with average quality-score below of the defined.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_minimum_quality_score</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-6" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_minimum_quality_score</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_minimum_quality_score [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_minimum_quality_score [options]

<span class="ex">It</span> discards reads with average quality-score below value.

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="ex">-m</span>, --min=<span class="op">&lt;</span>int<span class="op">&gt;</span>       The minimum average quality-score 
                          <span class="kw">(</span><span class="ex">Value</span> 25 or 30 is commonly used<span class="kw">)</span>
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>        Output FASTQ file format (stdout)

<span class="ex">Example</span>: ./gto2_fq_minimum_quality_score -m <span class="op">&lt;</span>min<span class="op">&gt;</span> <span class="op">&lt;</span> 
<span class="ex">input.fastq</span> <span class="op">&gt;</span> output.fastq

<span class="ex">Console</span> output example:
<span class="op">&lt;</span><span class="ex">FASTQ</span> non-filtered reads<span class="op">&gt;</span>
<span class="ex">Total</span> reads    : value
<span class="ex">Filtered</span> reads : value</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="op">54599&lt;&gt;</span><span class="ex">77977</span>==6=?I6IBI::<span class="op">33344235521677999&gt;&gt;&gt;&lt;&lt;&lt;</span>@@A@BBCDGGBFF</code></pre></div>
</div>
<div id="output-6" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_minimum_quality_score</strong> program is a set of all the filtered FASTQ reads, followed by the execution report. Using the input above with the minimum averge value as 30, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">Total</span> reads    : 2
<span class="ex">Filtered</span> reads : 1</code></pre></div>
</div>
</div>
<div id="program-gto2_fq_minimum_read_size" class="section level2">
<h2><span class="header-section-number">3.8</span> Program gto2_fq_minimum_read_size</h2>
<p>The <strong>gto2_fq_minimum_read_size</strong> filters the FASTQ reads with the length smaller than the value defined.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_minimum_read_size</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-7" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_minimum_read_size</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_minimum_read_size [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_minimum_read_size [options]

<span class="ex">It</span> filters the FASTQ reads with the length smaller than the 
<span class="ex">value</span> defined. 
<span class="ex">If</span> present, it will erase the second header (after +)<span class="ex">.</span>

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="ex">-s</span>, --size=<span class="op">&lt;</span>int<span class="op">&gt;</span>      The minimum read length
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>        Output FASTQ file format (stdout)

<span class="ex">Example</span>: ./gto2_fq_minimum_read_size -s <span class="op">&lt;</span>size<span class="op">&gt;</span> <span class="op">&lt;</span> input.fastq
<span class="op">&gt;</span> <span class="ex">output.fastq</span>

<span class="ex">Console</span> output example:
<span class="op">&lt;</span><span class="ex">FASTQ</span> non-filtered reads<span class="op">&gt;</span>
<span class="ex">Total</span> reads    : value
<span class="ex">Filtered</span> reads : value</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=50
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAAC</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-7" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_minimum_read_size</strong> program is a set of all the filtered FASTQ reads, followed by the execution report. The execution report only appears in the console. Using the input above with the size values as 55, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span>
<span class="ex">Total</span> reads    : 2
<span class="ex">Filtered</span> reads : 1</code></pre></div>
</div>
</div>
<div id="program-gto2_fq_rand_extra_chars" class="section level2">
<h2><span class="header-section-number">3.9</span> Program gto2_fq_rand_extra_chars</h2>
<p>The <strong>gto2_fq_rand_extra_chars</strong> substitutes the outside ACGT chars by random ACGT symbols in the DNA sequence of FASTQ files.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_rand_extra_chars</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-8" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_rand_extra_chars</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_rand_extra_chars [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_rand_extra_chars [options]

<span class="ex">It</span> substitues in the FASTQ files, the DNA sequence the 
<span class="ex">outside</span> ACGT chars by random ACGT symbols.

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>        Output FASTQ file format (stdout)

<span class="ex">Example</span>: ./gto2_fq_rand_extra_chars <span class="op">&lt;</span> input.fastq <span class="op">&gt;</span> 
<span class="ex">output.fastq</span></code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-8" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_rand_extra_chars</strong> program is a FASTQ file. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GTGTGATGGCCGCTGCCGATGGCGCATAATCCCACCAACATACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGATTGTATTTTAAGAATCTGCAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
</div>
<div id="program-gto2_fq_from_seq" class="section level2">
<h2><span class="header-section-number">3.10</span> Program gto2_fq_from_seq</h2>
<p>The <strong>gto2_fq_from_seq</strong> converts a genomic sequence to pseudo FASTQ file format.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_from_seq</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-9" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_from_seq</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a sequence group file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_from_seq [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_from_seq [options]

<span class="ex">It</span> converts a genomic sequence to pseudo FASTQ file format.

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.seq</span>           Input sequence file (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>        Output FASTQ file format (stdout)

<span class="ex">Optional</span> options
    <span class="ex">-n</span>, --name=<span class="op">&lt;</span>str<span class="op">&gt;</span>      The read<span class="st">&#39;s header</span>
<span class="st">    -l, --lineSize=&lt;int&gt;  The maximum of chars for line</span>

<span class="st">Example: ./gto2_fq_from_seq -l &lt;lineSize&gt; -n &lt;name&gt; </span>
<span class="st">&lt; input.seq &gt; output.fastq</span></code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">ACAAGACGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCCTGGAGGGTCCACCGCT</span>
<span class="ex">GCCCTGCTGCCATTGTCCCCGGCCCCACCTAAGGAAAAGCAGCCTCCTGACTTTCCTCGC</span>
<span class="ex">TTGGGCCGAGACAGCGAGCATATGCAGGAAGCGGCAGGAAGTGGTTTGAGTGGACCTCCG</span>
<span class="ex">GGCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCGGCAGGAAGCAGGCCAGTGCC</span>
<span class="ex">GCGAATCCGCGCGCCGGGACAGAATCTCCTGCAAAGCCCTGCAGGAACTTCTTCTGGAAG</span></code></pre></div>
</div>
<div id="output-9" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_from_seq</strong> program is a pseudo FASTQ file. An example, using the size line as 60 and the read's header as ''SeqToFastq'', for the input, is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SeqToFastq1</span>
<span class="ex">ACAAGACGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCCTGGAGGGTCCACCGCT</span>
<span class="ex">+</span>
<span class="ex">FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF</span>
<span class="ex">@SeqToFastq2</span>
<span class="ex">GCCCTGCTGCCATTGTCCCCGGCCCCACCTAAGGAAAAGCAGCCTCCTGACTTTCCTCGC</span>
<span class="ex">+</span>
<span class="ex">FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF</span>
<span class="ex">@SeqToFastq3</span>
<span class="ex">TTGGGCCGAGACAGCGAGCATATGCAGGAAGCGGCAGGAAGTGGTTTGAGTGGACCTCCG</span>
<span class="ex">+</span>
<span class="ex">FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF</span>
<span class="ex">@SeqToFastq4</span>
<span class="ex">GGCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCGGCAGGAAGCAGGCCAGTGCC</span>
<span class="ex">+</span>
<span class="ex">FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF</span>
<span class="ex">@SeqToFastq5</span>
<span class="ex">GCGAATCCGCGCGCCGGGACAGAATCTCCTGCAAAGCCCTGCAGGAACTTCTTCTGGAAG</span>
<span class="ex">+</span>
<span class="ex">FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF</span></code></pre></div>
</div>
</div>
<div id="program-gto2_fq_mutate" class="section level2">
<h2><span class="header-section-number">3.11</span> Program gto2_fq_mutate</h2>
<p>The <strong>gto2_fq_mutate</strong> creates a synthetic mutation of a FASTQ file given specific rates of mutations, deletions and additions. All these paramenters are defined by the user, and their are optional.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_mutate</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-10" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_mutate</strong> program needs two streams for the computation, namely the input and output standard. However, optional settings can be supplied too, such as the starting point to the random generator, and the edition, deletion and insertion rates. Also, the user can choose to use the ACGTN alphabet in the synthetic mutation. The input stream is a FASTQ File.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_mutate [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_mutate [options]

<span class="ex">Creates</span> a synthetic mutation of a FASTQ file given specific 
<span class="ex">rates</span> of mutations, deletions and additions.

    <span class="ex">-h</span>, --help      show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fasta</span>   Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fasta</span>  Output FASTQ file format (stdout)

<span class="ex">Optional</span>
    <span class="ex">-s</span>              Starting point to the random generator
    <span class="ex">-m</span>              Defines the mutation rate (default 0.0)
    <span class="ex">-d</span>              Defines the deletion rate (default 0.0)
    <span class="ex">-i</span>              Defines the insertion rate (default 0.0)
    <span class="ex">-a</span>              When active, the application uses the 
                    <span class="ex">ACGTN</span> alphabet

<span class="ex">Example</span>: ./gto2_fq_mutate -s <span class="op">&lt;</span>seed<span class="op">&gt;</span> -m <span class="op">&lt;</span>mutation rate<span class="op">&gt;</span> 
<span class="ex">-d</span> <span class="op">&lt;</span>deletion rate<span class="op">&gt;</span> -i <span class="op">&lt;</span>insertion rate<span class="op">&gt;</span> -a 
<span class="op">&lt;</span> <span class="ex">input.fastq</span> <span class="op">&gt;</span> output.fastq</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-10" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_mutate</strong> program is a FASTQ file whith the synthetic mutation of input file. Using the input above with the seed value as 1 and the mutation rate as 0.5, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGACTTTGAGGTGTGGCGATAGACTGAAAACACTTCAGGGTAAAATCACTCGCAAAAGTG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGAGCCTTTACCGTAGGGGTGTAAGATTTTATACAAAAAGTCCAGGTCAAGAGGAA</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
</div>
<div id="program-gto2_fq_split" class="section level2">
<h2><span class="header-section-number">3.12</span> Program gto2_fq_split</h2>
<p>The <strong>gto2_fq_split</strong> splits Paired End files according to the direction of the strand ('/1' or '/2'). It writes by default singleton reads as forward stands.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_split</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-11" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_split</strong> program needs a stream for the computation, namely the input standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_split [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_split [options]

<span class="ex">It</span> writes by default singleton reads as forward stands.

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="ex">-f</span>, --forward=<span class="op">&lt;</span>str<span class="op">&gt;</span>   Output forward file
    <span class="ex">-r</span>, --reverse=<span class="op">&lt;</span>str<span class="op">&gt;</span>   Output reverse file
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output</span>              Output read information (stdout)

<span class="ex">Example</span>: ./gto2_fq_split -f <span class="op">&lt;</span>output_forward.fastq<span class="op">&gt;</span> 
<span class="ex">-r</span> <span class="op">&lt;</span>output_reverse.fastq<span class="op">&gt;</span> <span class="op">&lt;</span> input.fastq <span class="op">&gt;</span> output

<span class="ex">Output</span> example :
<span class="ex">Total</span> reads      : value
<span class="ex">Singleton</span> reads  : value
<span class="ex">Forward</span> reads    : value
<span class="ex">Reverse</span> reads    : value</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60 1
<span class="ex">GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60 2
<span class="ex">NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-11" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_split</strong> program is a set of information related to the file read. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Total</span> reads     : 2
<span class="ex">Singleton</span> reads : 0
<span class="ex">Forward</span> reads   : 65536
<span class="ex">Reverse</span> reads   : 1</code></pre></div>
<p>Also, this program generates two FASTQ files, with the reverse and forward reads.</p>
<p>An example of the forward reads, for the input, is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60 1
<span class="ex">GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span></code></pre></div>
</div>
</div>
<div id="program-gto2_fq_pack" class="section level2">
<h2><span class="header-section-number">3.13</span> Program gto2_fq_pack</h2>
<p>The <strong>gto2_fq_pack</strong> packages each FASTQ read in a single line. It can show the read score first or the dna sequence, depending on the execution mode.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_pack</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-12" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_pack</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_pack [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_pack [options]

<span class="ex">It</span> packages each FASTQ read in a single line.

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastqpack</span>    Output packaged FASTQ file format 
                          <span class="kw">(</span><span class="ex">stdout</span><span class="kw">)</span>

<span class="ex">Optional</span>
    <span class="ex">-s</span>, --scores          When active, the application show 
                          <span class="ex">the</span> scores first

<span class="ex">Example</span>: ./gto2_fq_pack -s <span class="op">&lt;</span> input.fastq <span class="op">&gt;</span> output.fastqpack</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-12" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_pack</strong> program is a packaged FASTQ file. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60+ 0
<span class="ex">NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span>
<span class="ex">SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60+ 1</code></pre></div>
<p>Another example for the same input, but using the scores first (flag ''s''), is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG</span>
<span class="ex">SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60+ 0
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span>
<span class="ex">NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60+ 1</code></pre></div>
</div>
</div>
<div id="program-gto2_fq_unpack" class="section level2">
<h2><span class="header-section-number">3.14</span> Program gto2_fq_unpack</h2>
<p>The <strong>gto2_fq_unpack</strong> unpacks the FASTQ reads packaged using the <strong>gto2_fq_pack</strong> tool.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_unpack</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-13" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_unpack</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a packaged FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_unpack [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_unpack [options]

<span class="ex">It</span> unpacks the FASTQ reads packaged using the gto2_fq_pack 
<span class="ex">tool.</span>

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>        Output FASTQ file format (stdout)

<span class="ex">Optional</span>
    <span class="ex">-s</span>, --scores          When active, the application show 
                          <span class="ex">the</span> scores first
    
<span class="ex">Example</span>: ./gto2_fq_unpack -s <span class="op">&lt;</span> input.fastqpack <span class="op">&gt;</span> out.fastq</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60+ 0
<span class="ex">NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span>
<span class="ex">SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60+ 1</code></pre></div>
</div>
<div id="output-13" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_unpack</strong> program is a FASTQ file. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
</div>
<div id="program-gto2_fq_quality_score_info" class="section level2">
<h2><span class="header-section-number">3.15</span> Program gto2_fq_quality_score_info</h2>
<p>The <strong>gto2_fq_quality_score_info</strong> analyses the quality-scores of a FASTQ file.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_quality_score_info</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-14" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_quality_score_info</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_quality_score_info [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_quality_score_info [options]

<span class="ex">It</span> analyses the quality-scores of a FASTQ file.

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output</span>              Output read information (stdout)
    
<span class="ex">Optional</span>
    <span class="ex">-m</span>, --max=<span class="op">&lt;</span>int<span class="op">&gt;</span>       The lenght of the maximum window

<span class="ex">Example</span>: ./gto2_fq_quality_score_info -m <span class="op">&lt;</span>max<span class="op">&gt;</span> <span class="op">&lt;</span> input.fastq
<span class="op">&gt;</span> <span class="ex">output</span>

<span class="ex">Output</span> example :
<span class="ex">Total</span> reads     : value
<span class="ex">Max</span> read length : value
<span class="ex">Min</span> read length : value
<span class="ex">Min</span> QS value    : value
<span class="ex">Max</span> QS value    : value
<span class="ex">QS</span> range        : value</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@111</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60 1
<span class="ex">GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@222</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60 2
<span class="ex">NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-14" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_quality_score_info</strong> program is a set of information related to the file read. Using the input above with the max window value as 30, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Total</span> reads     : 2
<span class="ex">Max</span> read length : 60
<span class="ex">Min</span> read length : 60
<span class="ex">Min</span> QS value    : 54
<span class="ex">Max</span> QS value    : 73
<span class="ex">QS</span> range        : 20
 <span class="ex">1</span>  ...  24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 
<span class="ex">--+</span> ... +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
<span class="ex">73</span>  ...  73 73 73 73 73 73 73 65 73 62 65 69 70 73 73 73 73 
                               <span class="ex">*</span>     *  *  *  *             
                               <span class="ex">*</span>     *  *  *  *             
                               <span class="ex">*</span>     *  *  *  *             
                               <span class="ex">*</span>     *  *  *                
                               <span class="ex">*</span>     *  *                   
                               <span class="ex">*</span>     *  *                   
                               <span class="ex">*</span>     *  *                   
                               <span class="ex">*</span>     *  *                   
                                     <span class="ex">*</span>                      
                                     <span class="ex">*</span>                      
                                     <span class="ex">*</span>   </code></pre></div>
</div>
</div>
<div id="program-gto2_fq_quality_score_min" class="section level2">
<h2><span class="header-section-number">3.16</span> Program gto2_fq_quality_score_min</h2>
<p>The <strong>gto2_fq_quality_score_min</strong> analyses the minimal quality-scores of a FASTQ file.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_quality_score_min</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-15" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_quality_score_min</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_quality_score_min [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_quality_score_min [options]

<span class="ex">It</span> analyses the minimal quality-scores of a FASTQ file.

    <span class="ex">-h</span>, --help        show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>     Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output</span>          Output read information (stdout)

<span class="ex">Optional</span>
    <span class="ex">-m</span>, --max=<span class="op">&lt;</span>int<span class="op">&gt;</span>   The maximum window length (default 40)

<span class="ex">Example</span>: ./gto2_fq_quality_score_min -m <span class="op">&lt;</span>max<span class="op">&gt;</span> <span class="op">&lt;</span> input.fastq 
<span class="op">&gt;</span> <span class="ex">output</span></code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@111</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60 1
<span class="ex">GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@222</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60 2
<span class="ex">NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-15" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_quality_score_min</strong> program is a set of information related to the file read, considering the minimum quality scores. Using the input above with the max window value as 20, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"> <span class="ex">1</span>  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 
<span class="ex">--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+</span>
<span class="ex">73</span> 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73</code></pre></div>
</div>
</div>
<div id="program-gto2_fq_quality_score_max" class="section level2">
<h2><span class="header-section-number">3.17</span> Program gto2_fq_quality_score_max</h2>
<p>The <strong>gto2_fq_quality_score_max</strong> analyses the maximal quality-scores of a FASTQ file.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_quality_score_max</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-16" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_quality_score_max</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_quality_score_max [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_quality_score_max [options]

<span class="ex">It</span> analyses the maximal quality-scores of a FASTQ file.

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output</span>              Output read information (stdout)

<span class="ex">Optional</span>
    <span class="ex">-m</span>, --max=<span class="op">&lt;</span>int<span class="op">&gt;</span>       The maximum window length (default 40)

<span class="ex">Example</span>: ./gto2_fq_quality_score_max -m <span class="op">&lt;</span>max<span class="op">&gt;</span> 
<span class="op">&lt;</span> <span class="ex">input.fastq</span> <span class="op">&gt;</span> output</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@111</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60 1
<span class="ex">GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@222</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60 2
<span class="ex">NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-16" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_quality_score_max</strong> program is a set of information related to the file read, considering the maximal quality scores. Using the input above with the max window value as 20, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"> <span class="ex">1</span>  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 
<span class="ex">--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+</span>
<span class="ex">73</span> 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73</code></pre></div>
</div>
</div>
<div id="program-gto2_fq_cut" class="section level2">
<h2><span class="header-section-number">3.18</span> Program gto2_fq_cut</h2>
<p>The <strong>gto2_fq_cut</strong> cuts read sequences in a FASTQ file. It requires that the initial and end positions for the cut.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_cut</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-17" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_cut</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_cut [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_cut [options]

<span class="ex">It</span> cuts read sequences in a FASTQ file.

    <span class="ex">-h</span>, --help            show this help message and exit

<span class="ex">Basic</span> options
    <span class="ex">-i</span>, --initial=<span class="op">&lt;</span>int<span class="op">&gt;</span>   Starting position to the cut
    <span class="ex">-e</span>, --end=<span class="op">&lt;</span>int<span class="op">&gt;</span>       Ending position to the cut
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>        Output FASTQ file format (stdout)

<span class="ex">Example</span>: ./gto2_fq_cut -i <span class="op">&lt;</span>initial<span class="op">&gt;</span> -e <span class="op">&lt;</span>end<span class="op">&gt;</span> <span class="op">&lt;</span> input.fastq 
<span class="op">&gt;</span> <span class="ex">output.fastq</span></code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-17" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_cut</strong> program is a FASTQ file cut. Using the initial value as 10 and the end value as 30, an example of this input, is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">CGCTGCCGATGGCGTCAAATC</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIII9</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">ACGACGTTTGTATTTTAAGAA</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIII</span></code></pre></div>
</div>
</div>
<div id="program-gto2_fq_minimum_local_quality_score_forward" class="section level2">
<h2><span class="header-section-number">3.19</span> Program gto2_fq_minimum_local_quality_score_forward</h2>
<p>The <strong>gto2_fq_minimum_local_quality_score_forward</strong> filters the reads considering the quality score average of a defined window size of bases.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_minimum_local_quality_score_forward</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-18" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_minimum_local_quality_score_forward</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_minimum_local_quality_score_forward [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_minimum_local_quality_score_forward [options]

<span class="ex">It</span> filters the reads considering the quality score average 
<span class="ex">of</span> a defined window size of bases.

    <span class="ex">-h</span>, --help        show this help message and exit

<span class="ex">Basic</span> options
    <span class="ex">-k</span>                The window size of bases (default 5)
    <span class="ex">-w</span>                The minimum average of quality score 
                      <span class="kw">(</span><span class="ex">default</span> 25<span class="kw">)</span>
    <span class="ex">-m</span>                The minimum value of the quality score 
                      <span class="kw">(</span><span class="ex">default</span> 33<span class="kw">)</span>
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>     Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>    Output FASTQ file format (stdout)

<span class="ex">Example</span>: ./gto2_fq_minimum_local_quality_score_forward 
<span class="ex">-k</span> <span class="op">&lt;</span>windowsize<span class="op">&gt;</span> -w <span class="op">&lt;</span>minavg<span class="op">&gt;</span> -m <span class="op">&lt;</span>minqs<span class="op">&gt;</span> 
<span class="op">&lt;</span> <span class="ex">input.fastq</span> <span class="op">&gt;</span> output.fastq

<span class="ex">Console</span> output example:
<span class="ex">Minimum</span> QS       : value
<span class="op">&lt;</span><span class="ex">FASTQ</span> output<span class="op">&gt;</span>
<span class="ex">Total</span> reads      : value
<span class="ex">Trimmed</span> reads    : value</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIII-I</span>)<span class="ex">8IIIIIIIIIIIIIIIIII</span></code></pre></div>
</div>
<div id="output-18" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_minimum_local_quality_score_forward</strong> program is a FASTQ file with the reads filtered following a quality score average of a defined window of bases. The execution report only appears in the console. Using the input above with the default values, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Minimum</span> QS     : 33
<span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAA</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBII</span>
<span class="ex">Total</span> reads    : 2
<span class="ex">Trimmed</span> reads  : 1</code></pre></div>
</div>
</div>
<div id="program-gto2_fq_minimum_local_quality_score_reverse" class="section level2">
<h2><span class="header-section-number">3.20</span> Program gto2_fq_minimum_local_quality_score_reverse</h2>
<p>The <strong>gto2_fq_minimum_local_quality_score_reverse</strong> filters the reverse reads, considering the quality score average of a defined window size of bases.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_minimum_local_quality_score_reverse</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-19" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_minimum_local_quality_score_reverse</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_minimum_local_quality_score_reverse [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_minimum_local_quality_score_reverse [options]

<span class="ex">It</span> filters the reverse reads, considering the quality score 
<span class="ex">average</span> of a defined 
<span class="ex">window</span> size of bases.

    <span class="ex">-h</span>, --help        show this help message and exit

<span class="ex">Basic</span> options
    <span class="ex">-k</span>                The window size of bases (default 5)
    <span class="ex">-w</span>                The minimum average of quality score 
                      <span class="kw">(</span><span class="ex">default</span> 25<span class="kw">)</span>
    <span class="ex">-m</span>                The minimum value of the quality score
                      <span class="kw">(</span><span class="ex">default</span> 33<span class="kw">)</span>
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>     Input FASTQ file format (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>    Output FASTQ file format (stdout)

<span class="ex">Example</span>: ./gto2_fq_minimum_local_quality_score_reverse 
<span class="ex">-k</span> <span class="op">&lt;</span>windowsize<span class="op">&gt;</span> -w <span class="op">&lt;</span>minavg<span class="op">&gt;</span> -m <span class="op">&lt;</span>minqs<span class="op">&gt;</span> 
<span class="op">&lt;</span> <span class="ex">input.fastq</span> <span class="op">&gt;</span> output.fastq

<span class="ex">Console</span> output example:
<span class="ex">Minimum</span> QS       : value
<span class="op">&lt;</span><span class="ex">FASTQ</span> output<span class="op">&gt;</span>
<span class="ex">Total</span> reads      : value
<span class="ex">Trimmed</span> reads    : value</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIII-I</span>)<span class="ex">8IIIIIIIIIIIIIIIIII</span></code></pre></div>
</div>
<div id="output-19" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_minimum_local_quality_score_reverse</strong> program is a FASTQ file with the reads filtered following a quality score average of a defined window of bases. The execution report only appears in the console. Using the input above with the default values, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Minimum</span> QS: 33
<span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIII</span>
<span class="ex">Total</span> reads    : 2
<span class="ex">Trimmed</span> reads  : 1</code></pre></div>
</div>
</div>
<div id="program-gto2_fq_xs" class="section level2">
<h2><span class="header-section-number">3.21</span> Program gto2_fq_xs</h2>
<p>The <strong>gto2_fq_xs</strong> is a skilled FASTQ read simulation tool, flexible, portable (does not need a reference sequence) and tunable in terms of sequence complexity. XS handles Ion Torrent, Roche-454, Illumina and ABI-SOLiD simulation sequencing types. It has several running modes, depending on the time and memory available, and is aimed at testing computing infrastructures, namely cloud computing of large-scale projects, and testing FASTQ compression algorithms. Moreover, XS offers the possibility of simulating the three main FASTQ components individually (headers, DNA sequences and quality-scores). Quality-scores can be simulated using uniform and Gaussian distributions.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_xs</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-20" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_xs</strong> program needs a FASTQ file to compute.</p>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-20" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_xs</strong> program is a FASTQ file. Using the input above using the common usage with 5 reads (-n 5), an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"></code></pre></div>
</div>
</div>
<div id="program-gto2_fq_complement" class="section level2">
<h2><span class="header-section-number">3.22</span> Program gto2_fq_complement</h2>
<p>The <strong>gto2_fq_complement</strong> replaces the ACGT bases with their complements in a FASTQ file format.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_complement</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-21" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_complement</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_complement [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_complement [options]

<span class="ex">It</span> replaces the ACGT bases with their complements in a FASTQ
<span class="fu">file</span> format.

    <span class="ex">-h</span>, --help            Show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>        Output FASTQ file (stdout)

<span class="ex">Example</span>: ./gto2_fq_complement <span class="op">&lt;</span> input.fastq <span class="op">&gt;</span> output.fastq</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-21" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_complement</strong> program is the FASTQ file with the ACGT base complements. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">CCCACTACCGGCGACGGCTACCGCAGTTTAGGGTGGTTCAATGGGAATTGTTGAATTCCC</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">CAAGTCCCTATGCTGCAAACATAAAATTCTTAGACTTCGTCTTCAGCTACTATTATGCGC</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
</div>
<div id="program-gto2_fq_reverse" class="section level2">
<h2><span class="header-section-number">3.23</span> Program gto2_fq_reverse</h2>
<p>The <strong>gto2_fq_reverse</strong> reverses the ACGT bases order for each read in a FASTQ file format.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_reverse</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-22" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_reverse</strong> program needs two streams for the computation, namely the input and output standard. The input stream is a FASTQ file.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_reverse [options] [[--] args]
   <span class="ex">or</span>: ./gto2_fq_reverse [options]

<span class="ex">It</span> reverses the ACGT bases order for each read in a FASTQ 
<span class="ex">file.</span>

    <span class="ex">-h</span>, --help            Show this help message and exit

<span class="ex">Basic</span> options
    <span class="op">&lt;</span> <span class="ex">input.fastq</span>         Input FASTQ file (stdin)
    <span class="op">&gt;</span> <span class="ex">output.fastq</span>        Output FASTQ file (stdout)

<span class="ex">Example</span>: ./gto2_fq_reverse <span class="op">&lt;</span> input.fastq <span class="op">&gt;</span> output.fastq</code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-22" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_reverse</strong> program is the FASTQ file complement with the flag ''(Reversed)'' added in the header. Using the input above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 (Reversed)
<span class="ex">GGGAATTCAACAATTCCCATTGAACCACCCTAAACTGCGGTAGCCGTCGCCGGTAGTGGG</span>
<span class="ex">+</span>
<span class="ex">IIIDIIIIIIIIIIIIIIIIIIIICI9GI9IIIIIIIIIIIIIIIIIIIIIIIIIIIIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 (Reversed)
<span class="ex">GCGCATAATAGTAGCTGAAGACGAAGTCTAAGAATTTTATGTTTGCAGCATAGGGACTTG</span>
<span class="ex">+</span>
<span class="ex">IGIIIIIIIIIIIIIIIIIIIIIIIBI6IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII</span></code></pre></div>
</div>
</div>
<div id="program-gto2_fq_variation_map" class="section level2">
<h2><span class="header-section-number">3.24</span> Program gto2_fq_variation_map</h2>
<p>The <strong>gto2_fq_variation_map</strong> identifies the variation that occours in the sequences relative to the reads or a set of reads.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_variation_map</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-23" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_variation_map</strong> program needs FASTQ, FASTA or SEQ files to be used as reference and target files.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_variation_map <span class="op">&lt;</span>OPTIONS<span class="op">&gt;</span>... [FILE]:<span class="op">&lt;</span>...<span class="op">&gt;</span> 
[<span class="ex">FILE</span>]:<span class="op">&lt;</span>...<span class="op">&gt;</span>

<span class="ex">The</span> gto2_fq_variation_map is a tool to map relative 
<span class="ex">singularity</span> regions. The (probabilistic) <span class="ex">Bloom</span> filter is 
<span class="ex">automatically</span> set.   
                                                         
  <span class="ex">-v</span>                       verbose mode,                 
  <span class="ex">-a</span>                       about CHESTER,                
  <span class="ex">-s</span> <span class="op">&lt;</span>value<span class="op">&gt;</span>               bloom size,                   
  <span class="ex">-i</span>                       use inversions,               
  <span class="ex">-p</span>                       show positions/words,         
  <span class="ex">-k</span> <span class="op">&lt;</span>value<span class="op">&gt;</span>               k-mer size (up to 30),        
                                                         
  [<span class="ex">rFile1</span>]:<span class="op">&lt;</span>rFile2<span class="op">&gt;</span>:<span class="op">&lt;</span>...<span class="op">&gt;</span>  reference file(s),            
  [<span class="ex">tFile1</span>]:<span class="op">&lt;</span>tFile2<span class="op">&gt;</span>:<span class="op">&lt;</span>...<span class="op">&gt;</span>  target file(s)<span class="bu">.</span>               
                                                         
<span class="ex">The</span> reference files may be FASTA, FASTQ or DNA SEQ,      
<span class="kw">while</span> <span class="ex">the</span> target files may be FASTA OR DNA SEQ.          
<span class="ex">Report</span> bugs to <span class="op">&lt;</span><span class="dt">{pratas,raquelsilva,ap,pjf}</span>@ua.pt<span class="op">&gt;</span>.</code></pre></div>
<p>An example of a reference file (Multi-FASTA format) is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="op">&gt;</span><span class="ex">AB000264</span> <span class="kw">|</span><span class="va">acc=</span>AB000264<span class="kw">|</span><span class="va">descr=</span>Homo <span class="ex">sapiens</span> mRNA 
<span class="ex">ACAAGACGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCCTGGAGGGTCCACCGCT</span>
<span class="ex">CCCTGCTGCCATTGTCCCCGGCCCCACCTAAGGAAAAGCAGCCTCCTGACTTTCCTCGCT</span>
<span class="ex">TGGGCCGAGACAGCGAGCATATGCAGGAAGCGGCAGGAAGTGGTTTGAGTGGACCTCCGG</span>
<span class="ex">GCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCGGCAGGAAGCAGGCCAGTGCCG</span>
<span class="ex">CGAATCCGCGCGCCGGGACAGAATCTCCTGCAAAGCCCTGCAGGAACTTCTTCTGGAAGA</span>
<span class="ex">CCTTCTCCACCCCCCCAGCTAAAACCTCACCCATGAATGCTCACGCAAGTTTAATTACAG</span>
<span class="ex">ACCTGAA</span>
<span class="op">&gt;</span><span class="ex">AB000263</span> <span class="kw">|</span><span class="va">acc=</span>AB000263<span class="kw">|</span><span class="va">descr=</span>Homo <span class="ex">sapiens</span> mRNA 
<span class="ex">ACAAGATGCCATTGTCCCCCGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCACCG</span>
<span class="ex">CTGCCCTGCCCCTGGAGGGTGGCCCCACCGGCCGAGACAGCGAGCATATGCAGGAAGCGG</span>
<span class="ex">CAGGAATAAGGAAAAGCAGCCTCCTGACTTTCCTCGCTTGGTGGTTTGAGTGGACCTCCC</span>
<span class="ex">AGGCCAGTGCCGGGCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCGGCAGGAAG</span>
<span class="ex">GCGCACCCCCCCAGCAATCCGCGCGCCGGGACAGAATGCCCTGCAGGAACTTCTTCTGGA</span>
<span class="ex">AGACCTTCTCCTCCTGCAAATAAAACCTCACCCATGAATGCTCACGCAAGTTTAATTACA</span>
<span class="ex">GACCTGAA</span></code></pre></div>
<p>An example for the target file (FASTQ format) is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-23" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_variation_map</strong> program is a text file identifying the relative regions. Using the inputs above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">111111111111111111111111111110000000000000000000000000000000</span>
<span class="ex">111111111111111111111111111110000000000000000000000000000000</span>
<span class="ex">000000000000</span></code></pre></div>
</div>
</div>
<div id="program-gto2_fq_variation_filter" class="section level2">
<h2><span class="header-section-number">3.25</span> Program gto2_fq_variation_filter</h2>
<p>The <strong>gto2_fq_variation_filter</strong> filters and segments the regions of singularity from the output of <strong>gto2_fq_variation_map</strong>.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_variation_filter</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-24" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_variation_filter</strong> program needs a the output of <strong>gto2_fq_variation_map</strong> to compute.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_variation_filter <span class="op">&lt;</span>OPTIONS<span class="op">&gt;</span>... [FILE]:<span class="op">&lt;</span>...<span class="op">&gt;</span>      
<span class="ex">The</span> gto2_fq_variation_filter is a tool to filter maps.  
                                                     
  <span class="ex">-v</span>                       verbose mode,             
  <span class="ex">-a</span>                       about CHESTER,            
  <span class="ex">-t</span> <span class="op">&lt;</span>value<span class="op">&gt;</span>               threshold [0.0<span class="kw">;</span><span class="ex">1.0</span>],      
  <span class="ex">-w</span> <span class="op">&lt;</span>value<span class="op">&gt;</span>               window size,              
  <span class="ex">-u</span> <span class="op">&lt;</span>value<span class="op">&gt;</span>               sub-sampling,             
                                                     
  [<span class="ex">tFile1</span>]:<span class="op">&lt;</span>tFile2<span class="op">&gt;</span>:<span class="op">&lt;</span>...<span class="op">&gt;</span>  target file(s)<span class="bu">.</span>           
                                                     
<span class="ex">The</span> target files may be generated by gto2_fq_variation_map.    
<span class="ex">Report</span> bugs to <span class="op">&lt;</span><span class="dt">{pratas,raquelsilva,ap,pjf}</span>@ua.pt<span class="op">&gt;</span>. </code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">111111111111111111111111111110000000000000000000000000000000</span>
<span class="ex">111111111111111111111111111110000000000000000000000000000000</span>
<span class="ex">000000000000</span></code></pre></div>
</div>
<div id="output-24" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_variation_filter</strong> program is a text file with the coordenates of the segmented regions. Using the inputs above, an output example of this is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="co">#132#132</span>
<span class="ex">30</span>:60
<span class="ex">90</span>:130</code></pre></div>
</div>
</div>
<div id="program-gto2_fq_variation_visual" class="section level2">
<h2><span class="header-section-number">3.26</span> Program gto2_fq_variation_visual</h2>
<p>The <strong>gto2_fq_variation_visual</strong> depites the regions of singularity using the output from <strong>gto2_fq_variation_filter</strong> into an SVG image.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_variation_visual</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-25" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_variation_visual</strong> program needs a the output of <strong>gto2_fq_variation_filter</strong> to compute.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">Usage</span>: ./gto2_fq_variation_visual <span class="op">&lt;</span>OPTIONS<span class="op">&gt;</span>... [FILE]:<span class="op">&lt;</span>...<span class="op">&gt;</span>
<span class="ex">The</span> gto2_fq_variation_visual is a tool to visualize relative
<span class="ex">singularity</span> regions.
                                                     
  <span class="ex">-v</span>                       verbose mode,             
  <span class="ex">-a</span>                       about CHESTER,            
  <span class="ex">-e</span> <span class="op">&lt;</span>value<span class="op">&gt;</span>               enlarge painted regions,  
                                                     
  [<span class="ex">tFile1</span>]:<span class="op">&lt;</span>tFile2<span class="op">&gt;</span>:<span class="op">&lt;</span>...<span class="op">&gt;</span>  target file(s)<span class="bu">.</span>           
                                                     
<span class="ex">Report</span> bugs to <span class="op">&lt;</span><span class="dt">{pratas,raquelsilva,ap,pjf}</span>@ua.pt<span class="op">&gt;</span>. </code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="co">#132#132</span>
<span class="ex">30</span>:60
<span class="ex">90</span>:130</code></pre></div>
</div>
<div id="output-25" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_variation_visual</strong> program is a SVG plot with the maps. In the following Figure, is represented the plot using the input above.</p>
<div class="figure"><span id="fig:filters"></span>
<img src="images/gto2FqVariationVisual.png" alt="Execution plot of the variation visual tool using the previous input." width="100%" />
<p class="caption">
Figure 3.1: Execution plot of the variation visual tool using the previous input.
</p>
</div>
</div>
</div>
<div id="program-gto2_fq_metagenomics" class="section level2">
<h2><span class="header-section-number">3.27</span> Program gto2_fq_metagenomics</h2>
<p>The <strong>gto2_fq_metagenomics</strong> is an ultra-fast method to infer metagenomic composition of sequenced reads relative to a database. gto2_fq_metagenomics measures similarity between any FASTQ file (or FASTA), independently from the size, against any multi-FASTA database, such as the entire set of complete genomes from the NCBI. gto2_fq_metagenomics supports single reads, paired-end reads, and compositions of both. It has been tested in many plataforms, such as Illumina MySeq, HiSeq, Novaseq, IonTorrent.</p>
<p>gto2_fq_metagenomics is efficient to detect the presence and authenticate a given species in the FASTQ reads. The core of the method is based on relative data compression. gto2_fq_metagenomics uses variable multi-threading, without multiplying the memory for each thread, being able to run efficiently in a common laptop.</p>
<p>For help type:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">./gto2_fq_metagenomics</span> -h</code></pre></div>
<p>In the following subsections, we explain the input and output parameters.</p>
<div id="input-parameters-26" class="section level3 unnumbered">
<h3>Input parameters</h3>
<p>The <strong>gto2_fq_metagenomics</strong> program needs a FASTQ file to compute.</p>
<p>The attribution is given according to:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">NAME</span>                                                                     
      <span class="ex">The</span> gto2_fq_metagenomics is a tool to infer 
      <span class="ex">metagenomic</span> composition.            
                                                                         
<span class="ex">SYNOPSIS</span>                                                                 
      <span class="ex">gto2_fq_metagenomics</span> [OPTION]... [FILE1]:[FILE2]:... 
      [<span class="ex">FILE</span>]                      
                                                                         
<span class="ex">SAMPLE</span>                                                                   
      <span class="ex">gto2_fq_metagenomics</span> -v -F -l 47 -Z -y pro.com 
      <span class="ex">reads1.fq</span>:reads2.fq DB.fa         
                                                                         
<span class="ex">DESCRIPTION</span>                                                              
      <span class="ex">It</span> infers metagenomic sample composition of sequenced 
      <span class="ex">reads.</span> The core of the method uses a cooperation 
      <span class="ex">between</span> multiple context and tolerant context models 
      <span class="ex">with</span> several depths. The reference sequences must be 
      <span class="kw">in</span> <span class="ex">a</span> multi-FASTA format. The sequenced reads must be 
      <span class="ex">trimmed</span> and in FASTQ format.           
                                                                         
      <span class="ex">Non-mandatory</span> arguments:                                           
                                                                         
      <span class="ex">-h</span>                   give this help,                               
      <span class="ex">-F</span>                   force mode (overwrites top file),             
      <span class="ex">-V</span>                   display version number,                       
      <span class="ex">-v</span>                   verbose mode (more information),              
      <span class="ex">-Z</span>                   database local similarity,                    
      <span class="ex">-s</span>                   show compression levels,                      
                                                                         
      <span class="ex">-l</span> <span class="op">&lt;</span>level<span class="op">&gt;</span>           compression level [1<span class="kw">;</span><span class="ex">47</span>],                    
      <span class="ex">-p</span> <span class="op">&lt;</span>sample<span class="op">&gt;</span>          subsampling (default: 1),                    
      <span class="ex">-t</span> <span class="op">&lt;</span>top<span class="op">&gt;</span>             top of similarity (default: 20),              
      <span class="ex">-n</span> <span class="op">&lt;</span>nThreads<span class="op">&gt;</span>        number of threads (default: 2),              
                                                                         
      <span class="ex">-x</span> <span class="op">&lt;</span>FILE<span class="op">&gt;</span>            similarity top filename,                      
      <span class="ex">-y</span> <span class="op">&lt;</span>FILE<span class="op">&gt;</span>            profile filename (-Z must be on)<span class="bu">.</span>             
                                                                         
      <span class="ex">Mandatory</span> arguments:                                               
                                                                         
      [<span class="ex">FILE1</span>]:[FILE2]:...  metagenomic filename (FASTQ),                 
                           <span class="ex">Use</span> <span class="st">&quot;:&quot;</span> for splitting files.                
                                                                         
      [<span class="ex">FILE</span>]               database filename (Multi-FASTA)<span class="bu">.</span>                </code></pre></div>
<p>An example of such an input file is:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">@SRR001666.1</span> 071112_SLXA-EAS1_s_7:5:1:817:345 length=60
<span class="ex">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII</span>
<span class="ex">@SRR001666.2</span> 071112_SLXA-EAS1_s_7:5:1:801:338 length=60
<span class="ex">GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG</span>
<span class="ex">+</span>
<span class="ex">IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI</span></code></pre></div>
</div>
<div id="output-26" class="section level3 unnumbered">
<h3>Output</h3>
<p>The output of the <strong>gto2_fq_metagenomics</strong> program is a CSV file (top.csv) with the highest probability of being contained in the samples. An example for this CSV file is the following:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">2</span>  66725   12.263   NC_037703.1_Saccharomycodes_ludwigii...
<span class="ex">1</span>  66725   12.263   NC_037703.1_Saccharomycodes_ludwigii...
<span class="ex">3</span>  107123  11.492   NC_012621.1_Nakaseomyces_bacillispor...
<span class="ex">4</span>  107123  11.492   NC_012621.1_Nakaseomyces_bacillispor...
<span class="ex">5</span>  16592   11.153   NC_024030.1_Equus_przewalskii_mitoch...
<span class="ex">6</span>  14583   10.851   NC_021120.1_Bursaphelenchus_mucronat...
<span class="ex">7</span>  162504  10.607   NC_018415.1_Candidatus_Carsonella_ru...
<span class="ex">8</span>  10315   10.586   NC_016117.1_Mnemiopsis_leidyi_mitoch...
<span class="ex">9</span>  162589  10.550   NC_018414.1_Candidatus_Carsonella_ru...
<span class="ex">10</span> 166163  10.476   NC_018416.1_Candidatus_Carsonella_ru...</code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="fasta-tools.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="amino-acid-tools.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["GTO2.pdf"],
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
