<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 General Purpose Tools | GTO 2: The genomics-proteomics toolkit</title>
  <meta name="description" content="To do" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 General Purpose Tools | GTO 2: The genomics-proteomics toolkit" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="To do" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 General Purpose Tools | GTO 2: The genomics-proteomics toolkit" />
  
  <meta name="twitter:description" content="To do" />
  

<meta name="author" content="J. R. Almeida, A. J. Pinho, J. L. Oliveira, D. Pratas" />


<meta name="date" content="2021-04-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="genomic-tools.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="/">GTO 2: The genomics-proteomics toolkit</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#testing"><i class="fa fa-check"></i><b>1.2</b> Testing</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#execution-control"><i class="fa fa-check"></i><b>1.3</b> Execution control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="fasta-tools.html"><a href="fasta-tools.html"><i class="fa fa-check"></i><b>2</b> FASTA Tools</a><ul>
<li class="chapter" data-level="2.1" data-path="fasta-tools.html"><a href="fasta-tools.html#gto2_fa_to_fq"><i class="fa fa-check"></i><b>2.1</b> gto2_fa_to_fq</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="fastq-tools.html"><a href="fastq-tools.html"><i class="fa fa-check"></i><b>3</b> FASTQ Tools</a><ul>
<li class="chapter" data-level="3.1" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_to_fa"><i class="fa fa-check"></i><b>3.1</b> Program gto2_fq_to_fa</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_to_mfa"><i class="fa fa-check"></i><b>3.2</b> Program gto2_fq_to_mfa</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-1"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-1"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_exclude_n"><i class="fa fa-check"></i><b>3.3</b> Program gto2_fq_exclude_n</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-2"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-2"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_extract_quality_scores"><i class="fa fa-check"></i><b>3.4</b> Program gto2_fq_extract_quality_scores</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-3"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-3"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_info"><i class="fa fa-check"></i><b>3.5</b> Program gto2_fq_info</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-4"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-4"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_maximum_read_size"><i class="fa fa-check"></i><b>3.6</b> Program gto2_fq_maximum_read_size</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-5"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-5"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_minimum_quality_score"><i class="fa fa-check"></i><b>3.7</b> Program gto2_fq_minimum_quality_score</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-6"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-6"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_minimum_read_size"><i class="fa fa-check"></i><b>3.8</b> Program gto2_fq_minimum_read_size</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-7"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-7"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_rand_extra_chars"><i class="fa fa-check"></i><b>3.9</b> Program gto2_fq_rand_extra_chars</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-8"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-8"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.10" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_from_seq"><i class="fa fa-check"></i><b>3.10</b> Program gto2_fq_from_seq</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-9"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-9"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.11" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_mutate"><i class="fa fa-check"></i><b>3.11</b> Program gto2_fq_mutate</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-10"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-10"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.12" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_split"><i class="fa fa-check"></i><b>3.12</b> Program gto2_fq_split</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-11"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-11"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.13" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_pack"><i class="fa fa-check"></i><b>3.13</b> Program gto2_fq_pack</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-12"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-12"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.14" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_unpack"><i class="fa fa-check"></i><b>3.14</b> Program gto2_fq_unpack</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-13"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-13"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.15" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_quality_score_info"><i class="fa fa-check"></i><b>3.15</b> Program gto2_fq_quality_score_info</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-14"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-14"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.16" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_quality_score_min"><i class="fa fa-check"></i><b>3.16</b> Program gto2_fq_quality_score_min</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-15"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-15"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.17" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_quality_score_max"><i class="fa fa-check"></i><b>3.17</b> Program gto2_fq_quality_score_max</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-16"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-16"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.18" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_cut"><i class="fa fa-check"></i><b>3.18</b> Program gto2_fq_cut</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-17"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-17"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.19" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_minimum_local_quality_score_forward"><i class="fa fa-check"></i><b>3.19</b> Program gto2_fq_minimum_local_quality_score_forward</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-18"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-18"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.20" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_minimum_local_quality_score_reverse"><i class="fa fa-check"></i><b>3.20</b> Program gto2_fq_minimum_local_quality_score_reverse</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-19"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-19"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.21" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_xs"><i class="fa fa-check"></i><b>3.21</b> Program gto2_fq_xs</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-20"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-20"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.22" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_complement"><i class="fa fa-check"></i><b>3.22</b> Program gto2_fq_complement</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-21"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-21"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.23" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_reverse"><i class="fa fa-check"></i><b>3.23</b> Program gto2_fq_reverse</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-22"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-22"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.24" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_variation_map"><i class="fa fa-check"></i><b>3.24</b> Program gto2_fq_variation_map</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-23"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-23"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.25" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_variation_filter"><i class="fa fa-check"></i><b>3.25</b> Program gto2_fq_variation_filter</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-24"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-24"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.26" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_variation_visual"><i class="fa fa-check"></i><b>3.26</b> Program gto2_fq_variation_visual</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-25"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-25"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.27" data-path="fastq-tools.html"><a href="fastq-tools.html#program-gto2_fq_metagenomics"><i class="fa fa-check"></i><b>3.27</b> Program gto2_fq_metagenomics</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters-26"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output-26"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html"><i class="fa fa-check"></i><b>4</b> Amino Acid Tools</a><ul>
<li class="chapter" data-level="4.1" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#program-gto2_aa_to_group"><i class="fa fa-check"></i><b>4.1</b> Program gto2_aa_to_group</a><ul>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#input-parameters-27"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#output-27"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#program-gto2_aa_to_pseudo_dna"><i class="fa fa-check"></i><b>4.2</b> Program gto2_aa_to_pseudo_dna</a><ul>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#input-parameters-28"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#output-28"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#program-gto2_aa_compressor"><i class="fa fa-check"></i><b>4.3</b> Program gto2_aa_compressor</a></li>
<li class="chapter" data-level="4.4" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#program-gto2_aa_from_fa"><i class="fa fa-check"></i><b>4.4</b> Program gto2_aa_from_fa</a><ul>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#input-parameters-29"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#output-29"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#program-gto2_aa_from_fq"><i class="fa fa-check"></i><b>4.5</b> Program gto2_aa_from_fq</a><ul>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#input-parameters-30"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#output-30"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#program-gto2_aa_from_seq"><i class="fa fa-check"></i><b>4.6</b> Program gto2_aa_from_seq</a><ul>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#input-parameters-31"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#output-31"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genomic-tools.html"><a href="genomic-tools.html"><i class="fa fa-check"></i><b>5</b> Genomic Tools</a><ul>
<li class="chapter" data-level="5.1" data-path="genomic-tools.html"><a href="genomic-tools.html#gto2_dna"><i class="fa fa-check"></i><b>5.1</b> gto2_dna</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="general-purpose-tools.html"><a href="general-purpose-tools.html"><i class="fa fa-check"></i><b>6</b> General Purpose Tools</a><ul>
<li class="chapter" data-level="6.1" data-path="general-purpose-tools.html"><a href="general-purpose-tools.html#gto2_"><i class="fa fa-check"></i><b>6.1</b> gto2_</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GTO 2: The genomics-proteomics toolkit</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="general-purpose-tools" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> General Purpose Tools</h1>
<div id="gto2_" class="section level2">
<h2><span class="header-section-number">6.1</span> gto2_</h2>
<p>to do</p>

<div id="refs" class="references">
<div>
<p>Afgan, Enis, Dannon Baker, Bérénice Batut, Marius Van Den Beek, Dave Bouvier, Martin Čech, John Chilton, et al. 2018. “The Galaxy Platform for Accessible, Reproducible and Collaborative Biomedical Analyses: 2018 Update.” <em>Nucleic Acids Research</em> 46 (W1). Oxford University Press: W537–W544.</p>
</div>
<div>
<p>Baruzzo, Giacomo, Katharina E Hayer, Eun Ji Kim, Barbara Di Camillo, Garret A FitzGerald, and Gregory R Grant. 2017. “Simulation-Based Comprehensive Benchmarking of Rna-Seq Aligners.” <em>Nature Methods</em> 14 (2). Nature Publishing Group: 135.</p>
</div>
<div>
<p>Blankenberg, Daniel, Assaf Gordon, Gregory Von Kuster, Nathan Coraor, James Taylor, Anton Nekrutenko, and Galaxy Team. 2010. “Manipulation of Fastq Data with Galaxy.” <em>Bioinformatics</em> 26 (14). Oxford University Press: 1783–5.</p>
</div>
<div>
<p>Brouwer, CPJM, Thuy Duong Vu, Miaomiao Zhou, Gianluigi Cardinali, Mick M Welling, Nathalie van de Wiele, and Vincent Robert. 2016. “Current Opportunities and Challenges of Next Generation Sequencing (Ngs) of Dna; Determining Health and Disease.” <em>British Biotechnology Journal</em> 13 (4). Sciencedomain International.</p>
</div>
<div>
<p>Cock, Peter JA, Christopher J Fields, Naohisa Goto, Michael L Heuer, and Peter M Rice. 2009. “The Sanger Fastq File Format for Sequences with Quality Scores, and the Solexa/Illumina Fastq Variants.” <em>Nucleic Acids Research</em> 38 (6). Oxford University Press: 1767–71.</p>
</div>
<div>
<p>Deorowicz, Sebastian, Agnieszka Danek, and Marcin Niemiec. 2015. “GDC 2: Compression of Large Collections of Genomes.” <em>Scientific Reports</em> 5. Nature Publishing Group: 11565.</p>
</div>
<div>
<p>DePristo, Mark A, Eric Banks, Ryan Poplin, Kiran V Garimella, Jared R Maguire, Christopher Hartl, Anthony A Philippakis, et al. 2011. “A Framework for Variation Discovery and Genotyping Using Next-Generation Dna Sequencing Data.” <em>Nature Genetics</em> 43 (5). Nature Publishing Group: 491.</p>
</div>
<div>
<p>Droop, Alastair P. 2016. “Fqtools: An Efficient Software Suite for Modern Fastq File Manipulation.” <em>Bioinformatics</em> 32 (12). Oxford University Press: 1883–4.</p>
</div>
<div>
<p>Escalona, Merly, Sara Rocha, and David Posada. 2016. “A Comparison of Tools for the Simulation of Genomic Next-Generation Sequencing Data.” <em>Nature Reviews Genetics</em> 17 (8). Nature Publishing Group: 459.</p>
</div>
<div>
<p>Goecks, Jeremy, Anton Nekrutenko, and James Taylor. 2010. “Galaxy: A Comprehensive Approach for Supporting Accessible, Reproducible, and Transparent Computational Research in the Life Sciences.” <em>Genome Biology</em> 11 (8). BioMed Central: R86.</p>
</div>
<div>
<p>Gordon, Assaf, GJ Hannon, and others. 2010. “Fastx-Toolkit.” <em>FASTQ/A Short-Reads Preprocessing Tools (Unpublished) Http://Hannonlab. Cshl. Edu/Fastx_toolkit</em> 5.</p>
</div>
<div>
<p>Hernaez, Mikel, Dmitri Pavlichin, Tsachy Weissman, and Idoia Ochoa. 2019. “Genomic Data Compression.” <em>Annual Review of Biomedical Data Science</em> 2. Annual Reviews.</p>
</div>
<div>
<p>Hosseini, Morteza, Diogo Pratas, and Armando Pinho. 2016. “A Survey on Data Compression Methods for Biological Sequences.” <em>Information</em> 7 (4). Multidisciplinary Digital Publishing Institute: 56.</p>
</div>
<div>
<p>Hosseini, Morteza, Diogo Pratas, and Armando J Pinho. 2019. “AC: A Compression Tool for Amino Acid Sequences.” <em>Interdisciplinary Sciences: Computational Life Sciences</em>. Springer, 1–9.</p>
</div>
<div>
<p>Huang, Weichun, Leping Li, Jason R Myers, and Gabor T Marth. 2011. “ART: A Next-Generation Sequencing Read Simulator.” <em>Bioinformatics</em> 28 (4). Oxford University Press: 593–94.</p>
</div>
<div>
<p>Kumar, Sudhir, Glen Stecher, and Koichiro Tamura. 2016. “MEGA7: Molecular Evolutionary Genetics Analysis Version 7.0 for Bigger Datasets.” <em>Molecular Biology and Evolution</em> 33 (7). Society for Molecular Biology; Evolution: 1870–4.</p>
</div>
<div>
<p>Liu, Lin, Yinhu Li, Siliang Li, Ni Hu, Yimin He, Ray Pong, Danni Lin, Lihua Lu, and Maggie Law. 2012. “Comparison of Next-Generation Sequencing Systems.” <em>BioMed Research International</em> 2012. Hindawi Publishing Corporation.</p>
</div>
<div>
<p>Liu, Yuansheng, Hui Peng, Limsoon Wong, and Jinyan Li. 2017. “High-Speed and High-Ratio Referential Genome Compression.” <em>Bioinformatics</em> 33 (21). Oxford University Press: 3364–72.</p>
</div>
<div>
<p>Mardis, Elaine R. 2017. “DNA Sequencing Technologies: 2006–2016.” <em>Nature Protocols</em> 12 (2). Nature Publishing Group: 213.</p>
</div>
<div>
<p>Nalbantoglu, Özkan, David Russell, and Khalid Sayood. 2010. “Data Compression Concepts and Algorithms and Their Applications to Bioinformatics.” <em>Entropy</em> 12 (1). Molecular Diversity Preservation International: 34–52.</p>
</div>
<div>
<p>Ochoa, Idoia, Mikel Hernaez, and Tsachy Weissman. 2014. “IDoComp: A Compression Scheme for Assembled Genomes.” <em>Bioinformatics</em> 31 (5). Oxford University Press: 626–33.</p>
</div>
<div>
<p>Pratas, Diogo, Morteza Hosseini, and Armando J Pinho. 2018. “Compression of Amino Acid Sequences.” In <em>International Conference on Practical Applications of Computational Biology &amp; Bioinformatics</em>, 105–13. Springer.</p>
</div>
<div>
<p>Pratas, Diogo, Armando J Pinho, and Paulo JSG Ferreira. 2016. “Efficient Compression of Genomic Sequences.” In <em>2016 Data Compression Conference (Dcc)</em>, 231–40. IEEE.</p>
</div>
<div>
<p>Pratas, Diogo, Armando J Pinho, and João MOS Rodrigues. 2014. “XS: A Fastq Read Simulator.” <em>BMC Research Notes</em> 7 (1). BioMed Central: 40.</p>
</div>
<div>
<p>Shen, Wei, Shuai Le, Yan Li, and Fuquan Hu. 2016. “SeqKit: A Cross-Platform and Ultrafast Toolkit for Fasta/Q File Manipulation.” <em>PLoS One</em> 11 (10). Public Library of Science: e0163962.</p>
</div>
<div>
<p>Tamura, Koichiro, Daniel Peterson, Nicholas Peterson, Glen Stecher, Masatoshi Nei, and Sudhir Kumar. 2011. “MEGA5: Molecular Evolutionary Genetics Analysis Using Maximum Likelihood, Evolutionary Distance, and Maximum Parsimony Methods.” <em>Molecular Biology and Evolution</em> 28 (10). Oxford University Press: 2731–9.</p>
</div>
<div>
<p>Van der Auwera, Geraldine A, Mauricio O Carneiro, Christopher Hartl, Ryan Poplin, Guillermo Del Angel, Ami Levy-Moonshine, Tadeusz Jordan, et al. 2013. “From Fastq Data to High-Confidence Variant Calls: The Genome Analysis Toolkit Best Practices Pipeline.” <em>Current Protocols in Bioinformatics</em> 43 (1). Wiley Online Library: 11–10.</p>
</div>
<div>
<p>Zhang, Hongen. 2016. “Overview of Sequence Data Formats.” In <em>Statistical Genomics</em>, 3–17. Springer.</p>
</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="genomic-tools.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["GTO2.pdf"],
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
