\documentclass[11pt,]{krantz}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage[]{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\PassOptionsToPackage{hyphens}{url} % url is loaded by hyperref
\usepackage[unicode=true]{hyperref}
\PassOptionsToPackage{usenames,dvipsnames}{color} % color is loaded by hyperref
\hypersetup{
            pdftitle={GTO 2: The genomics-proteomics toolkit},
            pdfauthor={J. R. Almeida, A. J. Pinho, J. L. Oliveira, D. Pratas},
            colorlinks=true,
            linkcolor=Maroon,
            citecolor=Blue,
            urlcolor=Blue,
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{natbib}
\bibliographystyle{apalike}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.27,0.27,0.27}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.27,0.27,0.27}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.06,0.06,0.06}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.06,0.06,0.06}{#1}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.06,0.06,0.06}{#1}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0,0,0}{#1}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.5,0.5,0.5}{#1}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0,0,0}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.5,0.5,0.5}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.5,0.5,0.5}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.5,0.5,0.5}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textit{#1}}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textbf{\textit{#1}}}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textbf{\textit{#1}}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textbf{\textit{#1}}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0,0,0}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0,0,0}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.27,0.27,0.27}{\textbf{#1}}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.43,0.43,0.43}{\textbf{#1}}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textit{#1}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.61,0.61,0.61}{#1}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textbf{\textit{#1}}}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textbf{\textit{#1}}}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.33,0.33,0.33}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.14,0.14,0.14}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\usepackage{longtable,booktabs}
% Fix footnotes in tables (requires footnote package)
\IfFileExists{footnote.sty}{\usepackage{footnote}\makesavenoteenv{long table}}{}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

% set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother

\usepackage{booktabs}
\usepackage{longtable}
\usepackage[bf,singlelinecheck=off]{caption}

\usepackage{framed,color}
\definecolor{shadecolor}{RGB}{248,248,248}

\renewcommand{\textfraction}{0.05}
\renewcommand{\topfraction}{0.8}
\renewcommand{\bottomfraction}{0.8}
\renewcommand{\floatpagefraction}{0.75}

\renewenvironment{quote}{\begin{VF}}{\end{VF}}
\let\oldhref\href
\renewcommand{\href}[2]{#2\footnote{\url{#1}}}

\ifxetex
  \usepackage{letltxmacro}
  \setlength{\XeTeXLinkMargin}{1pt}
  \LetLtxMacro\SavedIncludeGraphics\includegraphics
  \def\includegraphics#1#{% #1 catches optional stuff (star/opt. arg.)
    \IncludeGraphicsAux{#1}%
  }%
  \newcommand*{\IncludeGraphicsAux}[2]{%
    \XeTeXLinkBox{%
      \SavedIncludeGraphics#1{#2}%
    }%
  }%
\fi

\usepackage{makeidx}
\makeindex

\urlstyle{tt}

\usepackage{amsthm}
\makeatletter
\def\thm@space@setup{%
  \thm@preskip=8pt plus 2pt minus 4pt
  \thm@postskip=\thm@preskip
}
\makeatother

\frontmatter

\title{GTO 2: The genomics-proteomics toolkit}
\author{J. R. Almeida, A. J. Pinho, J. L. Oliveira, D. Pratas}
\date{2021-04-24}

\begin{document}
\maketitle

{
\hypersetup{linkcolor=black}
\setcounter{tocdepth}{1}
\tableofcontents
}
\listoftables
\listoffigures
\chapter*{Preface}\label{preface}


\section*{License}\label{license}


This document was written in
\href{https://rmarkdown.rstudio.com}{RMarkdown} using the
\href{https://bookdown.org}{bookdown} package.

\mainmatter

\chapter{Introduction}\label{introduction}

Recent advances in DNA sequencing, specifically in next-generation
sequencing (NGS), revolutionised the field of genomics, making possible
the generation of large amounts of sequencing data very rapidly and at
substantially low cost\citep{mardis2017dna}. This new technology also
brought with it several challenges, namely in what concerns the
analysis, storage, and transmission of the generated
sequences\citep[\citet{liu2012comparison}]{brouwer2016current}. As a
consequence, several specialised tools were developed throughout the
years in order to deal with these challenges.

Firstly, the storage of the raw data generated by NGS experiments is
possible by using several file formats, the FASTQ and FASTA are the most
commonly used\citep{zhang2016overview}. FASTQ is an extension of the
FASTA format, that besides the nucleotide sequence, also stores
associated per base quality score and it is considered the standard
format for sequencing data storage and exchange\citep{cock2009sanger}.

Regarding the analysis and manipulation of these sequencing data files
many software applications emerged, including
\textbf{fqtools}\citep{droop2016fqtools},
\textbf{FASTX-Toolkit}\citep{gordon2010fastx},
\textbf{GALAXY}\citep{afgan2018galaxy},
\textbf{GATK}\citep{depristo2011framework},
\textbf{MEGA}\citep{kumar2016mega7},
\textbf{SeqKit}\citep{shen2016seqkit}, among others. \textbf{Fqtools} is
a suite of tools to view, manipulate and summarise FASTQ data. This
software also identifies invalid FASTQ files\citep{droop2016fqtools}.
\textbf{GALAXY}, in its turn, is an open, web-based scientific platform
for analysing genomic data\citep{goecks2010galaxy}. This platform
integrates several specialised sets of tools, e.g.~for manipulating
FASTQ files\citep{blankenberg2010manipulation}. \textbf{FASTX-Toolkit}
is a collection of command-line tools to process FASTA and FASTQ files.
This toolkit is available in two forms: as a command-line, or integrated
into the web-based platform \textbf{GALAXY}\citep{gordon2010fastx}.
\textbf{SeqKit} is another toolkit used to process FASTA and FASTQ files
and is available for all major operating systems\citep{shen2016seqkit}.
The Genome Analysis Toolkit (\textbf{GATK}) was designed as a structured
programming framework to simplify the development of analysis tools.
However, nowadays, it is a suite of tools focused on variant discovering
and genotyping\citep{van2013fastq}. More towards the evolutionary
perspectives, Molecular Evolutionary Genetics Analysis (\textbf{MEGA})
software provides tools to analyse DNA and protein sequences
statistically\citep{tamura2011mega5}. Several of these frameworks lack
on variety, namely the ability to perform multiple tasks using only one
toolkit.

Compression is another important aspect when dealing with
high-throughput sequencing data, as it reduces storage space and
accelerates data transmission. A survey on DNA compressors and amino
acid sequence compression can be found in\citep{hosseini2016survey}.
Currently, the DNA sequence compressors HiRGC\citep{liu2017high},
iDoComp\citep{ochoa2014idocomp}, GeCo\citep{pratas2016efficient}, and
GDC\citep{deorowicz2015gdc} are considered to have the best
performance\citep{hernaez2019genomic}. Of these four approaches, GeCo is
the only one that can be used for reference-free and reference-based
compression. Furthermore, GeCo can be used as an analysis tool to
determine absolute measures for many distance computations and local
measures\citep{pratas2016efficient}.

Amino acid sequences are known to be very hard to
compress\citep{nalbantoglu2010data}, however, Hosseini et
al.\citep{hosseini2019ac} recently developed AC, a state-of-the-art for
lossless amino acid sequence compression.
In\citep{pratas2018compression} the authors compared the performance of
AC, in terms of bit-rate, to several general-purpose lossless
compressors and several protein compressors, using different proteomes.
They concluded that in average AC provides the best bit-rates.

Another relevant subject is genomic data simulation. Read simulations
tools are fundamental for the development, testing and evaluation of
methods and computational tools\citep[price2017simulome]{huang2011art}.
Despite the availability of a large number of real sequence reads, read
simulation data is necessary due to the inability to know the ground
truth of real data\citep{baruzzo2017simulation}. Escalona
\textit{et al.}\citep{escalona2016comparison}, recently, reviewed 23 NGS
simulation tools. XS\citep{pratas2014xs}, a FASTQ read simulation tool,
stands out in relation to the other 22 simulation tools because it is
the only one that does not need a reference sequence. Furthermore, XS is
the only open-source tool for simulation of FASTQ reads produced by the
four most used sequencing machines, Roche-454, Illumina, ABI SOLiD and
Ion Torrent.

Although a large number of tools are available for analysing,
compressing, and simulation, these tools are specialised in only a
specific task. Besides, in many cases the output of one tool cannot be
used directly as input for another tool, e.g.~the output of a simulation
tool cannot always be used directly as input for an analysis tool. Thus,
unique software that includes several specialised tools is necessary.

In this document, we describe \textbf{GTO2}, a complete toolkit for
genomics and proteomics, namely for FASTQ, FASTA and SEQ formats, with
many complementary tools. The toolkit is for Unix-based systems, built
for ultra-fast computations. \textbf{GTO2} supports pipes for easy
integration with the sub-programs belonging to \textbf{GTO2} as well as
external tools. \textbf{GTO2} works as \textbf{LEGOs}, since it allows
the construction of multiple pipelines with many combinations.

\textbf{GTO2} includes tools for information display, randomisation,
edition, conversion, extraction, search, calculation, compression,
simulation and visualisation. \textbf{GTO2} is prepared to deal with
very large datasets, typically in the scale of Gigabytes or Terabytes
(but not limited). The complete toolkit is an optimised command-line
version, using the prefix \texttt{gto2\_} followed by the suffix with
the respective name of the program. \textbf{GTO2} is implemented in
\textbf{C} language and it is available, under the MIT license, at
\url{https://github.com/cobilab/gto2}

\section{Installation}\label{installation}

To install \textbf{GTO2} through the GitHub repository:

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{git}\NormalTok{ clone https://github.com/cobilab/gto2.git}
\BuiltInTok{cd}\NormalTok{ gto2/src/}
\FunctionTok{make}
\end{Highlighting}
\end{Shaded}

Or by installing them directly using the Cobilab channel from Conda:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{conda}\NormalTok{ install -c cobilab gto2 --yes}
\end{Highlighting}
\end{Shaded}

\section{Testing}\label{testing}

The examples provided in this document are available in the repository.
Therefore, each example can be easily reproduced, which it will also
test and validate each tool. To replicate those tests, it can be done in
two different ways:

\begin{itemize}
\tightlist
\item
  Running one test for a specific tool:

  \begin{itemize}
  \tightlist
  \item
    cd gto2/tester/gto2\_\{tool\}
  \item
    sh runExample.sh
  \end{itemize}
\item
  Running the batch of tests for all the tools:

  \begin{itemize}
  \tightlist
  \item
    cd gto2/tester/
  \item
    sh runAllTests.sh
  \end{itemize}
\end{itemize}

Some of these tests require internet connection to download external
files and it will create new files.

\section{Execution control}\label{execution-control}

The quality control in Unix/Linux pipelines using GTO's tools is made in
three ways:

\begin{itemize}
\item
  Input verification: where the tools verify the format of the input
  file;
\item
  Stderr logs: Some execution errors are directly sent for the stderr
  channel.
\item
  Scripting validation: In complex pipelines, the verification of all
  the tools in the pipeline were executed properly, it is used the
  PIPESTATUS variable, e.g.:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{gto2_fa_rand_extra_chars} \OperatorTok{<}\NormalTok{ input.fa }\KeywordTok{|} \KeywordTok{\textbackslash{}}
\ExtensionTok{gto2_fa_to_seq} \OperatorTok{>}\NormalTok{ output.seq }
\BuiltInTok{echo} \StringTok{"}\VariableTok{$\{PIPESTATUS[0]\}}\StringTok{ }\VariableTok{$\{PIPESTATUS[1]\}}\StringTok{"} 
\ExtensionTok{0}\NormalTok{ 0 }
\end{Highlighting}
\end{Shaded}
\end{itemize}

\chapter{FASTA Tools}\label{fasta-tools}

\section{gto2\_fa\_to\_fq}\label{gto2_fa_to_fq}

to do

\chapter{FASTQ Tools}\label{fastq-tools}

The toolkit has a set of tools dedicated to manipulating FASTQ files.
Some of these tools allow the data conversion to/from different formats,
i. e., there are tools designed to convert a FASTQ file into a sequence
or a FASTA/Multi-FASTA format, or converting DNA in some of those
formats to FASTQ.

There are also tools for data manipulation in this format, which are
designed to exclude `N', remove low quality scored reads, following
different metrics and randomize DNA sequences. Succeeding the
manipulation, it is also possible to perform analyses over these files,
simulations and mutations. The current available tools for FASTQ format
analysis and manipulation include:

\begin{itemize}
\tightlist
\item
  \textbf{gto2\_fq\_to\_fa}: it converts a FASTQ file format to a pseudo
  FASTA file.
\item
  \textbf{gto2\_fq\_to\_mfa}: it converts a FASTQ file format to a
  pseudo Multi-FASTA file.
\item
  \textbf{gto2\_fq\_exclude\_n}: it discards the FASTQ reads with the
  minimum number of ``N'' symbols.
\item
  \textbf{gto2\_fq\_extract\_quality\_scores}: it extracts all the
  quality-scores from FASTQ reads.
\item
  \textbf{gto2\_fq\_info}: it analyses the basic information of FASTQ
  file format.
\item
  \textbf{gto2\_fq\_maximum\_read\_size}: it filters the FASTQ reads
  with the length higher than the value defined.
\item
  \textbf{gto2\_fq\_minimum\_quality\_score}: it discards reads with
  average quality-score below of the defined.
\item
  \textbf{gto2\_fq\_minimum\_read\_size}: it filters the FASTQ reads
  with the length smaller than the value defined.
\item
  \textbf{gto2\_fq\_rand\_extra\_chars}: it substitues in the FASTQ
  files, the DNA sequence the outside ACGT chars by random ACGT symbols.
\item
  \textbf{gto2\_fq\_from\_seq}: it converts a genomic sequence to pseudo
  FASTQ file format.
\item
  \textbf{gto2\_fq\_mutate}: it creates a synthetic mutation of a FASTQ
  file given specific rates of mutations, deletions and additions.
\item
  \textbf{gto2\_fq\_split}: it splits Paired End files according to the
  direction of the strand (`/1' or `/2').
\item
  \textbf{gto2\_fq\_pack}: it packages each FASTQ read in a single line.
\item
  \textbf{gto2\_fq\_unpack}: it unpacks the FASTQ reads packaged using
  the \textbf{gto2\_fastq\_pack} tool.
\item
  \textbf{gto2\_fq\_quality\_score\_info}: it analyses the
  quality-scores of a FASTQ file.
\item
  \textbf{gto2\_fq\_quality\_score\_min}: it analyses the minimal
  quality-scores of a FASTQ file.
\item
  \textbf{gto2\_fq\_quality\_score\_max}: it analyses the maximal
  quality-scores of a FASTQ file.
\item
  \textbf{gto2\_fq\_cut}: it cuts read sequences in a FASTQ file.
\item
  \textbf{gto2\_fq\_minimum\_local\_quality\_score\_forward}: it filters
  the reads considering the quality score average of a defined window
  size of bases.
\item
  \textbf{gto2\_fq\_minimum\_local\_quality\_score\_reverse}: it filters
  the reverse reads, considering the average window size score defined
  by the bases.
\item
  \textbf{gto2\_fq\_xs}: it is a skilled FASTQ read simulation tool,
  flexible, portable and tunable in terms of sequence complexity.
\item
  \textbf{gto2\_fq\_complement}: it replaces the ACGT bases with their
  complements in a FASTQ file format.
\item
  \textbf{gto2\_fq\_reverse}: it reverses the ACGT bases order for each
  read in a FASTQ file format.
\item
  \textbf{gto2\_fq\_variation\_map}: it identifies the variation that
  occours in the sequences relative to the reads or a set of reads.
\item
  \textbf{gto2\_fq\_variation\_filter}: it filters and segments the
  regions of singularity from the output of
  \textbf{gto2\_fq\_variation\_map}.
\item
  \textbf{gto2\_fq\_variation\_visual}: it depites the regions of
  singularity using the output from \textbf{gto2\_fq\_variation\_filter}
  into an SVG image.
\item
  \textbf{gto2\_fq\_metagenomics}: it measures similarity between any
  FASTQ file, independently from the size, against any multi-FASTA
  database.
\end{itemize}

\section{gto2\_fq\_to\_fa}\label{gto2_fq_to_fa}

The \textbf{gto2\_fq\_to\_fa} converts a FASTQ file format to a pseudo
FASTA file. However, it does not align the sequence. Also, it extracts
the sequence and adds a pseudo header.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_to_fa}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output paramters.

\subsection*{Input parameters}\label{input-parameters}


The \textbf{gto2\_fq\_to\_fa} program needs two streams for the
computation, namely the input and output standard. The input stream is a
FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_to_fa [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_to_fa [options]}

\ExtensionTok{It}\NormalTok{ converts a FASTQ file format to a pseudo FASTA file.}
\ExtensionTok{It}\NormalTok{ does NOT align the sequence.}
\ExtensionTok{It}\NormalTok{ extracts the sequence and adds a pseudo header.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fasta}\NormalTok{        Output FASTA file format (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_to_fa }\OperatorTok{<}\NormalTok{ input.fastq }\OperatorTok{>}\NormalTok{ output.fasta}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output}


The output of the \textbf{gto2\_fq\_to\_fa} program a FASTA file. Using
the input above, an output example for this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\OperatorTok{>} \ExtensionTok{Computed}\NormalTok{ with Fastq2Fasta}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\end{Highlighting}
\end{Shaded}

\section{gto2\_fq\_to\_mfa}\label{gto2_fq_to_mfa}

to do

\section{gto2\_fq\_exclude\_n}\label{gto2_fq_exclude_n}

to do

\section{gto2\_fq\_extract\_quality\_scores}\label{gto2_fq_extract_quality_scores}

to do

\section{gto2\_fq\_info}\label{gto2_fq_info}

to do

\section{gto2\_fq\_maximum\_read\_size}\label{gto2_fq_maximum_read_size}

to do

\section{gto2\_fq\_minimum\_quality\_score}\label{gto2_fq_minimum_quality_score}

to do

\section{gto2\_fq\_minimum\_read\_size}\label{gto2_fq_minimum_read_size}

to do

\section{gto2\_fq\_rand\_extra\_chars}\label{gto2_fq_rand_extra_chars}

to do

\section{gto2\_fq\_from\_seq}\label{gto2_fq_from_seq}

to do

\section{gto2\_fq\_mutate}\label{gto2_fq_mutate}

to do

\section{gto2\_fq\_split}\label{gto2_fq_split}

to do

\section{gto2\_fq\_pack}\label{gto2_fq_pack}

to do

\section{gto2\_fq\_unpack}\label{gto2_fq_unpack}

to do

\section{gto2\_fq\_quality\_score\_info}\label{gto2_fq_quality_score_info}

to do

\section{gto2\_fq\_quality\_score\_min}\label{gto2_fq_quality_score_min}

to do

\section{gto2\_fq\_quality\_score\_max}\label{gto2_fq_quality_score_max}

to do

\section{gto2\_fq\_cut}\label{gto2_fq_cut}

to do

\section{gto2\_fq\_minimum\_local\_quality\_score\_forward}\label{gto2_fq_minimum_local_quality_score_forward}

to do

\section{gto2\_fq\_minimum\_local\_quality\_score\_reverse}\label{gto2_fq_minimum_local_quality_score_reverse}

to do

\section{gto2\_fq\_xs}\label{gto2_fq_xs}

to do

\section{gto2\_fq\_complement}\label{gto2_fq_complement}

to do

\section{gto2\_fq\_reverse}\label{gto2_fq_reverse}

to do

\section{gto2\_fq\_variation\_map}\label{gto2_fq_variation_map}

to do

\section{gto2\_fq\_variation\_filter}\label{gto2_fq_variation_filter}

to do

\section{gto2\_fq\_variation\_visual}\label{gto2_fq_variation_visual}

to do

\section{gto2\_fq\_metagenomics}\label{gto2_fq_metagenomics}

to do

\chapter{Amino Acid Tools}\label{amino-acid-tools}

\section{gto2\_aa}\label{gto2_aa}

to do

\chapter{Genomic Tools}\label{genomic-tools}

\section{gto2\_dna}\label{gto2_dna}

to do

\chapter{General Purpose Tools}\label{general-purpose-tools}

\section{gto2\_}\label{gto2_}

to do

\bibliography{refs.bib}

\printindex

\end{document}
