\documentclass[11pt,]{krantz}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage[]{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\PassOptionsToPackage{hyphens}{url} % url is loaded by hyperref
\usepackage[unicode=true]{hyperref}
\PassOptionsToPackage{usenames,dvipsnames}{color} % color is loaded by hyperref
\hypersetup{
            pdftitle={GTO 2: The genomics-proteomics toolkit},
            pdfauthor={J. R. Almeida, A. J. Pinho, J. L. Oliveira, D. Pratas},
            colorlinks=true,
            linkcolor=Maroon,
            citecolor=Blue,
            urlcolor=Blue,
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{natbib}
\bibliographystyle{apalike}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.27,0.27,0.27}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.27,0.27,0.27}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.06,0.06,0.06}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.06,0.06,0.06}{#1}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.06,0.06,0.06}{#1}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0,0,0}{#1}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.5,0.5,0.5}{#1}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0,0,0}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.5,0.5,0.5}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.5,0.5,0.5}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.5,0.5,0.5}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textit{#1}}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textbf{\textit{#1}}}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textbf{\textit{#1}}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textbf{\textit{#1}}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0,0,0}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0,0,0}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.27,0.27,0.27}{\textbf{#1}}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.43,0.43,0.43}{\textbf{#1}}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textit{#1}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.61,0.61,0.61}{#1}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textbf{\textit{#1}}}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textbf{\textit{#1}}}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.33,0.33,0.33}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.14,0.14,0.14}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\usepackage{longtable,booktabs}
% Fix footnotes in tables (requires footnote package)
\IfFileExists{footnote.sty}{\usepackage{footnote}\makesavenoteenv{long table}}{}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

% set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother

\usepackage{booktabs}
\usepackage{longtable}
\usepackage[bf,singlelinecheck=off]{caption}

\usepackage{framed,color}
\definecolor{shadecolor}{RGB}{248,248,248}

\renewcommand{\textfraction}{0.05}
\renewcommand{\topfraction}{0.8}
\renewcommand{\bottomfraction}{0.8}
\renewcommand{\floatpagefraction}{0.75}

\renewenvironment{quote}{\begin{VF}}{\end{VF}}
\let\oldhref\href
\renewcommand{\href}[2]{#2\footnote{\url{#1}}}

\ifxetex
  \usepackage{letltxmacro}
  \setlength{\XeTeXLinkMargin}{1pt}
  \LetLtxMacro\SavedIncludeGraphics\includegraphics
  \def\includegraphics#1#{% #1 catches optional stuff (star/opt. arg.)
    \IncludeGraphicsAux{#1}%
  }%
  \newcommand*{\IncludeGraphicsAux}[2]{%
    \XeTeXLinkBox{%
      \SavedIncludeGraphics#1{#2}%
    }%
  }%
\fi

\usepackage{makeidx}
\makeindex

\urlstyle{tt}

\usepackage{amsthm}
\makeatletter
\def\thm@space@setup{%
  \thm@preskip=8pt plus 2pt minus 4pt
  \thm@postskip=\thm@preskip
}
\makeatother

\frontmatter

\title{GTO 2: The genomics-proteomics toolkit}
\author{J. R. Almeida, A. J. Pinho, J. L. Oliveira, D. Pratas}
\date{2021-04-24}

\begin{document}
\maketitle

{
\hypersetup{linkcolor=black}
\setcounter{tocdepth}{1}
\tableofcontents
}
\listoftables
\listoffigures
\chapter*{Preface}\label{preface}


\section*{License}\label{license}


This document was written in
\href{https://rmarkdown.rstudio.com}{RMarkdown} using the
\href{https://bookdown.org}{bookdown} package.

\mainmatter

\chapter{Introduction}\label{introduction}

Recent advances in DNA sequencing, specifically in next-generation
sequencing (NGS), revolutionised the field of genomics, making possible
the generation of large amounts of sequencing data very rapidly and at
substantially low cost\citep{mardis2017dna}. This new technology also
brought with it several challenges, namely in what concerns the
analysis, storage, and transmission of the generated
sequences\citep[\citet{liu2012comparison}]{brouwer2016current}. As a
consequence, several specialised tools were developed throughout the
years in order to deal with these challenges.

Firstly, the storage of the raw data generated by NGS experiments is
possible by using several file formats, the FASTQ and FASTA are the most
commonly used\citep{zhang2016overview}. FASTQ is an extension of the
FASTA format, that besides the nucleotide sequence, also stores
associated per base quality score and it is considered the standard
format for sequencing data storage and exchange\citep{cock2009sanger}.

Regarding the analysis and manipulation of these sequencing data files
many software applications emerged, including
\textbf{fqtools}\citep{droop2016fqtools},
\textbf{FASTX-Toolkit}\citep{gordon2010fastx},
\textbf{GALAXY}\citep{afgan2018galaxy},
\textbf{GATK}\citep{depristo2011framework},
\textbf{MEGA}\citep{kumar2016mega7},
\textbf{SeqKit}\citep{shen2016seqkit}, among others. \textbf{Fqtools} is
a suite of tools to view, manipulate and summarise FASTQ data. This
software also identifies invalid FASTQ files\citep{droop2016fqtools}.
\textbf{GALAXY}, in its turn, is an open, web-based scientific platform
for analysing genomic data\citep{goecks2010galaxy}. This platform
integrates several specialised sets of tools, e.g.~for manipulating
FASTQ files\citep{blankenberg2010manipulation}. \textbf{FASTX-Toolkit}
is a collection of command-line tools to process FASTA and FASTQ files.
This toolkit is available in two forms: as a command-line, or integrated
into the web-based platform \textbf{GALAXY}\citep{gordon2010fastx}.
\textbf{SeqKit} is another toolkit used to process FASTA and FASTQ files
and is available for all major operating systems\citep{shen2016seqkit}.
The Genome Analysis Toolkit (\textbf{GATK}) was designed as a structured
programming framework to simplify the development of analysis tools.
However, nowadays, it is a suite of tools focused on variant discovering
and genotyping\citep{van2013fastq}. More towards the evolutionary
perspectives, Molecular Evolutionary Genetics Analysis (\textbf{MEGA})
software provides tools to analyse DNA and protein sequences
statistically\citep{tamura2011mega5}. Several of these frameworks lack
on variety, namely the ability to perform multiple tasks using only one
toolkit.

Compression is another important aspect when dealing with
high-throughput sequencing data, as it reduces storage space and
accelerates data transmission. A survey on DNA compressors and amino
acid sequence compression can be found in\citep{hosseini2016survey}.
Currently, the DNA sequence compressors HiRGC\citep{liu2017high},
iDoComp\citep{ochoa2014idocomp}, GeCo\citep{pratas2016efficient}, and
GDC\citep{deorowicz2015gdc} are considered to have the best
performance\citep{hernaez2019genomic}. Of these four approaches, GeCo is
the only one that can be used for reference-free and reference-based
compression. Furthermore, GeCo can be used as an analysis tool to
determine absolute measures for many distance computations and local
measures\citep{pratas2016efficient}.

Amino acid sequences are known to be very hard to
compress\citep{nalbantoglu2010data}, however, Hosseini et
al.\citep{hosseini2019ac} recently developed AC, a state-of-the-art for
lossless amino acid sequence compression.
In\citep{pratas2018compression} the authors compared the performance of
AC, in terms of bit-rate, to several general-purpose lossless
compressors and several protein compressors, using different proteomes.
They concluded that in average AC provides the best bit-rates.

Another relevant subject is genomic data simulation. Read simulations
tools are fundamental for the development, testing and evaluation of
methods and computational tools\citep[price2017simulome]{huang2011art}.
Despite the availability of a large number of real sequence reads, read
simulation data is necessary due to the inability to know the ground
truth of real data\citep{baruzzo2017simulation}. Escalona
\textit{et al.}\citep{escalona2016comparison}, recently, reviewed 23 NGS
simulation tools. XS\citep{pratas2014xs}, a FASTQ read simulation tool,
stands out in relation to the other 22 simulation tools because it is
the only one that does not need a reference sequence. Furthermore, XS is
the only open-source tool for simulation of FASTQ reads produced by the
four most used sequencing machines, Roche-454, Illumina, ABI SOLiD and
Ion Torrent.

Although a large number of tools are available for analysing,
compressing, and simulation, these tools are specialised in only a
specific task. Besides, in many cases the output of one tool cannot be
used directly as input for another tool, e.g.~the output of a simulation
tool cannot always be used directly as input for an analysis tool. Thus,
unique software that includes several specialised tools is necessary.

In this document, we describe \textbf{GTO2}, a complete toolkit for
genomics and proteomics, namely for FASTQ, FASTA and SEQ formats, with
many complementary tools. The toolkit is for Unix-based systems, built
for ultra-fast computations. \textbf{GTO2} supports pipes for easy
integration with the sub-programs belonging to \textbf{GTO2} as well as
external tools. \textbf{GTO2} works as \textbf{LEGOs}, since it allows
the construction of multiple pipelines with many combinations.

\textbf{GTO2} includes tools for information display, randomisation,
edition, conversion, extraction, search, calculation, compression,
simulation and visualisation. \textbf{GTO2} is prepared to deal with
very large datasets, typically in the scale of Gigabytes or Terabytes
(but not limited). The complete toolkit is an optimised command-line
version, using the prefix \texttt{gto2\_} followed by the suffix with
the respective name of the program. \textbf{GTO2} is implemented in
\textbf{C} language and it is available, under the MIT license, at
\url{https://github.com/cobilab/gto2}

\section{Installation}\label{installation}

To install \textbf{GTO2} through the GitHub repository:

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{git}\NormalTok{ clone https://github.com/cobilab/gto2.git}
\BuiltInTok{cd}\NormalTok{ gto2/src/}
\FunctionTok{make}
\end{Highlighting}
\end{Shaded}

Or by installing them directly using the Cobilab channel from Conda:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{conda}\NormalTok{ install -c cobilab gto2 -y}
\end{Highlighting}
\end{Shaded}

\section{Testing}\label{testing}

The examples provided in this document are available in the repository.
Therefore, each example can be easily reproduced, which it will also
test and validate each tool. To replicate those tests, it can be done in
two different ways:

\begin{itemize}
\tightlist
\item
  Running one test for a specific tool:

  \begin{itemize}
  \tightlist
  \item
    cd gto2/tester/gto2\_\{tool\}
  \item
    sh runExample.sh
  \end{itemize}
\item
  Running the batch of tests for all the tools:

  \begin{itemize}
  \tightlist
  \item
    cd gto2/tester/
  \item
    sh runAllTests.sh
  \end{itemize}
\end{itemize}

Some of these tests require internet connection to download external
files and it will create new files.

\section{Execution control}\label{execution-control}

The quality control in Unix/Linux pipelines using GTO's tools is made in
three ways:

\begin{itemize}
\item
  Input verification: where the tools verify the format of the input
  file;
\item
  Stderr logs: Some execution errors are directly sent for the stderr
  channel.
\item
  Scripting validation: In complex pipelines, the verification of all
  the tools in the pipeline were executed properly, it is used the
  PIPESTATUS variable, e.g.:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{gto2_fa_rand_extra_chars} \OperatorTok{<}\NormalTok{ input.fa }\KeywordTok{|} \KeywordTok{\textbackslash{}}
\ExtensionTok{gto2_fa_to_seq} \OperatorTok{>}\NormalTok{ output.seq }
\BuiltInTok{echo} \StringTok{"}\VariableTok{$\{PIPESTATUS[0]\}}\StringTok{ }\VariableTok{$\{PIPESTATUS[1]\}}\StringTok{"} 
\ExtensionTok{0}\NormalTok{ 0 }
\end{Highlighting}
\end{Shaded}
\end{itemize}

\chapter{FASTA Tools}\label{fasta-tools}

\section{gto2\_fa\_to\_fq}\label{gto2_fa_to_fq}

to do

\chapter{FASTQ Tools}\label{fastq-tools}

The toolkit has a set of tools dedicated to manipulating FASTQ files.
Some of these tools allow the data conversion to/from different formats,
i. e., there are tools designed to convert a FASTQ file into a sequence
or a FASTA/Multi-FASTA format, or converting DNA in some of those
formats to FASTQ.

There are also tools for data manipulation in this format, which are
designed to exclude `N', remove low quality scored reads, following
different metrics and randomize DNA sequences. Succeeding the
manipulation, it is also possible to perform analyses over these files,
simulations and mutations. The current available tools for FASTQ format
analysis and manipulation include:

\begin{itemize}
\tightlist
\item
  \textbf{gto2\_fq\_to\_fa}: to convert a FASTQ file format to a pseudo
  FASTA file.
\item
  \textbf{gto2\_fq\_to\_mfa}: to convert a FASTQ file format to a pseudo
  Multi-FASTA file.
\item
  \textbf{gto2\_fq\_exclude\_n}: to discard the FASTQ reads with the
  minimum number of ``N'' symbols.
\item
  \textbf{gto2\_fq\_extract\_quality\_scores}: to extract all the
  quality-scores from FASTQ reads.
\item
  \textbf{gto2\_fq\_info}: to analyse the basic information of FASTQ
  file format.
\item
  \textbf{gto2\_fq\_maximum\_read\_size}: to filter the FASTQ reads with
  the length higher than the value defined.
\item
  \textbf{gto2\_fq\_minimum\_quality\_score}: to discard reads with
  average quality-score below of the defined.
\item
  \textbf{gto2\_fq\_minimum\_read\_size}: to filter the FASTQ reads with
  the length smaller than the value defined.
\item
  \textbf{gto2\_fq\_rand\_extra\_chars}: to substitue in the FASTQ
  files, the DNA sequence the outside ACGT chars by random ACGT symbols.
\item
  \textbf{gto2\_fq\_from\_seq}: to convert a genomic sequence to pseudo
  FASTQ file format.
\item
  \textbf{gto2\_fq\_mutate}: to create a synthetic mutation of a FASTQ
  file given specific rates of mutations, deletions and additions.
\item
  \textbf{gto2\_fq\_split}: to split Paired End files according to the
  direction of the strand (`/1' or `/2').
\item
  \textbf{gto2\_fq\_pack}: to package each FASTQ read in a single line.
\item
  \textbf{gto2\_fq\_unpack}: to unpack the FASTQ reads packaged using
  the \textbf{gto2\_fq\_pack} tool.
\item
  \textbf{gto2\_fq\_quality\_score\_info}: to analyse the quality-scores
  of a FASTQ file.
\item
  \textbf{gto2\_fq\_quality\_score\_min}: to analyse the minimal
  quality-scores of a FASTQ file.
\item
  \textbf{gto2\_fq\_quality\_score\_max}: to analyse the maximal
  quality-scores of a FASTQ file.
\item
  \textbf{gto2\_fq\_cut}: to cut read sequences in a FASTQ file.
\item
  \textbf{gto2\_fq\_minimum\_local\_quality\_score\_forward}: to filter
  the reads considering the quality score average of a defined window
  size of bases.
\item
  \textbf{gto2\_fq\_minimum\_local\_quality\_score\_reverse}: to filter
  the reverse reads, considering the average window size score defined
  by the bases.
\item
  \textbf{gto2\_fq\_xs}: a skilled FASTQ read simulation tool, flexible,
  portable and tunable in terms of sequence complexity.
\item
  \textbf{gto2\_fq\_complement}: to replace the ACGT bases with their
  complements in a FASTQ file format.
\item
  \textbf{gto2\_fq\_reverse}: to reverse the ACGT bases order for each
  read in a FASTQ file format.
\item
  \textbf{gto2\_fq\_variation\_map}: to identify the variation that
  occours in the sequences relative to the reads or a set of reads.
\item
  \textbf{gto2\_fq\_variation\_filter}: to filter and segments the
  regions of singularity from the output of
  \textbf{gto2\_fq\_variation\_map}.
\item
  \textbf{gto2\_fq\_variation\_visual}: to depict the regions of
  singularity using the output from \textbf{gto2\_fq\_variation\_filter}
  into an SVG image.
\item
  \textbf{gto2\_fq\_metagenomics}: to measure the similarity between any
  FASTQ file, independently from the size, against any multi-FASTA
  database.
\end{itemize}

\section{Program gto2\_fq\_to\_fa}\label{program-gto2_fq_to_fa}

The \textbf{gto2\_fq\_to\_fa} converts a FASTQ file format to a pseudo
FASTA file. However, this tool does not align the sequence, instead, it
extracts the sequence and adds a pseudo-header.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_to_fa}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters}


The \textbf{gto2\_fq\_to\_fa} program needs two streams for the
computation, namely the input and output standard. The input stream is a
FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_to_fa [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_to_fa [options]}

\ExtensionTok{It}\NormalTok{ converts a FASTQ file format to a pseudo FASTA file.}
\ExtensionTok{It}\NormalTok{ does NOT align the sequence.}
\ExtensionTok{It}\NormalTok{ extracts the sequence and adds a pseudo header.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fasta}\NormalTok{        Output FASTA file format (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_to_fa }\OperatorTok{<}\NormalTok{ input.fastq }\OperatorTok{>}\NormalTok{ output.fasta}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output}


The output of the \textbf{gto2\_fq\_to\_fa} program is a FASTA file.
Using the input above, an output example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\OperatorTok{>} \ExtensionTok{Computed}\NormalTok{ with Fastq2Fasta}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\end{Highlighting}
\end{Shaded}

\section{Program gto2\_fq\_to\_mfa}\label{program-gto2_fq_to_mfa}

The \textbf{gto2\_fq\_to\_mfa} converts a FASTQ file format to a pseudo
Multi-FASTA file. However, this tool does not align the sequence,
instead, it extracts the sequence and adds a pseudo header.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_to_mfa}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-1}


The \textbf{gto2\_fq\_to\_mfa} program needs two streams for the
computation, namely the input and output standard. The input stream is a
FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_to_mfa [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_to_mfa [options]}

\ExtensionTok{It}\NormalTok{ converts a FASTQ file format to a pseudo Multi-FASTA }
\ExtensionTok{file.}\NormalTok{ It does NOT align the sequence. It extracts the }
\ExtensionTok{sequence}\NormalTok{ and adds each header in a Multi-FASTA format.}

    \ExtensionTok{-h}\NormalTok{, --help       show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{    Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.mfasta}\NormalTok{  Output Multi-FASTA file format (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_to_mfa }\OperatorTok{<}\NormalTok{ input.fastq }\OperatorTok{>}\NormalTok{ output.mfasta}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-1}


The output of the \textbf{gto2\_fq\_to\_mfa} program is a Multi-FASTA
file. Using the input above, an output example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\OperatorTok{>}\ExtensionTok{SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\OperatorTok{>}\ExtensionTok{SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\end{Highlighting}
\end{Shaded}

\section{Program gto2\_fq\_exclude\_n}\label{program-gto2_fq_exclude_n}

The \textbf{gto2\_fq\_exclude\_n} discards the FASTQ reads with the
minimum number of `'N'' symbols, and it will erase the second header
(after +), if presented.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_exclude_n}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-2}


The \textbf{gto2\_fq\_exclude\_n} program needs two streams for the
computation, namely the input and output standard. The input stream is a
FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_exclude_n [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_exclude_n [options]}

\ExtensionTok{It}\NormalTok{ discards the FASTQ reads with the minimum number of }\StringTok{"N"} 
\ExtensionTok{symbols.} 
\ExtensionTok{If}\NormalTok{ present, it will erase the second header (after +)}\ExtensionTok{.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \ExtensionTok{-m}\NormalTok{, --max=}\OperatorTok{<}\NormalTok{int}\OperatorTok{>}\NormalTok{       The maximum of of }\StringTok{"N"}\NormalTok{ symbols in }
                          \ExtensionTok{the}\NormalTok{ read}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{        Output FASTQ file format (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_exclude_n -m }\OperatorTok{<}\NormalTok{max}\OperatorTok{>} \OperatorTok{<}\NormalTok{ input.fastq }\OperatorTok{>} 
\ExtensionTok{output.fastq}

\ExtensionTok{Console}\NormalTok{ output example :}
\OperatorTok{<}\ExtensionTok{FASTQ}\NormalTok{ non-filtered reads}\OperatorTok{>}
\ExtensionTok{Total}\NormalTok{ reads    : value}
\ExtensionTok{Filtered}\NormalTok{ reads : value}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-2}


The output of the \textbf{gto2\_fq\_exclude\_n} program is a set of all
the filtered FASTQ reads, followed by the execution report. The
execution report only appears in the console.

Using the input above with the max value as 5, an output example for
this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\ExtensionTok{Total}\NormalTok{ reads    : 2}
\ExtensionTok{Filtered}\NormalTok{ reads : 1}
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_extract\_quality\_scores}\label{program-gto2_fq_extract_quality_scores}

The \textbf{gto2\_fq\_extract\_quality\_scores} extracts all the
quality-scores from FASTQ reads.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_extract_quality_scores}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-3}


The \textbf{gto2\_fq\_extract\_quality\_scores} program needs two
streams for the computation, namely the input and output standard. The
input stream is a FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_extract_quality_scores [options] [[--]args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_extract_quality_scores [options]}

\ExtensionTok{It}\NormalTok{ extracts all the quality-scores from FASTQ reads.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{        Output FASTQ file format (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_extract_quality_scores }\OperatorTok{<}\NormalTok{ input.fastq }\OperatorTok{>}
\ExtensionTok{output.fastq}

\ExtensionTok{Console}\NormalTok{ output example:}
\OperatorTok{<}\ExtensionTok{FASTQ}\NormalTok{ quality scores}\OperatorTok{>}
\ExtensionTok{Total}\NormalTok{ reads          : value}
\ExtensionTok{Total}\NormalTok{ Quality-Scores : value}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-3}


The output of the \textbf{gto2\_fq\_extract\_quality\_scores} program is
a set of all the quality scores from the FASTQ reads, followed by the
execution report. The execution report only appears in the console.
Using the input above, an output example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\ExtensionTok{Total}\NormalTok{ reads          : 2}
\ExtensionTok{Total}\NormalTok{ Quality-Scores : 144}
\end{Highlighting}
\end{Shaded}

\section{Program gto2\_fq\_info}\label{program-gto2_fq_info}

The \textbf{gto2\_fq\_info} analyses the basic information of FASTQ file
format.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_info}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-4}


The \textbf{gto2\_fq\_info} program needs two streams for the
computation, namely the input and output standard. The input stream is a
FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_info [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_info [options]}

\ExtensionTok{It}\NormalTok{ analyses the basic information of FASTQ file format.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output}\NormalTok{              Output read information (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_info }\OperatorTok{<}\NormalTok{ input.fastq }\OperatorTok{>}\NormalTok{ output}

\ExtensionTok{Output}\NormalTok{ example:}
\ExtensionTok{Total}\NormalTok{ reads     : value}
\ExtensionTok{Max}\NormalTok{ read length : value}
\ExtensionTok{Min}\NormalTok{ read length : value}
\ExtensionTok{Min}\NormalTok{ QS value    : value}
\ExtensionTok{Max}\NormalTok{ QS value    : value}
\ExtensionTok{QS}\NormalTok{ range        : value}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-4}


The output of the \textbf{gto2\_fq\_info} program is a set of
information related to the file read. Using the input above, an output
example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Total}\NormalTok{ reads     : 2}
\ExtensionTok{Max}\NormalTok{ read length : 72}
\ExtensionTok{Min}\NormalTok{ read length : 72}
\ExtensionTok{Min}\NormalTok{ QS value    : 41}
\ExtensionTok{Max}\NormalTok{ QS value    : 73}
\ExtensionTok{QS}\NormalTok{ range        : 33}
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_maximum\_read\_size}\label{program-gto2_fq_maximum_read_size}

The \textbf{gto2\_fq\_maximum\_read\_size} filters the FASTQ reads with
the length higher than the value defined.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_maximum_read_size}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-5}


The \textbf{gto2\_fq\_maximum\_read\_size} program needs two streams for
the computation, namely the input and output standard. The input stream
is a FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_maximum_read_size [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_maximum_read_size [options]}

\ExtensionTok{It}\NormalTok{ filters the FASTQ reads with the length higher than the }
\ExtensionTok{value}\NormalTok{ defined. }
\ExtensionTok{If}\NormalTok{ present, it will erase the second header (after +)}\ExtensionTok{.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \ExtensionTok{-s}\NormalTok{, --size=}\OperatorTok{<}\NormalTok{int}\OperatorTok{>}\NormalTok{      The maximum read length}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{        Output FASTQ file format (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_maximum_read_size -s }\OperatorTok{<}\NormalTok{size}\OperatorTok{>} \OperatorTok{<}\NormalTok{ input.fastq }
\OperatorTok{>} \ExtensionTok{output.fastq}

\ExtensionTok{Console}\NormalTok{ output example :}
\OperatorTok{<}\ExtensionTok{FASTQ}\NormalTok{ non-filtered reads}\OperatorTok{>}
\ExtensionTok{Total}\NormalTok{ reads    : value}
\ExtensionTok{Filtered}\NormalTok{ reads : value}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=59}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-5}


The output of the \textbf{gto2\_fq\_maximum\_read\_size} program is a
set of all the filtered FASTQ reads, followed by the execution report.
The execution report only appears in the console.

Using the input above with the size values as 59, an output example for
this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=59}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDII}
\ExtensionTok{Total}\NormalTok{ reads    : 2}
\ExtensionTok{Filtered}\NormalTok{ reads : 1}
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_minimum\_quality\_score}\label{program-gto2_fq_minimum_quality_score}

The \textbf{gto2\_fq\_minimum\_quality\_score} discards reads with
average quality-score below of the defined.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_minimum_quality_score}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-6}


The \textbf{gto2\_fq\_minimum\_quality\_score} program needs two streams
for the computation, namely the input and output standard. The input
stream is a FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_minimum_quality_score [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_minimum_quality_score [options]}

\ExtensionTok{It}\NormalTok{ discards reads with average quality-score below value.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \ExtensionTok{-m}\NormalTok{, --min=}\OperatorTok{<}\NormalTok{int}\OperatorTok{>}\NormalTok{       The minimum average quality-score }
                          \KeywordTok{(}\ExtensionTok{Value}\NormalTok{ 25 or 30 is commonly used}\KeywordTok{)}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{        Output FASTQ file format (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_minimum_quality_score -m }\OperatorTok{<}\NormalTok{min}\OperatorTok{>} \OperatorTok{<} 
\ExtensionTok{input.fastq} \OperatorTok{>}\NormalTok{ output.fastq}

\ExtensionTok{Console}\NormalTok{ output example:}
\OperatorTok{<}\ExtensionTok{FASTQ}\NormalTok{ non-filtered reads}\OperatorTok{>}
\ExtensionTok{Total}\NormalTok{ reads    : value}
\ExtensionTok{Filtered}\NormalTok{ reads : value}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\OperatorTok{54599<>}\ExtensionTok{77977}\NormalTok{==6=?I6IBI::}\OperatorTok{33344235521677999>>><<<}\NormalTok{@@A@BBCDGGBFF}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-6}


The output of the \textbf{gto2\_fq\_minimum\_quality\_score} program is
a set of all the filtered FASTQ reads, followed by the execution report.
Using the input above with the minimum averge value as 30, an output
example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{Total}\NormalTok{ reads    : 2}
\ExtensionTok{Filtered}\NormalTok{ reads : 1}
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_minimum\_read\_size}\label{program-gto2_fq_minimum_read_size}

The \textbf{gto2\_fq\_minimum\_read\_size} filters the FASTQ reads with
the length smaller than the value defined.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_minimum_read_size}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-7}


The \textbf{gto2\_fq\_minimum\_read\_size} program needs two streams for
the computation, namely the input and output standard. The input stream
is a FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_minimum_read_size [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_minimum_read_size [options]}

\ExtensionTok{It}\NormalTok{ filters the FASTQ reads with the length smaller than the }
\ExtensionTok{value}\NormalTok{ defined. }
\ExtensionTok{If}\NormalTok{ present, it will erase the second header (after +)}\ExtensionTok{.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \ExtensionTok{-s}\NormalTok{, --size=}\OperatorTok{<}\NormalTok{int}\OperatorTok{>}\NormalTok{      The minimum read length}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{        Output FASTQ file format (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_minimum_read_size -s }\OperatorTok{<}\NormalTok{size}\OperatorTok{>} \OperatorTok{<}\NormalTok{ input.fastq}
\OperatorTok{>} \ExtensionTok{output.fastq}

\ExtensionTok{Console}\NormalTok{ output example:}
\OperatorTok{<}\ExtensionTok{FASTQ}\NormalTok{ non-filtered reads}\OperatorTok{>}
\ExtensionTok{Total}\NormalTok{ reads    : value}
\ExtensionTok{Filtered}\NormalTok{ reads : value}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=50}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAAC}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-7}


The output of the \textbf{gto2\_fq\_minimum\_read\_size} program is a
set of all the filtered FASTQ reads, followed by the execution report.
The execution report only appears in the console. Using the input above
with the size values as 55, an output example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\ExtensionTok{Total}\NormalTok{ reads    : 2}
\ExtensionTok{Filtered}\NormalTok{ reads : 1}
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_rand\_extra\_chars}\label{program-gto2_fq_rand_extra_chars}

The \textbf{gto2\_fq\_rand\_extra\_chars} substitutes the outside ACGT
chars by random ACGT symbols in the DNA sequence of FASTQ files.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_rand_extra_chars}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-8}


The \textbf{gto2\_fq\_rand\_extra\_chars} program needs two streams for
the computation, namely the input and output standard. The input stream
is a FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_rand_extra_chars [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_rand_extra_chars [options]}

\ExtensionTok{It}\NormalTok{ substitues in the FASTQ files, the DNA sequence the }
\ExtensionTok{outside}\NormalTok{ ACGT chars by random ACGT symbols.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{        Output FASTQ file format (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_rand_extra_chars }\OperatorTok{<}\NormalTok{ input.fastq }\OperatorTok{>} 
\ExtensionTok{output.fastq}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-8}


The output of the \textbf{gto2\_fq\_rand\_extra\_chars} program is a
FASTQ file. Using the input above, an output example of this is the
following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GTGTGATGGCCGCTGCCGATGGCGCATAATCCCACCAACATACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGATTGTATTTTAAGAATCTGCAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\section{Program gto2\_fq\_from\_seq}\label{program-gto2_fq_from_seq}

The \textbf{gto2\_fq\_from\_seq} converts a genomic sequence to pseudo
FASTQ file format.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_from_seq}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-9}


The \textbf{gto2\_fq\_from\_seq} program needs two streams for the
computation, namely the input and output standard. The input stream is a
sequence group file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_from_seq [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_from_seq [options]}

\ExtensionTok{It}\NormalTok{ converts a genomic sequence to pseudo FASTQ file format.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.seq}\NormalTok{           Input sequence file (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{        Output FASTQ file format (stdout)}

\ExtensionTok{Optional}\NormalTok{ options}
    \ExtensionTok{-n}\NormalTok{, --name=}\OperatorTok{<}\NormalTok{str}\OperatorTok{>}\NormalTok{      The read}\StringTok{'s header}
\StringTok{    -l, --lineSize=<int>  The maximum of chars for line}

\StringTok{Example: ./gto2_fq_from_seq -l <lineSize> -n <name> }
\StringTok{< input.seq > output.fastq}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{ACAAGACGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCCTGGAGGGTCCACCGCT}
\ExtensionTok{GCCCTGCTGCCATTGTCCCCGGCCCCACCTAAGGAAAAGCAGCCTCCTGACTTTCCTCGC}
\ExtensionTok{TTGGGCCGAGACAGCGAGCATATGCAGGAAGCGGCAGGAAGTGGTTTGAGTGGACCTCCG}
\ExtensionTok{GGCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCGGCAGGAAGCAGGCCAGTGCC}
\ExtensionTok{GCGAATCCGCGCGCCGGGACAGAATCTCCTGCAAAGCCCTGCAGGAACTTCTTCTGGAAG}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-9}


The output of the \textbf{gto2\_fq\_from\_seq} program is a pseudo FASTQ
file. An example, using the size line as 60 and the read's header as
`'SeqToFastq'', for the input, is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SeqToFastq1}
\ExtensionTok{ACAAGACGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCCTGGAGGGTCCACCGCT}
\ExtensionTok{+}
\ExtensionTok{FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF}
\ExtensionTok{@SeqToFastq2}
\ExtensionTok{GCCCTGCTGCCATTGTCCCCGGCCCCACCTAAGGAAAAGCAGCCTCCTGACTTTCCTCGC}
\ExtensionTok{+}
\ExtensionTok{FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF}
\ExtensionTok{@SeqToFastq3}
\ExtensionTok{TTGGGCCGAGACAGCGAGCATATGCAGGAAGCGGCAGGAAGTGGTTTGAGTGGACCTCCG}
\ExtensionTok{+}
\ExtensionTok{FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF}
\ExtensionTok{@SeqToFastq4}
\ExtensionTok{GGCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCGGCAGGAAGCAGGCCAGTGCC}
\ExtensionTok{+}
\ExtensionTok{FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF}
\ExtensionTok{@SeqToFastq5}
\ExtensionTok{GCGAATCCGCGCGCCGGGACAGAATCTCCTGCAAAGCCCTGCAGGAACTTCTTCTGGAAG}
\ExtensionTok{+}
\ExtensionTok{FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF}
\end{Highlighting}
\end{Shaded}

\section{Program gto2\_fq\_mutate}\label{program-gto2_fq_mutate}

The \textbf{gto2\_fq\_mutate} creates a synthetic mutation of a FASTQ
file given specific rates of mutations, deletions and additions. All
these paramenters are defined by the user, and their are optional.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_mutate}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-10}


The \textbf{gto2\_fq\_mutate} program needs two streams for the
computation, namely the input and output standard. However, optional
settings can be supplied too, such as the starting point to the random
generator, and the edition, deletion and insertion rates. Also, the user
can choose to use the ACGTN alphabet in the synthetic mutation. The
input stream is a FASTQ File.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_mutate [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_mutate [options]}

\ExtensionTok{Creates}\NormalTok{ a synthetic mutation of a FASTQ file given specific }
\ExtensionTok{rates}\NormalTok{ of mutations, deletions and additions.}

    \ExtensionTok{-h}\NormalTok{, --help      show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fasta}\NormalTok{   Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fasta}\NormalTok{  Output FASTQ file format (stdout)}

\ExtensionTok{Optional}
    \ExtensionTok{-s}\NormalTok{              Starting point to the random generator}
    \ExtensionTok{-m}\NormalTok{              Defines the mutation rate (default 0.0)}
    \ExtensionTok{-d}\NormalTok{              Defines the deletion rate (default 0.0)}
    \ExtensionTok{-i}\NormalTok{              Defines the insertion rate (default 0.0)}
    \ExtensionTok{-a}\NormalTok{              When active, the application uses the }
                    \ExtensionTok{ACGTN}\NormalTok{ alphabet}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_mutate -s }\OperatorTok{<}\NormalTok{seed}\OperatorTok{>}\NormalTok{ -m }\OperatorTok{<}\NormalTok{mutation rate}\OperatorTok{>} 
\ExtensionTok{-d} \OperatorTok{<}\NormalTok{deletion rate}\OperatorTok{>}\NormalTok{ -i }\OperatorTok{<}\NormalTok{insertion rate}\OperatorTok{>}\NormalTok{ -a }
\OperatorTok{<} \ExtensionTok{input.fastq} \OperatorTok{>}\NormalTok{ output.fastq}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-10}


The output of the \textbf{gto2\_fq\_mutate} program is a FASTQ file
whith the synthetic mutation of input file. Using the input above with
the seed value as 1 and the mutation rate as 0.5, an output example of
this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGACTTTGAGGTGTGGCGATAGACTGAAAACACTTCAGGGTAAAATCACTCGCAAAAGTG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGAGCCTTTACCGTAGGGGTGTAAGATTTTATACAAAAAGTCCAGGTCAAGAGGAA}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\section{Program gto2\_fq\_split}\label{program-gto2_fq_split}

The \textbf{gto2\_fq\_split} splits Paired End files according to the
direction of the strand (`/1' or `/2'). It writes by default singleton
reads as forward stands.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_split}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-11}


The \textbf{gto2\_fq\_split} program needs a stream for the computation,
namely the input standard. The input stream is a FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_split [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_split [options]}

\ExtensionTok{It}\NormalTok{ writes by default singleton reads as forward stands.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \ExtensionTok{-f}\NormalTok{, --forward=}\OperatorTok{<}\NormalTok{str}\OperatorTok{>}\NormalTok{   Output forward file}
    \ExtensionTok{-r}\NormalTok{, --reverse=}\OperatorTok{<}\NormalTok{str}\OperatorTok{>}\NormalTok{   Output reverse file}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output}\NormalTok{              Output read information (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_split -f }\OperatorTok{<}\NormalTok{output_forward.fastq}\OperatorTok{>} 
\ExtensionTok{-r} \OperatorTok{<}\NormalTok{output_reverse.fastq}\OperatorTok{>} \OperatorTok{<}\NormalTok{ input.fastq }\OperatorTok{>}\NormalTok{ output}

\ExtensionTok{Output}\NormalTok{ example :}
\ExtensionTok{Total}\NormalTok{ reads      : value}
\ExtensionTok{Singleton}\NormalTok{ reads  : value}
\ExtensionTok{Forward}\NormalTok{ reads    : value}
\ExtensionTok{Reverse}\NormalTok{ reads    : value}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60 1}
\ExtensionTok{GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60 2}
\ExtensionTok{NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-11}


The output of the \textbf{gto2\_fq\_split} program is a set of
information related to the file read. Using the input above, an output
example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Total}\NormalTok{ reads     : 2}
\ExtensionTok{Singleton}\NormalTok{ reads : 0}
\ExtensionTok{Forward}\NormalTok{ reads   : 65536}
\ExtensionTok{Reverse}\NormalTok{ reads   : 1}
\end{Highlighting}
\end{Shaded}

Also, this program generates two FASTQ files, with the reverse and
forward reads.

An example of the forward reads, for the input, is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60 1}
\ExtensionTok{GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\end{Highlighting}
\end{Shaded}

\section{Program gto2\_fq\_pack}\label{program-gto2_fq_pack}

The \textbf{gto2\_fq\_pack} packages each FASTQ read in a single line.
It can show the read score first or the dna sequence, depending on the
execution mode.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_pack}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-12}


The \textbf{gto2\_fq\_pack} program needs two streams for the
computation, namely the input and output standard. The input stream is a
FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_pack [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_pack [options]}

\ExtensionTok{It}\NormalTok{ packages each FASTQ read in a single line.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastqpack}\NormalTok{    Output packaged FASTQ file format }
                          \KeywordTok{(}\ExtensionTok{stdout}\KeywordTok{)}

\ExtensionTok{Optional}
    \ExtensionTok{-s}\NormalTok{, --scores          When active, the application show }
                          \ExtensionTok{the}\NormalTok{ scores first}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_pack -s }\OperatorTok{<}\NormalTok{ input.fastq }\OperatorTok{>}\NormalTok{ output.fastqpack}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-12}


The output of the \textbf{gto2\_fq\_pack} program is a packaged FASTQ
file. Using the input above, an output example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60+ 0}
\ExtensionTok{NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\ExtensionTok{SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60+ 1}
\end{Highlighting}
\end{Shaded}

Another example for the same input, but using the scores first (flag
`'s''), is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG}
\ExtensionTok{SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60+ 0}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\ExtensionTok{NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60+ 1}
\end{Highlighting}
\end{Shaded}

\section{Program gto2\_fq\_unpack}\label{program-gto2_fq_unpack}

The \textbf{gto2\_fq\_unpack} unpacks the FASTQ reads packaged using the
\textbf{gto2\_fq\_pack} tool.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_unpack}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-13}


The \textbf{gto2\_fq\_unpack} program needs two streams for the
computation, namely the input and output standard. The input stream is a
packaged FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_unpack [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_unpack [options]}

\ExtensionTok{It}\NormalTok{ unpacks the FASTQ reads packaged using the gto2_fq_pack }
\ExtensionTok{tool.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{        Output FASTQ file format (stdout)}

\ExtensionTok{Optional}
    \ExtensionTok{-s}\NormalTok{, --scores          When active, the application show }
                          \ExtensionTok{the}\NormalTok{ scores first}
    
\ExtensionTok{Example}\NormalTok{: ./gto2_fq_unpack -s }\OperatorTok{<}\NormalTok{ input.fastqpack }\OperatorTok{>}\NormalTok{ out.fastq}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60+ 0}
\ExtensionTok{NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\ExtensionTok{SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60+ 1}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-13}


The output of the \textbf{gto2\_fq\_unpack} program is a FASTQ file.
Using the input above, an output example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_quality\_score\_info}\label{program-gto2_fq_quality_score_info}

The \textbf{gto2\_fq\_quality\_score\_info} analyses the quality-scores
of a FASTQ file.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_quality_score_info}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-14}


The \textbf{gto2\_fq\_quality\_score\_info} program needs two streams
for the computation, namely the input and output standard. The input
stream is a FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_quality_score_info [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_quality_score_info [options]}

\ExtensionTok{It}\NormalTok{ analyses the quality-scores of a FASTQ file.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output}\NormalTok{              Output read information (stdout)}
    
\ExtensionTok{Optional}
    \ExtensionTok{-m}\NormalTok{, --max=}\OperatorTok{<}\NormalTok{int}\OperatorTok{>}\NormalTok{       The lenght of the maximum window}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_quality_score_info -m }\OperatorTok{<}\NormalTok{max}\OperatorTok{>} \OperatorTok{<}\NormalTok{ input.fastq}
\OperatorTok{>} \ExtensionTok{output}

\ExtensionTok{Output}\NormalTok{ example :}
\ExtensionTok{Total}\NormalTok{ reads     : value}
\ExtensionTok{Max}\NormalTok{ read length : value}
\ExtensionTok{Min}\NormalTok{ read length : value}
\ExtensionTok{Min}\NormalTok{ QS value    : value}
\ExtensionTok{Max}\NormalTok{ QS value    : value}
\ExtensionTok{QS}\NormalTok{ range        : value}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@111}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60 1}
\ExtensionTok{GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@222}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60 2}
\ExtensionTok{NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-14}


The output of the \textbf{gto2\_fq\_quality\_score\_info} program is a
set of information related to the file read. Using the input above with
the max window value as 30, an output example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Total}\NormalTok{ reads     : 2}
\ExtensionTok{Max}\NormalTok{ read length : 60}
\ExtensionTok{Min}\NormalTok{ read length : 60}
\ExtensionTok{Min}\NormalTok{ QS value    : 54}
\ExtensionTok{Max}\NormalTok{ QS value    : 73}
\ExtensionTok{QS}\NormalTok{ range        : 20}
 \ExtensionTok{1}\NormalTok{  ...  24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 }
\ExtensionTok{--+}\NormalTok{ ... +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+}
\ExtensionTok{73}\NormalTok{  ...  73 73 73 73 73 73 73 65 73 62 65 69 70 73 73 73 73 }
                               \ExtensionTok{*}\NormalTok{     *  *  *  *             }
                               \ExtensionTok{*}\NormalTok{     *  *  *  *             }
                               \ExtensionTok{*}\NormalTok{     *  *  *  *             }
                               \ExtensionTok{*}\NormalTok{     *  *  *                }
                               \ExtensionTok{*}\NormalTok{     *  *                   }
                               \ExtensionTok{*}\NormalTok{     *  *                   }
                               \ExtensionTok{*}\NormalTok{     *  *                   }
                               \ExtensionTok{*}\NormalTok{     *  *                   }
                                     \ExtensionTok{*}                      
                                     \ExtensionTok{*}                      
                                     \ExtensionTok{*}   
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_quality\_score\_min}\label{program-gto2_fq_quality_score_min}

The \textbf{gto2\_fq\_quality\_score\_min} analyses the minimal
quality-scores of a FASTQ file.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_quality_score_min}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-15}


The \textbf{gto2\_fq\_quality\_score\_min} program needs two streams for
the computation, namely the input and output standard. The input stream
is a FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_quality_score_min [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_quality_score_min [options]}

\ExtensionTok{It}\NormalTok{ analyses the minimal quality-scores of a FASTQ file.}

    \ExtensionTok{-h}\NormalTok{, --help        show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{     Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output}\NormalTok{          Output read information (stdout)}

\ExtensionTok{Optional}
    \ExtensionTok{-m}\NormalTok{, --max=}\OperatorTok{<}\NormalTok{int}\OperatorTok{>}\NormalTok{   The maximum window length (default 40)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_quality_score_min -m }\OperatorTok{<}\NormalTok{max}\OperatorTok{>} \OperatorTok{<}\NormalTok{ input.fastq }
\OperatorTok{>} \ExtensionTok{output}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@111}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60 1}
\ExtensionTok{GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@222}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60 2}
\ExtensionTok{NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-15}


The output of the \textbf{gto2\_fq\_quality\_score\_min} program is a
set of information related to the file read, considering the minimum
quality scores. Using the input above with the max window value as 20,
an output example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
 \ExtensionTok{1}\NormalTok{  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 }
\ExtensionTok{--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+}
\ExtensionTok{73}\NormalTok{ 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73}
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_quality\_score\_max}\label{program-gto2_fq_quality_score_max}

The \textbf{gto2\_fq\_quality\_score\_max} analyses the maximal
quality-scores of a FASTQ file.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_quality_score_max}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-16}


The \textbf{gto2\_fq\_quality\_score\_max} program needs two streams for
the computation, namely the input and output standard. The input stream
is a FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_quality_score_max [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_quality_score_max [options]}

\ExtensionTok{It}\NormalTok{ analyses the maximal quality-scores of a FASTQ file.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output}\NormalTok{              Output read information (stdout)}

\ExtensionTok{Optional}
    \ExtensionTok{-m}\NormalTok{, --max=}\OperatorTok{<}\NormalTok{int}\OperatorTok{>}\NormalTok{       The maximum window length (default 40)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_quality_score_max -m }\OperatorTok{<}\NormalTok{max}\OperatorTok{>} 
\OperatorTok{<} \ExtensionTok{input.fastq} \OperatorTok{>}\NormalTok{ output}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@111}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60 1}
\ExtensionTok{GNNTGATGGCCGCTGCCGATGGCGNANAATCCCACCAANATACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@222}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60 2}
\ExtensionTok{NTTCAGGGATACGACGNTTGTATTTTAAGAATCTGNAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-16}


The output of the \textbf{gto2\_fq\_quality\_score\_max} program is a
set of information related to the file read, considering the maximal
quality scores. Using the input above with the max window value as 20,
an output example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
 \ExtensionTok{1}\NormalTok{  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 }
\ExtensionTok{--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+}
\ExtensionTok{73}\NormalTok{ 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73 73}
\end{Highlighting}
\end{Shaded}

\section{Program gto2\_fq\_cut}\label{program-gto2_fq_cut}

The \textbf{gto2\_fq\_cut} cuts read sequences in a FASTQ file. It
requires that the initial and end positions for the cut.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_cut}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-17}


The \textbf{gto2\_fq\_cut} program needs two streams for the
computation, namely the input and output standard. The input stream is a
FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_cut [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_cut [options]}

\ExtensionTok{It}\NormalTok{ cuts read sequences in a FASTQ file.}

    \ExtensionTok{-h}\NormalTok{, --help            show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \ExtensionTok{-i}\NormalTok{, --initial=}\OperatorTok{<}\NormalTok{int}\OperatorTok{>}\NormalTok{   Starting position to the cut}
    \ExtensionTok{-e}\NormalTok{, --end=}\OperatorTok{<}\NormalTok{int}\OperatorTok{>}\NormalTok{       Ending position to the cut}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{        Output FASTQ file format (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_cut -i }\OperatorTok{<}\NormalTok{initial}\OperatorTok{>}\NormalTok{ -e }\OperatorTok{<}\NormalTok{end}\OperatorTok{>} \OperatorTok{<}\NormalTok{ input.fastq }
\OperatorTok{>} \ExtensionTok{output.fastq}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-17}


The output of the \textbf{gto2\_fq\_cut} program is a FASTQ file cut.
Using the initial value as 10 and the end value as 30, an example of
this input, is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{CGCTGCCGATGGCGTCAAATC}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIII9}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{ACGACGTTTGTATTTTAAGAA}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIII}
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_minimum\_local\_quality\_score\_forward}\label{program-gto2_fq_minimum_local_quality_score_forward}

The \textbf{gto2\_fq\_minimum\_local\_quality\_score\_forward} filters
the reads considering the quality score average of a defined window size
of bases.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_minimum_local_quality_score_forward}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-18}


The \textbf{gto2\_fq\_minimum\_local\_quality\_score\_forward} program
needs two streams for the computation, namely the input and output
standard. The input stream is a FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_minimum_local_quality_score_forward [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_minimum_local_quality_score_forward [options]}

\ExtensionTok{It}\NormalTok{ filters the reads considering the quality score average }
\ExtensionTok{of}\NormalTok{ a defined window size of bases.}

    \ExtensionTok{-h}\NormalTok{, --help        show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \ExtensionTok{-k}\NormalTok{                The window size of bases (default 5)}
    \ExtensionTok{-w}\NormalTok{                The minimum average of quality score }
                      \KeywordTok{(}\ExtensionTok{default}\NormalTok{ 25}\KeywordTok{)}
    \ExtensionTok{-m}\NormalTok{                The minimum value of the quality score }
                      \KeywordTok{(}\ExtensionTok{default}\NormalTok{ 33}\KeywordTok{)}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{     Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{    Output FASTQ file format (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_minimum_local_quality_score_forward }
\ExtensionTok{-k} \OperatorTok{<}\NormalTok{windowsize}\OperatorTok{>}\NormalTok{ -w }\OperatorTok{<}\NormalTok{minavg}\OperatorTok{>}\NormalTok{ -m }\OperatorTok{<}\NormalTok{minqs}\OperatorTok{>} 
\OperatorTok{<} \ExtensionTok{input.fastq} \OperatorTok{>}\NormalTok{ output.fastq}

\ExtensionTok{Console}\NormalTok{ output example:}
\ExtensionTok{Minimum}\NormalTok{ QS       : value}
\OperatorTok{<}\ExtensionTok{FASTQ}\NormalTok{ output}\OperatorTok{>}
\ExtensionTok{Total}\NormalTok{ reads      : value}
\ExtensionTok{Trimmed}\NormalTok{ reads    : value}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIII-I}\NormalTok{)}\ExtensionTok{8IIIIIIIIIIIIIIIIII}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-18}


The output of the
\textbf{gto2\_fq\_minimum\_local\_quality\_score\_forward} program is a
FASTQ file with the reads filtered following a quality score average of
a defined window of bases. The execution report only appears in the
console. Using the input above with the default values, an output
example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Minimum}\NormalTok{ QS     : 33}
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAA}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBII}
\ExtensionTok{Total}\NormalTok{ reads    : 2}
\ExtensionTok{Trimmed}\NormalTok{ reads  : 1}
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_minimum\_local\_quality\_score\_reverse}\label{program-gto2_fq_minimum_local_quality_score_reverse}

The \textbf{gto2\_fq\_minimum\_local\_quality\_score\_reverse} filters
the reverse reads, considering the quality score average of a defined
window size of bases.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_minimum_local_quality_score_reverse}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-19}


The \textbf{gto2\_fq\_minimum\_local\_quality\_score\_reverse} program
needs two streams for the computation, namely the input and output
standard. The input stream is a FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_minimum_local_quality_score_reverse [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_minimum_local_quality_score_reverse [options]}

\ExtensionTok{It}\NormalTok{ filters the reverse reads, considering the quality score }
\ExtensionTok{average}\NormalTok{ of a defined }
\ExtensionTok{window}\NormalTok{ size of bases.}

    \ExtensionTok{-h}\NormalTok{, --help        show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \ExtensionTok{-k}\NormalTok{                The window size of bases (default 5)}
    \ExtensionTok{-w}\NormalTok{                The minimum average of quality score }
                      \KeywordTok{(}\ExtensionTok{default}\NormalTok{ 25}\KeywordTok{)}
    \ExtensionTok{-m}\NormalTok{                The minimum value of the quality score}
                      \KeywordTok{(}\ExtensionTok{default}\NormalTok{ 33}\KeywordTok{)}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{     Input FASTQ file format (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{    Output FASTQ file format (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_minimum_local_quality_score_reverse }
\ExtensionTok{-k} \OperatorTok{<}\NormalTok{windowsize}\OperatorTok{>}\NormalTok{ -w }\OperatorTok{<}\NormalTok{minavg}\OperatorTok{>}\NormalTok{ -m }\OperatorTok{<}\NormalTok{minqs}\OperatorTok{>} 
\OperatorTok{<} \ExtensionTok{input.fastq} \OperatorTok{>}\NormalTok{ output.fastq}

\ExtensionTok{Console}\NormalTok{ output example:}
\ExtensionTok{Minimum}\NormalTok{ QS       : value}
\OperatorTok{<}\ExtensionTok{FASTQ}\NormalTok{ output}\OperatorTok{>}
\ExtensionTok{Total}\NormalTok{ reads      : value}
\ExtensionTok{Trimmed}\NormalTok{ reads    : value}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIII-I}\NormalTok{)}\ExtensionTok{8IIIIIIIIIIIIIIIIII}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-19}


The output of the
\textbf{gto2\_fq\_minimum\_local\_quality\_score\_reverse} program is a
FASTQ file with the reads filtered following a quality score average of
a defined window of bases. The execution report only appears in the
console. Using the input above with the default values, an output
example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Minimum}\NormalTok{ QS: 33}
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIII}
\ExtensionTok{Total}\NormalTok{ reads    : 2}
\ExtensionTok{Trimmed}\NormalTok{ reads  : 1}
\end{Highlighting}
\end{Shaded}

\section{Program gto2\_fq\_xs}\label{program-gto2_fq_xs}

The \textbf{gto2\_fq\_xs} is a skilled FASTQ read simulation tool,
flexible, portable (does not need a reference sequence) and tunable in
terms of sequence complexity. XS handles Ion Torrent, Roche-454,
Illumina and ABI-SOLiD simulation sequencing types. It has several
running modes, depending on the time and memory available, and is aimed
at testing computing infrastructures, namely cloud computing of
large-scale projects, and testing FASTQ compression algorithms.
Moreover, XS offers the possibility of simulating the three main FASTQ
components individually (headers, DNA sequences and quality-scores).
Quality-scores can be simulated using uniform and Gaussian
distributions.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_xs}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-20}


The \textbf{gto2\_fq\_xs} program needs a FASTQ file to compute.

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-20}


The output of the \textbf{gto2\_fq\_xs} program is a FASTQ file. Using
the input above using the common usage with 5 reads (-n 5), an output
example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]

\end{Highlighting}
\end{Shaded}

\section{Program gto2\_fq\_complement}\label{program-gto2_fq_complement}

The \textbf{gto2\_fq\_complement} replaces the ACGT bases with their
complements in a FASTQ file format.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_complement}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-21}


The \textbf{gto2\_fq\_complement} program needs two streams for the
computation, namely the input and output standard. The input stream is a
FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_complement [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_complement [options]}

\ExtensionTok{It}\NormalTok{ replaces the ACGT bases with their complements in a FASTQ}
\FunctionTok{file}\NormalTok{ format.}

    \ExtensionTok{-h}\NormalTok{, --help            Show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{        Output FASTQ file (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_complement }\OperatorTok{<}\NormalTok{ input.fastq }\OperatorTok{>}\NormalTok{ output.fastq}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-21}


The output of the \textbf{gto2\_fq\_complement} program is the FASTQ
file with the ACGT base complements. Using the input above, an output
example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{CCCACTACCGGCGACGGCTACCGCAGTTTAGGGTGGTTCAATGGGAATTGTTGAATTCCC}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{CAAGTCCCTATGCTGCAAACATAAAATTCTTAGACTTCGTCTTCAGCTACTATTATGCGC}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\section{Program gto2\_fq\_reverse}\label{program-gto2_fq_reverse}

The \textbf{gto2\_fq\_reverse} reverses the ACGT bases order for each
read in a FASTQ file format.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_reverse}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-22}


The \textbf{gto2\_fq\_reverse} program needs two streams for the
computation, namely the input and output standard. The input stream is a
FASTQ file.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_reverse [options] [[--] args]}
   \ExtensionTok{or}\NormalTok{: ./gto2_fq_reverse [options]}

\ExtensionTok{It}\NormalTok{ reverses the ACGT bases order for each read in a FASTQ }
\ExtensionTok{file.}

    \ExtensionTok{-h}\NormalTok{, --help            Show this help message and exit}

\ExtensionTok{Basic}\NormalTok{ options}
    \OperatorTok{<} \ExtensionTok{input.fastq}\NormalTok{         Input FASTQ file (stdin)}
    \OperatorTok{>} \ExtensionTok{output.fastq}\NormalTok{        Output FASTQ file (stdout)}

\ExtensionTok{Example}\NormalTok{: ./gto2_fq_reverse }\OperatorTok{<}\NormalTok{ input.fastq }\OperatorTok{>}\NormalTok{ output.fastq}
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-22}


The output of the \textbf{gto2\_fq\_reverse} program is the FASTQ file
complement with the flag `'(Reversed)'' added in the header. Using the
input above, an output example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 (Reversed)}
\ExtensionTok{GGGAATTCAACAATTCCCATTGAACCACCCTAAACTGCGGTAGCCGTCGCCGGTAGTGGG}
\ExtensionTok{+}
\ExtensionTok{IIIDIIIIIIIIIIIIIIIIIIIICI9GI9IIIIIIIIIIIIIIIIIIIIIIIIIIIIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 (Reversed)}
\ExtensionTok{GCGCATAATAGTAGCTGAAGACGAAGTCTAAGAATTTTATGTTTGCAGCATAGGGACTTG}
\ExtensionTok{+}
\ExtensionTok{IGIIIIIIIIIIIIIIIIIIIIIIIBI6IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII}
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_variation\_map}\label{program-gto2_fq_variation_map}

The \textbf{gto2\_fq\_variation\_map} identifies the variation that
occours in the sequences relative to the reads or a set of reads.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_variation_map}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-23}


The \textbf{gto2\_fq\_variation\_map} program needs FASTQ, FASTA or SEQ
files to be used as reference and target files.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_variation_map }\OperatorTok{<}\NormalTok{OPTIONS}\OperatorTok{>}\NormalTok{... [FILE]:}\OperatorTok{<}\NormalTok{...}\OperatorTok{>} 
\NormalTok{[}\ExtensionTok{FILE}\NormalTok{]:}\OperatorTok{<}\NormalTok{...}\OperatorTok{>}

\ExtensionTok{The}\NormalTok{ gto2_fq_variation_map is a tool to map relative }
\ExtensionTok{singularity}\NormalTok{ regions. The (probabilistic) }\ExtensionTok{Bloom}\NormalTok{ filter is }
\ExtensionTok{automatically}\NormalTok{ set.   }
                                                         
  \ExtensionTok{-v}\NormalTok{                       verbose mode,                 }
  \ExtensionTok{-a}\NormalTok{                       about CHESTER,                }
  \ExtensionTok{-s} \OperatorTok{<}\NormalTok{value}\OperatorTok{>}\NormalTok{               bloom size,                   }
  \ExtensionTok{-i}\NormalTok{                       use inversions,               }
  \ExtensionTok{-p}\NormalTok{                       show positions/words,         }
  \ExtensionTok{-k} \OperatorTok{<}\NormalTok{value}\OperatorTok{>}\NormalTok{               k-mer size (up to 30),        }
                                                         
\NormalTok{  [}\ExtensionTok{rFile1}\NormalTok{]:}\OperatorTok{<}\NormalTok{rFile2}\OperatorTok{>}\NormalTok{:}\OperatorTok{<}\NormalTok{...}\OperatorTok{>}\NormalTok{  reference file(s),            }
\NormalTok{  [}\ExtensionTok{tFile1}\NormalTok{]:}\OperatorTok{<}\NormalTok{tFile2}\OperatorTok{>}\NormalTok{:}\OperatorTok{<}\NormalTok{...}\OperatorTok{>}\NormalTok{  target file(s)}\BuiltInTok{.}               
                                                         
\ExtensionTok{The}\NormalTok{ reference files may be FASTA, FASTQ or DNA SEQ,      }
\KeywordTok{while} \ExtensionTok{the}\NormalTok{ target files may be FASTA OR DNA SEQ.          }
\ExtensionTok{Report}\NormalTok{ bugs to }\OperatorTok{<}\DataTypeTok{\{pratas,raquelsilva,ap,pjf\}}\NormalTok{@ua.pt}\OperatorTok{>}\NormalTok{.}
\end{Highlighting}
\end{Shaded}

An example of a reference file (Multi-FASTA format) is:

\begin{Shaded}
\begin{Highlighting}[]
\OperatorTok{>}\ExtensionTok{AB000264} \KeywordTok{|}\VariableTok{acc=}\NormalTok{AB000264}\KeywordTok{|}\VariableTok{descr=}\NormalTok{Homo }\ExtensionTok{sapiens}\NormalTok{ mRNA }
\ExtensionTok{ACAAGACGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCCTGGAGGGTCCACCGCT}
\ExtensionTok{CCCTGCTGCCATTGTCCCCGGCCCCACCTAAGGAAAAGCAGCCTCCTGACTTTCCTCGCT}
\ExtensionTok{TGGGCCGAGACAGCGAGCATATGCAGGAAGCGGCAGGAAGTGGTTTGAGTGGACCTCCGG}
\ExtensionTok{GCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCGGCAGGAAGCAGGCCAGTGCCG}
\ExtensionTok{CGAATCCGCGCGCCGGGACAGAATCTCCTGCAAAGCCCTGCAGGAACTTCTTCTGGAAGA}
\ExtensionTok{CCTTCTCCACCCCCCCAGCTAAAACCTCACCCATGAATGCTCACGCAAGTTTAATTACAG}
\ExtensionTok{ACCTGAA}
\OperatorTok{>}\ExtensionTok{AB000263} \KeywordTok{|}\VariableTok{acc=}\NormalTok{AB000263}\KeywordTok{|}\VariableTok{descr=}\NormalTok{Homo }\ExtensionTok{sapiens}\NormalTok{ mRNA }
\ExtensionTok{ACAAGATGCCATTGTCCCCCGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCACCG}
\ExtensionTok{CTGCCCTGCCCCTGGAGGGTGGCCCCACCGGCCGAGACAGCGAGCATATGCAGGAAGCGG}
\ExtensionTok{CAGGAATAAGGAAAAGCAGCCTCCTGACTTTCCTCGCTTGGTGGTTTGAGTGGACCTCCC}
\ExtensionTok{AGGCCAGTGCCGGGCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCGGCAGGAAG}
\ExtensionTok{GCGCACCCCCCCAGCAATCCGCGCGCCGGGACAGAATGCCCTGCAGGAACTTCTTCTGGA}
\ExtensionTok{AGACCTTCTCCTCCTGCAAATAAAACCTCACCCATGAATGCTCACGCAAGTTTAATTACA}
\ExtensionTok{GACCTGAA}
\end{Highlighting}
\end{Shaded}

An example for the target file (FASTQ format) is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-23}


The output of the \textbf{gto2\_fq\_variation\_map} program is a text
file identifying the relative regions. Using the inputs above, an output
example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{111111111111111111111111111110000000000000000000000000000000}
\ExtensionTok{111111111111111111111111111110000000000000000000000000000000}
\ExtensionTok{000000000000}
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_variation\_filter}\label{program-gto2_fq_variation_filter}

The \textbf{gto2\_fq\_variation\_filter} filters and segments the
regions of singularity from the output of
\textbf{gto2\_fq\_variation\_map}.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_variation_filter}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-24}


The \textbf{gto2\_fq\_variation\_filter} program needs a the output of
\textbf{gto2\_fq\_variation\_map} to compute.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_variation_filter }\OperatorTok{<}\NormalTok{OPTIONS}\OperatorTok{>}\NormalTok{... [FILE]:}\OperatorTok{<}\NormalTok{...}\OperatorTok{>}      
\ExtensionTok{The}\NormalTok{ gto2_fq_variation_filter is a tool to filter maps.  }
                                                     
  \ExtensionTok{-v}\NormalTok{                       verbose mode,             }
  \ExtensionTok{-a}\NormalTok{                       about CHESTER,            }
  \ExtensionTok{-t} \OperatorTok{<}\NormalTok{value}\OperatorTok{>}\NormalTok{               threshold [0.0}\KeywordTok{;}\ExtensionTok{1.0}\NormalTok{],      }
  \ExtensionTok{-w} \OperatorTok{<}\NormalTok{value}\OperatorTok{>}\NormalTok{               window size,              }
  \ExtensionTok{-u} \OperatorTok{<}\NormalTok{value}\OperatorTok{>}\NormalTok{               sub-sampling,             }
                                                     
\NormalTok{  [}\ExtensionTok{tFile1}\NormalTok{]:}\OperatorTok{<}\NormalTok{tFile2}\OperatorTok{>}\NormalTok{:}\OperatorTok{<}\NormalTok{...}\OperatorTok{>}\NormalTok{  target file(s)}\BuiltInTok{.}           
                                                     
\ExtensionTok{The}\NormalTok{ target files may be generated by gto2_fq_variation_map.    }
\ExtensionTok{Report}\NormalTok{ bugs to }\OperatorTok{<}\DataTypeTok{\{pratas,raquelsilva,ap,pjf\}}\NormalTok{@ua.pt}\OperatorTok{>}\NormalTok{. }
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{111111111111111111111111111110000000000000000000000000000000}
\ExtensionTok{111111111111111111111111111110000000000000000000000000000000}
\ExtensionTok{000000000000}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-24}


The output of the \textbf{gto2\_fq\_variation\_filter} program is a text
file with the coordenates of the segmented regions. Using the inputs
above, an output example of this is the following:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#132#132}
\ExtensionTok{30}\NormalTok{:60}
\ExtensionTok{90}\NormalTok{:130}
\end{Highlighting}
\end{Shaded}

\section{Program
gto2\_fq\_variation\_visual}\label{program-gto2_fq_variation_visual}

The \textbf{gto2\_fq\_variation\_visual} depites the regions of
singularity using the output from \textbf{gto2\_fq\_variation\_filter}
into an SVG image.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_variation_visual}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-25}


The \textbf{gto2\_fq\_variation\_visual} program needs a the output of
\textbf{gto2\_fq\_variation\_filter} to compute.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Usage}\NormalTok{: ./gto2_fq_variation_visual }\OperatorTok{<}\NormalTok{OPTIONS}\OperatorTok{>}\NormalTok{... [FILE]:}\OperatorTok{<}\NormalTok{...}\OperatorTok{>}
\ExtensionTok{The}\NormalTok{ gto2_fq_variation_visual is a tool to visualize relative}
\ExtensionTok{singularity}\NormalTok{ regions.}
                                                     
  \ExtensionTok{-v}\NormalTok{                       verbose mode,             }
  \ExtensionTok{-a}\NormalTok{                       about CHESTER,            }
  \ExtensionTok{-e} \OperatorTok{<}\NormalTok{value}\OperatorTok{>}\NormalTok{               enlarge painted regions,  }
                                                     
\NormalTok{  [}\ExtensionTok{tFile1}\NormalTok{]:}\OperatorTok{<}\NormalTok{tFile2}\OperatorTok{>}\NormalTok{:}\OperatorTok{<}\NormalTok{...}\OperatorTok{>}\NormalTok{  target file(s)}\BuiltInTok{.}           
                                                     
\ExtensionTok{Report}\NormalTok{ bugs to }\OperatorTok{<}\DataTypeTok{\{pratas,raquelsilva,ap,pjf\}}\NormalTok{@ua.pt}\OperatorTok{>}\NormalTok{. }
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#132#132}
\ExtensionTok{30}\NormalTok{:60}
\ExtensionTok{90}\NormalTok{:130}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-25}


The output of the \textbf{gto2\_fq\_variation\_visual} program is a SVG
plot with the maps. In the following Figure, is represented the plot
using the input above.

\begin{figure}
\includegraphics[width=1\linewidth]{images/gto2FqVariationVisual} \caption{Execution plot of the variation visual tool using the previous input.}\label{fig:filters}
\end{figure}

\section{Program
gto2\_fq\_metagenomics}\label{program-gto2_fq_metagenomics}

The \textbf{gto2\_fq\_metagenomics} is an ultra-fast method to infer
metagenomic composition of sequenced reads relative to a database.
gto2\_fq\_metagenomics measures similarity between any FASTQ file (or
FASTA), independently from the size, against any multi-FASTA database,
such as the entire set of complete genomes from the NCBI.
gto2\_fq\_metagenomics supports single reads, paired-end reads, and
compositions of both. It has been tested in many plataforms, such as
Illumina MySeq, HiSeq, Novaseq, IonTorrent.

gto2\_fq\_metagenomics is efficient to detect the presence and
authenticate a given species in the FASTQ reads. The core of the method
is based on relative data compression. gto2\_fq\_metagenomics uses
variable multi-threading, without multiplying the memory for each
thread, being able to run efficiently in a common laptop.

For help type:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./gto2_fq_metagenomics}\NormalTok{ -h}
\end{Highlighting}
\end{Shaded}

In the following subsections, we explain the input and output
parameters.

\subsection*{Input parameters}\label{input-parameters-26}


The \textbf{gto2\_fq\_metagenomics} program needs a FASTQ file to
compute.

The attribution is given according to:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{NAME}                                                                     
      \ExtensionTok{The}\NormalTok{ gto2_fq_metagenomics is a tool to infer }
      \ExtensionTok{metagenomic}\NormalTok{ composition.            }
                                                                         
\ExtensionTok{SYNOPSIS}                                                                 
      \ExtensionTok{gto2_fq_metagenomics}\NormalTok{ [OPTION]... [FILE1]:[FILE2]:... }
\NormalTok{      [}\ExtensionTok{FILE}\NormalTok{]                      }
                                                                         
\ExtensionTok{SAMPLE}                                                                   
      \ExtensionTok{gto2_fq_metagenomics}\NormalTok{ -v -F -l 47 -Z -y pro.com }
      \ExtensionTok{reads1.fq}\NormalTok{:reads2.fq DB.fa         }
                                                                         
\ExtensionTok{DESCRIPTION}                                                              
      \ExtensionTok{It}\NormalTok{ infers metagenomic sample composition of sequenced }
      \ExtensionTok{reads.}\NormalTok{ The core of the method uses a cooperation }
      \ExtensionTok{between}\NormalTok{ multiple context and tolerant context models }
      \ExtensionTok{with}\NormalTok{ several depths. The reference sequences must be }
      \KeywordTok{in} \ExtensionTok{a}\NormalTok{ multi-FASTA format. The sequenced reads must be }
      \ExtensionTok{trimmed}\NormalTok{ and in FASTQ format.           }
                                                                         
      \ExtensionTok{Non-mandatory}\NormalTok{ arguments:                                           }
                                                                         
      \ExtensionTok{-h}\NormalTok{                   give this help,                               }
      \ExtensionTok{-F}\NormalTok{                   force mode (overwrites top file),             }
      \ExtensionTok{-V}\NormalTok{                   display version number,                       }
      \ExtensionTok{-v}\NormalTok{                   verbose mode (more information),              }
      \ExtensionTok{-Z}\NormalTok{                   database local similarity,                    }
      \ExtensionTok{-s}\NormalTok{                   show compression levels,                      }
                                                                         
      \ExtensionTok{-l} \OperatorTok{<}\NormalTok{level}\OperatorTok{>}\NormalTok{           compression level [1}\KeywordTok{;}\ExtensionTok{47}\NormalTok{],                    }
      \ExtensionTok{-p} \OperatorTok{<}\NormalTok{sample}\OperatorTok{>}\NormalTok{          subsampling (default: 1),                    }
      \ExtensionTok{-t} \OperatorTok{<}\NormalTok{top}\OperatorTok{>}\NormalTok{             top of similarity (default: 20),              }
      \ExtensionTok{-n} \OperatorTok{<}\NormalTok{nThreads}\OperatorTok{>}\NormalTok{        number of threads (default: 2),              }
                                                                         
      \ExtensionTok{-x} \OperatorTok{<}\NormalTok{FILE}\OperatorTok{>}\NormalTok{            similarity top filename,                      }
      \ExtensionTok{-y} \OperatorTok{<}\NormalTok{FILE}\OperatorTok{>}\NormalTok{            profile filename (-Z must be on)}\BuiltInTok{.}             
                                                                         
      \ExtensionTok{Mandatory}\NormalTok{ arguments:                                               }
                                                                         
\NormalTok{      [}\ExtensionTok{FILE1}\NormalTok{]:[FILE2]:...  metagenomic filename (FASTQ),                 }
                           \ExtensionTok{Use} \StringTok{":"}\NormalTok{ for splitting files.                }
                                                                         
\NormalTok{      [}\ExtensionTok{FILE}\NormalTok{]               database filename (Multi-FASTA)}\BuiltInTok{.}                
\end{Highlighting}
\end{Shaded}

An example of such an input file is:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{@SRR001666.1}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:817:345 length=60}
\ExtensionTok{GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACCAAGTTACCCTTAACAACTTAAGGG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9ICIIIIIIIIIIIIIIIIIIIIDIII}
\ExtensionTok{@SRR001666.2}\NormalTok{ 071112_SLXA-EAS1_s_7:5:1:801:338 length=60}
\ExtensionTok{GTTCAGGGATACGACGTTTGTATTTTAAGAATCTGAAGCAGAAGTCGATGATAATACGCG}
\ExtensionTok{+}
\ExtensionTok{IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII6IBIIIIIIIIIIIIIIIIIIIIIIIGI}
\end{Highlighting}
\end{Shaded}

\subsection*{Output}\label{output-26}


The output of the \textbf{gto2\_fq\_metagenomics} program is a CSV file
(top.csv) with the highest probability of being contained in the
samples. An example for this CSV file is the following:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{2}\NormalTok{  66725   12.263   NC_037703.1_Saccharomycodes_ludwigii...}
\ExtensionTok{1}\NormalTok{  66725   12.263   NC_037703.1_Saccharomycodes_ludwigii...}
\ExtensionTok{3}\NormalTok{  107123  11.492   NC_012621.1_Nakaseomyces_bacillispor...}
\ExtensionTok{4}\NormalTok{  107123  11.492   NC_012621.1_Nakaseomyces_bacillispor...}
\ExtensionTok{5}\NormalTok{  16592   11.153   NC_024030.1_Equus_przewalskii_mitoch...}
\ExtensionTok{6}\NormalTok{  14583   10.851   NC_021120.1_Bursaphelenchus_mucronat...}
\ExtensionTok{7}\NormalTok{  162504  10.607   NC_018415.1_Candidatus_Carsonella_ru...}
\ExtensionTok{8}\NormalTok{  10315   10.586   NC_016117.1_Mnemiopsis_leidyi_mitoch...}
\ExtensionTok{9}\NormalTok{  162589  10.550   NC_018414.1_Candidatus_Carsonella_ru...}
\ExtensionTok{10}\NormalTok{ 166163  10.476   NC_018416.1_Candidatus_Carsonella_ru...}
\end{Highlighting}
\end{Shaded}

\chapter{Amino Acid Tools}\label{amino-acid-tools}

\section{gto2\_aa}\label{gto2_aa}

to do

\chapter{Genomic Tools}\label{genomic-tools}

\section{gto2\_dna}\label{gto2_dna}

to do

\chapter{General Purpose Tools}\label{general-purpose-tools}

\section{gto2\_}\label{gto2_}

to do

\bibliography{refs.bib}

\printindex

\end{document}
