<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Introduction | GTO 2: The genomics-proteomics toolkit</title>
  <meta name="description" content="To do" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Introduction | GTO 2: The genomics-proteomics toolkit" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="To do" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Introduction | GTO 2: The genomics-proteomics toolkit" />
  
  <meta name="twitter:description" content="To do" />
  

<meta name="author" content="J. R. Almeida, A. J. Pinho, J. L. Oliveira, D. Pratas" />


<meta name="date" content="2021-04-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="fasta-tools.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="/">GTO 2: The genomics-proteomics toolkit</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#testing"><i class="fa fa-check"></i><b>1.2</b> Testing</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#execution-control"><i class="fa fa-check"></i><b>1.3</b> Execution control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="fasta-tools.html"><a href="fasta-tools.html"><i class="fa fa-check"></i><b>2</b> FASTA Tools</a><ul>
<li class="chapter" data-level="2.1" data-path="fasta-tools.html"><a href="fasta-tools.html#gto2_fa_to_fq"><i class="fa fa-check"></i><b>2.1</b> gto2_fa_to_fq</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="fastq-tools.html"><a href="fastq-tools.html"><i class="fa fa-check"></i><b>3</b> FASTQ Tools</a><ul>
<li class="chapter" data-level="3.1" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_to_fa"><i class="fa fa-check"></i><b>3.1</b> gto2_fq_to_fa</a><ul>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#input-parameters"><i class="fa fa-check"></i>Input parameters</a></li>
<li class="chapter" data-level="" data-path="fastq-tools.html"><a href="fastq-tools.html#output"><i class="fa fa-check"></i>Output</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_to_mfa"><i class="fa fa-check"></i><b>3.2</b> gto2_fq_to_mfa</a></li>
<li class="chapter" data-level="3.3" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_exclude_n"><i class="fa fa-check"></i><b>3.3</b> gto2_fq_exclude_n</a></li>
<li class="chapter" data-level="3.4" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_extract_quality_scores"><i class="fa fa-check"></i><b>3.4</b> gto2_fq_extract_quality_scores</a></li>
<li class="chapter" data-level="3.5" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_info"><i class="fa fa-check"></i><b>3.5</b> gto2_fq_info</a></li>
<li class="chapter" data-level="3.6" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_maximum_read_size"><i class="fa fa-check"></i><b>3.6</b> gto2_fq_maximum_read_size</a></li>
<li class="chapter" data-level="3.7" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_minimum_quality_score"><i class="fa fa-check"></i><b>3.7</b> gto2_fq_minimum_quality_score</a></li>
<li class="chapter" data-level="3.8" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_minimum_read_size"><i class="fa fa-check"></i><b>3.8</b> gto2_fq_minimum_read_size</a></li>
<li class="chapter" data-level="3.9" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_rand_extra_chars"><i class="fa fa-check"></i><b>3.9</b> gto2_fq_rand_extra_chars</a></li>
<li class="chapter" data-level="3.10" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_from_seq"><i class="fa fa-check"></i><b>3.10</b> gto2_fq_from_seq</a></li>
<li class="chapter" data-level="3.11" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_mutate"><i class="fa fa-check"></i><b>3.11</b> gto2_fq_mutate</a></li>
<li class="chapter" data-level="3.12" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_split"><i class="fa fa-check"></i><b>3.12</b> gto2_fq_split</a></li>
<li class="chapter" data-level="3.13" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_pack"><i class="fa fa-check"></i><b>3.13</b> gto2_fq_pack</a></li>
<li class="chapter" data-level="3.14" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_unpack"><i class="fa fa-check"></i><b>3.14</b> gto2_fq_unpack</a></li>
<li class="chapter" data-level="3.15" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_quality_score_info"><i class="fa fa-check"></i><b>3.15</b> gto2_fq_quality_score_info</a></li>
<li class="chapter" data-level="3.16" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_quality_score_min"><i class="fa fa-check"></i><b>3.16</b> gto2_fq_quality_score_min</a></li>
<li class="chapter" data-level="3.17" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_quality_score_max"><i class="fa fa-check"></i><b>3.17</b> gto2_fq_quality_score_max</a></li>
<li class="chapter" data-level="3.18" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_cut"><i class="fa fa-check"></i><b>3.18</b> gto2_fq_cut</a></li>
<li class="chapter" data-level="3.19" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_minimum_local_quality_score_forward"><i class="fa fa-check"></i><b>3.19</b> gto2_fq_minimum_local_quality_score_forward</a></li>
<li class="chapter" data-level="3.20" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_minimum_local_quality_score_reverse"><i class="fa fa-check"></i><b>3.20</b> gto2_fq_minimum_local_quality_score_reverse</a></li>
<li class="chapter" data-level="3.21" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_xs"><i class="fa fa-check"></i><b>3.21</b> gto2_fq_xs</a></li>
<li class="chapter" data-level="3.22" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_complement"><i class="fa fa-check"></i><b>3.22</b> gto2_fq_complement</a></li>
<li class="chapter" data-level="3.23" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_reverse"><i class="fa fa-check"></i><b>3.23</b> gto2_fq_reverse</a></li>
<li class="chapter" data-level="3.24" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_variation_map"><i class="fa fa-check"></i><b>3.24</b> gto2_fq_variation_map</a></li>
<li class="chapter" data-level="3.25" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_variation_filter"><i class="fa fa-check"></i><b>3.25</b> gto2_fq_variation_filter</a></li>
<li class="chapter" data-level="3.26" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_variation_visual"><i class="fa fa-check"></i><b>3.26</b> gto2_fq_variation_visual</a></li>
<li class="chapter" data-level="3.27" data-path="fastq-tools.html"><a href="fastq-tools.html#gto2_fq_metagenomics"><i class="fa fa-check"></i><b>3.27</b> gto2_fq_metagenomics</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html"><i class="fa fa-check"></i><b>4</b> Amino Acid Tools</a><ul>
<li class="chapter" data-level="4.1" data-path="amino-acid-tools.html"><a href="amino-acid-tools.html#gto2_aa"><i class="fa fa-check"></i><b>4.1</b> gto2_aa</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genomic-tools.html"><a href="genomic-tools.html"><i class="fa fa-check"></i><b>5</b> Genomic Tools</a><ul>
<li class="chapter" data-level="5.1" data-path="genomic-tools.html"><a href="genomic-tools.html#gto2_dna"><i class="fa fa-check"></i><b>5.1</b> gto2_dna</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="general-purpose-tools.html"><a href="general-purpose-tools.html"><i class="fa fa-check"></i><b>6</b> General Purpose Tools</a><ul>
<li class="chapter" data-level="6.1" data-path="general-purpose-tools.html"><a href="general-purpose-tools.html#gto2_"><i class="fa fa-check"></i><b>6.1</b> gto2_</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GTO 2: The genomics-proteomics toolkit</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introduction" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Introduction</h1>
<p>Recent advances in DNA sequencing, specifically in next-generation sequencing (NGS), revolutionised the field of genomics, making possible the generation of large amounts of sequencing data very rapidly and at substantially low cost<span class="citation">(Mardis <a href="#ref-mardis2017dna">2017</a>)</span>. This new technology also brought with it several challenges, namely in what concerns the analysis, storage, and transmission of the generated sequences<span class="citation">(Brouwer et al. <a href="#ref-brouwer2016current">2016</a>,<span class="citation">L. Liu et al. (<a href="#ref-liu2012comparison">2012</a>)</span>)</span>. As a consequence, several specialised tools were developed throughout the years in order to deal with these challenges.</p>
<p>Firstly, the storage of the raw data generated by NGS experiments is possible by using several file formats, the FASTQ and FASTA are the most commonly used<span class="citation">(Zhang <a href="#ref-zhang2016overview">2016</a>)</span>. FASTQ is an extension of the FASTA format, that besides the nucleotide sequence, also stores associated per base quality score and it is considered the standard format for sequencing data storage and exchange<span class="citation">(Cock et al. <a href="#ref-cock2009sanger">2009</a>)</span>.</p>
<p>Regarding the analysis and manipulation of these sequencing data files many software applications emerged, including <strong>fqtools</strong><span class="citation">(Droop <a href="#ref-droop2016fqtools">2016</a>)</span>, <strong>FASTX-Toolkit</strong><span class="citation">(Gordon, Hannon, and others <a href="#ref-gordon2010fastx">2010</a>)</span>, <strong>GALAXY</strong><span class="citation">(Afgan et al. <a href="#ref-afgan2018galaxy">2018</a>)</span>, <strong>GATK</strong><span class="citation">(DePristo et al. <a href="#ref-depristo2011framework">2011</a>)</span>, <strong>MEGA</strong><span class="citation">(Kumar, Stecher, and Tamura <a href="#ref-kumar2016mega7">2016</a>)</span>, <strong>SeqKit</strong><span class="citation">(Shen et al. <a href="#ref-shen2016seqkit">2016</a>)</span>, among others. <strong>Fqtools</strong> is a suite of tools to view, manipulate and summarise FASTQ data. This software also identifies invalid FASTQ files<span class="citation">(Droop <a href="#ref-droop2016fqtools">2016</a>)</span>. <strong>GALAXY</strong>, in its turn, is an open, web-based scientific platform for analysing genomic data<span class="citation">(Goecks, Nekrutenko, and Taylor <a href="#ref-goecks2010galaxy">2010</a>)</span>. This platform integrates several specialised sets of tools, e.g. for manipulating FASTQ files<span class="citation">(Blankenberg et al. <a href="#ref-blankenberg2010manipulation">2010</a>)</span>. <strong>FASTX-Toolkit</strong> is a collection of command-line tools to process FASTA and FASTQ files. This toolkit is available in two forms: as a command-line, or integrated into the web-based platform <strong>GALAXY</strong><span class="citation">(Gordon, Hannon, and others <a href="#ref-gordon2010fastx">2010</a>)</span>. <strong>SeqKit</strong> is another toolkit used to process FASTA and FASTQ files and is available for all major operating systems<span class="citation">(Shen et al. <a href="#ref-shen2016seqkit">2016</a>)</span>. The Genome Analysis Toolkit (<strong>GATK</strong>) was designed as a structured programming framework to simplify the development of analysis tools. However, nowadays, it is a suite of tools focused on variant discovering and genotyping<span class="citation">(Van der Auwera et al. <a href="#ref-van2013fastq">2013</a>)</span>. More towards the evolutionary perspectives, Molecular Evolutionary Genetics Analysis (<strong>MEGA</strong>) software provides tools to analyse DNA and protein sequences statistically<span class="citation">(Tamura et al. <a href="#ref-tamura2011mega5">2011</a>)</span>. Several of these frameworks lack on variety, namely the ability to perform multiple tasks using only one toolkit.</p>
<p>Compression is another important aspect when dealing with high-throughput sequencing data, as it reduces storage space and accelerates data transmission. A survey on DNA compressors and amino acid sequence compression can be found in<span class="citation">(Hosseini, Pratas, and Pinho <a href="#ref-hosseini2016survey">2016</a>)</span>. Currently, the DNA sequence compressors HiRGC<span class="citation">(Y. Liu et al. <a href="#ref-liu2017high">2017</a>)</span>, iDoComp<span class="citation">(Ochoa, Hernaez, and Weissman <a href="#ref-ochoa2014idocomp">2014</a>)</span>, GeCo<span class="citation">(Pratas, Pinho, and Ferreira <a href="#ref-pratas2016efficient">2016</a>)</span>, and GDC<span class="citation">(Deorowicz, Danek, and Niemiec <a href="#ref-deorowicz2015gdc">2015</a>)</span> are considered to have the best performance<span class="citation">(Hernaez et al. <a href="#ref-hernaez2019genomic">2019</a>)</span>. Of these four approaches, GeCo is the only one that can be used for reference-free and reference-based compression. Furthermore, GeCo can be used as an analysis tool to determine absolute measures for many distance computations and local measures<span class="citation">(Pratas, Pinho, and Ferreira <a href="#ref-pratas2016efficient">2016</a>)</span>.</p>
<p>Amino acid sequences are known to be very hard to compress<span class="citation">(Nalbantoglu, Russell, and Sayood <a href="#ref-nalbantoglu2010data">2010</a>)</span>, however, Hosseini et al.<span class="citation">(Hosseini, Pratas, and Pinho <a href="#ref-hosseini2019ac">2019</a>)</span> recently developed AC, a state-of-the-art for lossless amino acid sequence compression. In<span class="citation">(Pratas, Hosseini, and Pinho <a href="#ref-pratas2018compression">2018</a>)</span> the authors compared the performance of AC, in terms of bit-rate, to several general-purpose lossless compressors and several protein compressors, using different proteomes. They concluded that in average AC provides the best bit-rates.</p>
<p>Another relevant subject is genomic data simulation. Read simulations tools are fundamental for the development, testing and evaluation of methods and computational tools<span class="citation">(Huang et al. <a href="#ref-huang2011art">2011</a>,price2017simulome)</span>. Despite the availability of a large number of real sequence reads, read simulation data is necessary due to the inability to know the ground truth of real data<span class="citation">(Baruzzo et al. <a href="#ref-baruzzo2017simulation">2017</a>)</span>. Escalona <span class="citation">(Escalona, Rocha, and Posada <a href="#ref-escalona2016comparison">2016</a>)</span>, recently, reviewed 23 NGS simulation tools. XS<span class="citation">(Pratas, Pinho, and Rodrigues <a href="#ref-pratas2014xs">2014</a>)</span>, a FASTQ read simulation tool, stands out in relation to the other 22 simulation tools because it is the only one that does not need a reference sequence. Furthermore, XS is the only open-source tool for simulation of FASTQ reads produced by the four most used sequencing machines, Roche-454, Illumina, ABI SOLiD and Ion Torrent.</p>
<p>Although a large number of tools are available for analysing, compressing, and simulation, these tools are specialised in only a specific task. Besides, in many cases the output of one tool cannot be used directly as input for another tool, e.g. the output of a simulation tool cannot always be used directly as input for an analysis tool. Thus, unique software that includes several specialised tools is necessary.</p>
<p>In this document, we describe <strong>GTO2</strong>, a complete toolkit for genomics and proteomics, namely for FASTQ, FASTA and SEQ formats, with many complementary tools. The toolkit is for Unix-based systems, built for ultra-fast computations. <strong>GTO2</strong> supports pipes for easy integration with the sub-programs belonging to <strong>GTO2</strong> as well as external tools. <strong>GTO2</strong> works as <strong>LEGOs</strong>, since it allows the construction of multiple pipelines with many combinations.</p>
<p><strong>GTO2</strong> includes tools for information display, randomisation, edition, conversion, extraction, search, calculation, compression, simulation and visualisation. <strong>GTO2</strong> is prepared to deal with very large datasets, typically in the scale of Gigabytes or Terabytes (but not limited). The complete toolkit is an optimised command-line version, using the prefix <code>gto2_</code> followed by the suffix with the respective name of the program. <strong>GTO2</strong> is implemented in <strong>C</strong> language and it is available, under the MIT license, at <a href="https://github.com/cobilab/gto2" class="uri">https://github.com/cobilab/gto2</a></p>
<div id="installation" class="section level2">
<h2><span class="header-section-number">1.1</span> Installation</h2>
<p>To install <strong>GTO2</strong> through the GitHub repository:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="fu">git</span> clone https://github.com/cobilab/gto2.git
<span class="bu">cd</span> gto2/src/
<span class="fu">make</span></code></pre></div>
<p>Or by installing them directly using the Cobilab channel from Conda:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">conda</span> install -c cobilab gto2 --yes</code></pre></div>
</div>
<div id="testing" class="section level2">
<h2><span class="header-section-number">1.2</span> Testing</h2>
<p>The examples provided in this document are available in the repository. Therefore, each example can be easily reproduced, which it will also test and validate each tool. To replicate those tests, it can be done in two different ways:</p>
<ul>
<li>Running one test for a specific tool:
<ul>
<li>cd gto2/tester/gto2_{tool}</li>
<li>sh runExample.sh</li>
</ul></li>
<li>Running the batch of tests for all the tools:
<ul>
<li>cd gto2/tester/</li>
<li>sh runAllTests.sh</li>
</ul></li>
</ul>
<p>Some of these tests require internet connection to download external files and it will create new files.</p>
</div>
<div id="execution-control" class="section level2">
<h2><span class="header-section-number">1.3</span> Execution control</h2>
<p>The quality control in Unix/Linux pipelines using GTO's tools is made in three ways:</p>
<ul>
<li>Input verification: where the tools verify the format of the input file;</li>
<li>Stderr logs: Some execution errors are directly sent for the stderr channel.</li>
<li><p>Scripting validation: In complex pipelines, the verification of all the tools in the pipeline were executed properly, it is used the PIPESTATUS variable, e.g.:</p>
<div class="sourceCode"><pre class="sourceCode sh"><code class="sourceCode bash"><span class="ex">gto2_fa_rand_extra_chars</span> <span class="op">&lt;</span> input.fa <span class="kw">|</span> <span class="kw">\</span>
<span class="ex">gto2_fa_to_seq</span> <span class="op">&gt;</span> output.seq 
<span class="bu">echo</span> <span class="st">&quot;</span><span class="va">${PIPESTATUS[0]}</span><span class="st"> </span><span class="va">${PIPESTATUS[1]}</span><span class="st">&quot;</span> 
<span class="ex">0</span> 0 </code></pre></div></li>
</ul>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-afgan2018galaxy">
<p>Afgan, Enis, Dannon Baker, Bérénice Batut, Marius Van Den Beek, Dave Bouvier, Martin Čech, John Chilton, et al. 2018. “The Galaxy Platform for Accessible, Reproducible and Collaborative Biomedical Analyses: 2018 Update.” <em>Nucleic Acids Research</em> 46 (W1). Oxford University Press: W537–W544.</p>
</div>
<div id="ref-baruzzo2017simulation">
<p>Baruzzo, Giacomo, Katharina E Hayer, Eun Ji Kim, Barbara Di Camillo, Garret A FitzGerald, and Gregory R Grant. 2017. “Simulation-Based Comprehensive Benchmarking of Rna-Seq Aligners.” <em>Nature Methods</em> 14 (2). Nature Publishing Group: 135.</p>
</div>
<div id="ref-blankenberg2010manipulation">
<p>Blankenberg, Daniel, Assaf Gordon, Gregory Von Kuster, Nathan Coraor, James Taylor, Anton Nekrutenko, and Galaxy Team. 2010. “Manipulation of Fastq Data with Galaxy.” <em>Bioinformatics</em> 26 (14). Oxford University Press: 1783–5.</p>
</div>
<div id="ref-brouwer2016current">
<p>Brouwer, CPJM, Thuy Duong Vu, Miaomiao Zhou, Gianluigi Cardinali, Mick M Welling, Nathalie van de Wiele, and Vincent Robert. 2016. “Current Opportunities and Challenges of Next Generation Sequencing (Ngs) of Dna; Determining Health and Disease.” <em>British Biotechnology Journal</em> 13 (4). Sciencedomain International.</p>
</div>
<div id="ref-cock2009sanger">
<p>Cock, Peter JA, Christopher J Fields, Naohisa Goto, Michael L Heuer, and Peter M Rice. 2009. “The Sanger Fastq File Format for Sequences with Quality Scores, and the Solexa/Illumina Fastq Variants.” <em>Nucleic Acids Research</em> 38 (6). Oxford University Press: 1767–71.</p>
</div>
<div id="ref-deorowicz2015gdc">
<p>Deorowicz, Sebastian, Agnieszka Danek, and Marcin Niemiec. 2015. “GDC 2: Compression of Large Collections of Genomes.” <em>Scientific Reports</em> 5. Nature Publishing Group: 11565.</p>
</div>
<div id="ref-depristo2011framework">
<p>DePristo, Mark A, Eric Banks, Ryan Poplin, Kiran V Garimella, Jared R Maguire, Christopher Hartl, Anthony A Philippakis, et al. 2011. “A Framework for Variation Discovery and Genotyping Using Next-Generation Dna Sequencing Data.” <em>Nature Genetics</em> 43 (5). Nature Publishing Group: 491.</p>
</div>
<div id="ref-droop2016fqtools">
<p>Droop, Alastair P. 2016. “Fqtools: An Efficient Software Suite for Modern Fastq File Manipulation.” <em>Bioinformatics</em> 32 (12). Oxford University Press: 1883–4.</p>
</div>
<div id="ref-escalona2016comparison">
<p>Escalona, Merly, Sara Rocha, and David Posada. 2016. “A Comparison of Tools for the Simulation of Genomic Next-Generation Sequencing Data.” <em>Nature Reviews Genetics</em> 17 (8). Nature Publishing Group: 459.</p>
</div>
<div id="ref-goecks2010galaxy">
<p>Goecks, Jeremy, Anton Nekrutenko, and James Taylor. 2010. “Galaxy: A Comprehensive Approach for Supporting Accessible, Reproducible, and Transparent Computational Research in the Life Sciences.” <em>Genome Biology</em> 11 (8). BioMed Central: R86.</p>
</div>
<div id="ref-gordon2010fastx">
<p>Gordon, Assaf, GJ Hannon, and others. 2010. “Fastx-Toolkit.” <em>FASTQ/A Short-Reads Preprocessing Tools (Unpublished) Http://Hannonlab. Cshl. Edu/Fastx_toolkit</em> 5.</p>
</div>
<div id="ref-hernaez2019genomic">
<p>Hernaez, Mikel, Dmitri Pavlichin, Tsachy Weissman, and Idoia Ochoa. 2019. “Genomic Data Compression.” <em>Annual Review of Biomedical Data Science</em> 2. Annual Reviews.</p>
</div>
<div id="ref-hosseini2016survey">
<p>Hosseini, Morteza, Diogo Pratas, and Armando Pinho. 2016. “A Survey on Data Compression Methods for Biological Sequences.” <em>Information</em> 7 (4). Multidisciplinary Digital Publishing Institute: 56.</p>
</div>
<div id="ref-hosseini2019ac">
<p>Hosseini, Morteza, Diogo Pratas, and Armando J Pinho. 2019. “AC: A Compression Tool for Amino Acid Sequences.” <em>Interdisciplinary Sciences: Computational Life Sciences</em>. Springer, 1–9.</p>
</div>
<div id="ref-huang2011art">
<p>Huang, Weichun, Leping Li, Jason R Myers, and Gabor T Marth. 2011. “ART: A Next-Generation Sequencing Read Simulator.” <em>Bioinformatics</em> 28 (4). Oxford University Press: 593–94.</p>
</div>
<div id="ref-kumar2016mega7">
<p>Kumar, Sudhir, Glen Stecher, and Koichiro Tamura. 2016. “MEGA7: Molecular Evolutionary Genetics Analysis Version 7.0 for Bigger Datasets.” <em>Molecular Biology and Evolution</em> 33 (7). Society for Molecular Biology; Evolution: 1870–4.</p>
</div>
<div id="ref-liu2012comparison">
<p>Liu, Lin, Yinhu Li, Siliang Li, Ni Hu, Yimin He, Ray Pong, Danni Lin, Lihua Lu, and Maggie Law. 2012. “Comparison of Next-Generation Sequencing Systems.” <em>BioMed Research International</em> 2012. Hindawi Publishing Corporation.</p>
</div>
<div id="ref-liu2017high">
<p>Liu, Yuansheng, Hui Peng, Limsoon Wong, and Jinyan Li. 2017. “High-Speed and High-Ratio Referential Genome Compression.” <em>Bioinformatics</em> 33 (21). Oxford University Press: 3364–72.</p>
</div>
<div id="ref-mardis2017dna">
<p>Mardis, Elaine R. 2017. “DNA Sequencing Technologies: 2006–2016.” <em>Nature Protocols</em> 12 (2). Nature Publishing Group: 213.</p>
</div>
<div id="ref-nalbantoglu2010data">
<p>Nalbantoglu, Özkan, David Russell, and Khalid Sayood. 2010. “Data Compression Concepts and Algorithms and Their Applications to Bioinformatics.” <em>Entropy</em> 12 (1). Molecular Diversity Preservation International: 34–52.</p>
</div>
<div id="ref-ochoa2014idocomp">
<p>Ochoa, Idoia, Mikel Hernaez, and Tsachy Weissman. 2014. “IDoComp: A Compression Scheme for Assembled Genomes.” <em>Bioinformatics</em> 31 (5). Oxford University Press: 626–33.</p>
</div>
<div id="ref-pratas2018compression">
<p>Pratas, Diogo, Morteza Hosseini, and Armando J Pinho. 2018. “Compression of Amino Acid Sequences.” In <em>International Conference on Practical Applications of Computational Biology &amp; Bioinformatics</em>, 105–13. Springer.</p>
</div>
<div id="ref-pratas2016efficient">
<p>Pratas, Diogo, Armando J Pinho, and Paulo JSG Ferreira. 2016. “Efficient Compression of Genomic Sequences.” In <em>2016 Data Compression Conference (Dcc)</em>, 231–40. IEEE.</p>
</div>
<div id="ref-pratas2014xs">
<p>Pratas, Diogo, Armando J Pinho, and João MOS Rodrigues. 2014. “XS: A Fastq Read Simulator.” <em>BMC Research Notes</em> 7 (1). BioMed Central: 40.</p>
</div>
<div id="ref-shen2016seqkit">
<p>Shen, Wei, Shuai Le, Yan Li, and Fuquan Hu. 2016. “SeqKit: A Cross-Platform and Ultrafast Toolkit for Fasta/Q File Manipulation.” <em>PLoS One</em> 11 (10). Public Library of Science: e0163962.</p>
</div>
<div id="ref-tamura2011mega5">
<p>Tamura, Koichiro, Daniel Peterson, Nicholas Peterson, Glen Stecher, Masatoshi Nei, and Sudhir Kumar. 2011. “MEGA5: Molecular Evolutionary Genetics Analysis Using Maximum Likelihood, Evolutionary Distance, and Maximum Parsimony Methods.” <em>Molecular Biology and Evolution</em> 28 (10). Oxford University Press: 2731–9.</p>
</div>
<div id="ref-van2013fastq">
<p>Van der Auwera, Geraldine A, Mauricio O Carneiro, Christopher Hartl, Ryan Poplin, Guillermo Del Angel, Ami Levy-Moonshine, Tadeusz Jordan, et al. 2013. “From Fastq Data to High-Confidence Variant Calls: The Genome Analysis Toolkit Best Practices Pipeline.” <em>Current Protocols in Bioinformatics</em> 43 (1). Wiley Online Library: 11–10.</p>
</div>
<div id="ref-zhang2016overview">
<p>Zhang, Hongen. 2016. “Overview of Sequence Data Formats.” In <em>Statistical Genomics</em>, 3–17. Springer.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="fasta-tools.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["GTO2.pdf"],
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
